// @the-convocation/twitter-scraper/dist/timeline-v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseThreadedConversation = exports.parseTimelineEntryItemContentRaw = exports.parseTimelineTweetsV2 = exports.parseLegacyTweet = void 0;
const timeline_tweet_util_1 = require("./timeline-tweet-util");
const type_util_1 = require("./type-util");
function parseLegacyTweet(user, tweet) {
    if (tweet == null) {
        return {
            success: false,
            err: new Error('Tweet was not found in the timeline object.'),
        };
    }
    if (user == null) {
        return {
            success: false,
            err: new Error('User was not found in the timeline object.'),
        };
    }
    if (!tweet.id_str) {
        if (!tweet.conversation_id_str) {
            return {
                success: false,
                err: new Error('Tweet ID was not found in object.'),
            };
        }
        tweet.id_str = tweet.conversation_id_str;
    }
    const hashtags = tweet.entities?.hashtags ?? [];
    const mentions = tweet.entities?.user_mentions ?? [];
    const media = tweet.extended_entities?.media ?? [];
    const pinnedTweets = new Set(user.pinned_tweet_ids_str ?? []);
    const urls = tweet.entities?.urls ?? [];
    const { photos, videos, sensitiveContent } = (0, timeline_tweet_util_1.parseMediaGroups)(media);
    const tw = {
        conversationId: tweet.conversation_id_str,
        id: tweet.id_str,
        hashtags: hashtags
            .filter((0, type_util_1.isFieldDefined)('text'))
            .map((hashtag) => hashtag.text),
        likes: tweet.favorite_count,
        mentions: mentions.filter((0, type_util_1.isFieldDefined)('id_str')).map((mention) => ({
            id: mention.id_str,
            username: mention.screen_name,
            name: mention.name,
        })),
        name: user.name,
        permanentUrl: `https://twitter.com/${user.screen_name}/status/${tweet.id_str}`,
        photos,
        replies: tweet.reply_count,
        retweets: tweet.retweet_count,
        text: tweet.full_text,
        thread: [],
        urls: urls
            .filter((0, type_util_1.isFieldDefined)('expanded_url'))
            .map((url) => url.expanded_url),
        userId: tweet.user_id_str,
        username: user.screen_name,
        videos,
        isQuoted: false,
        isReply: false,
        isRetweet: false,
        isPin: false,
        sensitiveContent: false,
    };
    if (tweet.created_at) {
        tw.timeParsed = new Date(Date.parse(tweet.created_at));
        tw.timestamp = Math.floor(tw.timeParsed.valueOf() / 1000);
    }
    if (tweet.place?.id) {
        tw.place = tweet.place;
    }
    const quotedStatusIdStr = tweet.quoted_status_id_str;
    const inReplyToStatusIdStr = tweet.in_reply_to_status_id_str;
    const retweetedStatusIdStr = tweet.retweeted_status_id_str;
    const retweetedStatusResult = tweet.retweeted_status_result?.result;
    if (quotedStatusIdStr) {
        tw.isQuoted = true;
        tw.quotedStatusId = quotedStatusIdStr;
    }
    if (inReplyToStatusIdStr) {
        tw.isReply = true;
        tw.inReplyToStatusId = inReplyToStatusIdStr;
    }
    if (retweetedStatusIdStr || retweetedStatusResult) {
        tw.isRetweet = true;
        tw.retweetedStatusId = retweetedStatusIdStr;
        if (retweetedStatusResult) {
            const parsedResult = parseLegacyTweet(retweetedStatusResult?.core?.user_results?.result?.legacy, retweetedStatusResult?.legacy);
            if (parsedResult.success) {
                tw.retweetedStatus = parsedResult.tweet;
            }
        }
    }
    const views = parseInt(tweet.ext_views?.count ?? '');
    if (!isNaN(views)) {
        tw.views = views;
    }
    if (pinnedTweets.has(tweet.id_str)) {
        // TODO: Update tests so this can be assigned at the tweet declaration
        tw.isPin = true;
    }
    if (sensitiveContent) {
        // TODO: Update tests so this can be assigned at the tweet declaration
        tw.sensitiveContent = true;
    }
    tw.html = (0, timeline_tweet_util_1.reconstructTweetHtml)(tweet, tw.photos, tw.videos);
    return { success: true, tweet: tw };
}
exports.parseLegacyTweet = parseLegacyTweet;
function parseResult(result) {
    const noteTweetResultText = result?.note_tweet?.note_tweet_results?.result?.text;
    if (result?.legacy && noteTweetResultText) {
        result.legacy.full_text = noteTweetResultText;
    }
    const tweetResult = parseLegacyTweet(result?.core?.user_results?.result?.legacy, result?.legacy);
    if (!tweetResult.success) {
        return tweetResult;
    }
    if (!tweetResult.tweet.views && result?.views?.count) {
        const views = parseInt(result.views.count);
        if (!isNaN(views)) {
            tweetResult.tweet.views = views;
        }
    }
    const quotedResult = result?.quoted_status_result?.result;
    if (quotedResult) {
        if (quotedResult.legacy && quotedResult.rest_id) {
            quotedResult.legacy.id_str = quotedResult.rest_id;
        }
        const quotedTweetResult = parseResult(quotedResult);
        if (quotedTweetResult.success) {
            tweetResult.tweet.quotedStatus = quotedTweetResult.tweet;
        }
    }
    return tweetResult;
}
function parseTimelineTweetsV2(timeline) {
    let bottomCursor;
    let topCursor;
    const tweets = [];
    const instructions = timeline.data?.user?.result?.timeline_v2?.timeline?.instructions ?? [];
    for (const instruction of instructions) {
        const entries = instruction.entries ?? [];
        for (const entry of entries) {
            const entryContent = entry.content;
            if (!entryContent)
                continue;
            if (entryContent.cursorType === 'Bottom') {
                bottomCursor = entryContent.value;
                continue;
            }
            else if (entryContent.cursorType === 'Top') {
                topCursor = entryContent.value;
                continue;
            }
            const idStr = entry.entryId;
            if (!idStr.startsWith('tweet')) {
                continue;
            }
            if (entryContent.itemContent) {
                parseAndPush(tweets, entryContent.itemContent, idStr);
            }
        }
    }
    return { tweets, next: bottomCursor, previous: topCursor };
}
exports.parseTimelineTweetsV2 = parseTimelineTweetsV2;
function parseTimelineEntryItemContentRaw(content, entryId, isConversation = false) {
    const result = content.tweet_results?.result ?? content.tweetResult?.result;
    if (result?.__typename === 'Tweet') {
        if (result.legacy) {
            result.legacy.id_str = entryId
                .replace('conversation-', '')
                .replace('tweet-', '');
        }
        const tweetResult = parseResult(result);
        if (tweetResult.success) {
            if (isConversation) {
                if (content?.tweetDisplayType === 'SelfThread') {
                    tweetResult.tweet.isSelfThread = true;
                }
            }
            return tweetResult.tweet;
        }
    }
    return null;
}
exports.parseTimelineEntryItemContentRaw = parseTimelineEntryItemContentRaw;
function parseAndPush(tweets, content, entryId, isConversation = false) {
    const tweet = parseTimelineEntryItemContentRaw(content, entryId, isConversation);
    if (tweet) {
        tweets.push(tweet);
    }
}
function parseThreadedConversation(conversation) {
    const tweets = [];
    const instructions = conversation.data?.threaded_conversation_with_injections_v2?.instructions ??
        [];
    for (const instruction of instructions) {
        const entries = instruction.entries ?? [];
        for (const entry of entries) {
            const entryContent = entry.content?.itemContent;
            if (entryContent) {
                parseAndPush(tweets, entryContent, entry.entryId, true);
            }
            for (const item of entry.content?.items ?? []) {
                const itemContent = item.item?.content;
                if (itemContent) {
                    parseAndPush(tweets, itemContent, entry.entryId, true);
                }
            }
        }
    }
    for (const tweet of tweets) {
        if (tweet.inReplyToStatusId) {
            for (const parentTweet of tweets) {
                if (parentTweet.id === tweet.inReplyToStatusId) {
                    tweet.inReplyToStatus = parentTweet;
                    break;
                }
            }
        }
        if (tweet.isSelfThread && tweet.conversationId === tweet.id) {
            for (const childTweet of tweets) {
                if (childTweet.isSelfThread && childTweet.id !== tweet.id) {
                    tweet.thread.push(childTweet);
                }
            }
            if (tweet.thread.length === 0) {
                tweet.isSelfThread = false;
            }
        }
    }
    return tweets;
}
exports.parseThreadedConversation = parseThreadedConversation;
