// twitter-openapi-typescript-generated/dist/models/AdditionalMediaInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalMediaInfoToJSON = exports.AdditionalMediaInfoFromJSONTyped = exports.AdditionalMediaInfoFromJSON = exports.instanceOfAdditionalMediaInfo = void 0;
/**
 * Check if a given object implements the AdditionalMediaInfo interface.
 */
function instanceOfAdditionalMediaInfo(value) {
    var isInstance = true;
    isInstance = isInstance && "monetizable" in value;
    return isInstance;
}
exports.instanceOfAdditionalMediaInfo = instanceOfAdditionalMediaInfo;
function AdditionalMediaInfoFromJSON(json) {
    return AdditionalMediaInfoFromJSONTyped(json, false);
}
exports.AdditionalMediaInfoFromJSON = AdditionalMediaInfoFromJSON;
function AdditionalMediaInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'monetizable': json['monetizable'],
    };
}
exports.AdditionalMediaInfoFromJSONTyped = AdditionalMediaInfoFromJSONTyped;
function AdditionalMediaInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'monetizable': value.monetizable,
    };
}
exports.AdditionalMediaInfoToJSON = AdditionalMediaInfoToJSON;
