// twitter-openapi-typescript-generated/dist/models/BirdwatchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BirdwatchEntityToJSON = exports.BirdwatchEntityFromJSONTyped = exports.BirdwatchEntityFromJSON = exports.instanceOfBirdwatchEntity = void 0;
var BirdwatchEntityRef_1 = require("./BirdwatchEntityRef");
/**
 * Check if a given object implements the BirdwatchEntity interface.
 */
function instanceOfBirdwatchEntity(value) {
    var isInstance = true;
    isInstance = isInstance && "fromIndex" in value;
    isInstance = isInstance && "ref" in value;
    isInstance = isInstance && "toIndex" in value;
    return isInstance;
}
exports.instanceOfBirdwatchEntity = instanceOfBirdwatchEntity;
function BirdwatchEntityFromJSON(json) {
    return BirdwatchEntityFromJSONTyped(json, false);
}
exports.BirdwatchEntityFromJSON = BirdwatchEntityFromJSON;
function BirdwatchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fromIndex': json['fromIndex'],
        'ref': (0, BirdwatchEntityRef_1.BirdwatchEntityRefFromJSON)(json['ref']),
        'toIndex': json['toIndex'],
    };
}
exports.BirdwatchEntityFromJSONTyped = BirdwatchEntityFromJSONTyped;
function BirdwatchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fromIndex': value.fromIndex,
        'ref': (0, BirdwatchEntityRef_1.BirdwatchEntityRefToJSON)(value.ref),
        'toIndex': value.toIndex,
    };
}
exports.BirdwatchEntityToJSON = BirdwatchEntityToJSON;
