// twitter-openapi-typescript-generated/dist/models/BirdwatchPivotNote.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BirdwatchPivotNoteToJSON = exports.BirdwatchPivotNoteFromJSONTyped = exports.BirdwatchPivotNoteFromJSON = exports.instanceOfBirdwatchPivotNote = void 0;
/**
 * Check if a given object implements the BirdwatchPivotNote interface.
 */
function instanceOfBirdwatchPivotNote(value) {
    var isInstance = true;
    isInstance = isInstance && "restId" in value;
    return isInstance;
}
exports.instanceOfBirdwatchPivotNote = instanceOfBirdwatchPivotNote;
function BirdwatchPivotNoteFromJSON(json) {
    return BirdwatchPivotNoteFromJSONTyped(json, false);
}
exports.BirdwatchPivotNoteFromJSON = BirdwatchPivotNoteFromJSON;
function BirdwatchPivotNoteFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'restId': json['rest_id'],
    };
}
exports.BirdwatchPivotNoteFromJSONTyped = BirdwatchPivotNoteFromJSONTyped;
function BirdwatchPivotNoteToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'rest_id': value.restId,
    };
}
exports.BirdwatchPivotNoteToJSON = BirdwatchPivotNoteToJSON;
