// twitter-openapi-typescript-generated/dist/models/BookmarksResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookmarksResponseDataToJSON = exports.BookmarksResponseDataFromJSONTyped = exports.BookmarksResponseDataFromJSON = exports.instanceOfBookmarksResponseData = void 0;
var BookmarksTimeline_1 = require("./BookmarksTimeline");
/**
 * Check if a given object implements the BookmarksResponseData interface.
 */
function instanceOfBookmarksResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "bookmarkTimelineV2" in value;
    return isInstance;
}
exports.instanceOfBookmarksResponseData = instanceOfBookmarksResponseData;
function BookmarksResponseDataFromJSON(json) {
    return BookmarksResponseDataFromJSONTyped(json, false);
}
exports.BookmarksResponseDataFromJSON = BookmarksResponseDataFromJSON;
function BookmarksResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'bookmarkTimelineV2': (0, BookmarksTimeline_1.BookmarksTimelineFromJSON)(json['bookmark_timeline_v2']),
    };
}
exports.BookmarksResponseDataFromJSONTyped = BookmarksResponseDataFromJSONTyped;
function BookmarksResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'bookmark_timeline_v2': (0, BookmarksTimeline_1.BookmarksTimelineToJSON)(value.bookmarkTimelineV2),
    };
}
exports.BookmarksResponseDataToJSON = BookmarksResponseDataToJSON;
