// twitter-openapi-typescript-generated/dist/models/BookmarksResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookmarksResponseToJSON = exports.BookmarksResponseFromJSONTyped = exports.BookmarksResponseFromJSON = exports.instanceOfBookmarksResponse = void 0;
var BookmarksResponseData_1 = require("./BookmarksResponseData");
/**
 * Check if a given object implements the BookmarksResponse interface.
 */
function instanceOfBookmarksResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfBookmarksResponse = instanceOfBookmarksResponse;
function BookmarksResponseFromJSON(json) {
    return BookmarksResponseFromJSONTyped(json, false);
}
exports.BookmarksResponseFromJSON = BookmarksResponseFromJSON;
function BookmarksResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, BookmarksResponseData_1.BookmarksResponseDataFromJSON)(json['data']),
    };
}
exports.BookmarksResponseFromJSONTyped = BookmarksResponseFromJSONTyped;
function BookmarksResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, BookmarksResponseData_1.BookmarksResponseDataToJSON)(value.data),
    };
}
exports.BookmarksResponseToJSON = BookmarksResponseToJSON;
