// twitter-openapi-typescript-generated/dist/models/Callback.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallbackToJSON = exports.CallbackFromJSONTyped = exports.CallbackFromJSON = exports.instanceOfCallback = void 0;
/**
 * Check if a given object implements the Callback interface.
 */
function instanceOfCallback(value) {
    var isInstance = true;
    isInstance = isInstance && "endpoint" in value;
    return isInstance;
}
exports.instanceOfCallback = instanceOfCallback;
function CallbackFromJSON(json) {
    return CallbackFromJSONTyped(json, false);
}
exports.CallbackFromJSON = CallbackFromJSON;
function CallbackFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'endpoint': json['endpoint'],
    };
}
exports.CallbackFromJSONTyped = CallbackFromJSONTyped;
function CallbackToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'endpoint': value.endpoint,
    };
}
exports.CallbackToJSON = CallbackToJSON;
