// twitter-openapi-typescript-generated/dist/models/ClientEventInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientEventInfoToJSON = exports.ClientEventInfoFromJSONTyped = exports.ClientEventInfoFromJSON = exports.instanceOfClientEventInfo = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ClientEventInfo interface.
 */
function instanceOfClientEventInfo(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfClientEventInfo = instanceOfClientEventInfo;
function ClientEventInfoFromJSON(json) {
    return ClientEventInfoFromJSONTyped(json, false);
}
exports.ClientEventInfoFromJSON = ClientEventInfoFromJSON;
function ClientEventInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'component': !(0, runtime_1.exists)(json, 'component') ? undefined : json['component'],
        'details': !(0, runtime_1.exists)(json, 'details') ? undefined : json['details'],
        'element': !(0, runtime_1.exists)(json, 'element') ? undefined : json['element'],
    };
}
exports.ClientEventInfoFromJSONTyped = ClientEventInfoFromJSONTyped;
function ClientEventInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'component': value.component,
        'details': value.details,
        'element': value.element,
    };
}
exports.ClientEventInfoToJSON = ClientEventInfoToJSON;
