// twitter-openapi-typescript-generated/dist/models/CommunitiesActions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunitiesActionsToJSON = exports.CommunitiesActionsFromJSONTyped = exports.CommunitiesActionsFromJSON = exports.instanceOfCommunitiesActions = void 0;
/**
 * Check if a given object implements the CommunitiesActions interface.
 */
function instanceOfCommunitiesActions(value) {
    var isInstance = true;
    isInstance = isInstance && "create" in value;
    return isInstance;
}
exports.instanceOfCommunitiesActions = instanceOfCommunitiesActions;
function CommunitiesActionsFromJSON(json) {
    return CommunitiesActionsFromJSONTyped(json, false);
}
exports.CommunitiesActionsFromJSON = CommunitiesActionsFromJSON;
function CommunitiesActionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'create': json['create'],
    };
}
exports.CommunitiesActionsFromJSONTyped = CommunitiesActionsFromJSONTyped;
function CommunitiesActionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'create': value.create,
    };
}
exports.CommunitiesActionsToJSON = CommunitiesActionsToJSON;
