// twitter-openapi-typescript-generated/dist/models/ContentEntryType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentEntryTypeToJSON = exports.ContentEntryTypeFromJSONTyped = exports.ContentEntryTypeFromJSON = exports.ContentEntryType = void 0;
/**
 *
 * @export
 */
exports.ContentEntryType = {
    TimelineTimelineItem: 'TimelineTimelineItem',
    TimelineTimelineCursor: 'TimelineTimelineCursor',
    TimelineTimelineModule: 'TimelineTimelineModule'
};
function ContentEntryTypeFromJSON(json) {
    return ContentEntryTypeFromJSONTyped(json, false);
}
exports.ContentEntryTypeFromJSON = ContentEntryTypeFromJSON;
function ContentEntryTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ContentEntryTypeFromJSONTyped = ContentEntryTypeFromJSONTyped;
function ContentEntryTypeToJSON(value) {
    return value;
}
exports.ContentEntryTypeToJSON = ContentEntryTypeToJSON;
