// twitter-openapi-typescript-generated/dist/models/ContentItemType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentItemTypeToJSON = exports.ContentItemTypeFromJSONTyped = exports.ContentItemTypeFromJSON = exports.ContentItemType = void 0;
/**
 *
 * @export
 */
exports.ContentItemType = {
    TimelineTweet: 'TimelineTweet',
    TimelineTimelineCursor: 'TimelineTimelineCursor',
    TimelineUser: 'TimelineUser',
    TimelinePrompt: 'TimelinePrompt',
    TimelineMessagePrompt: 'TimelineMessagePrompt',
    TimelineCommunity: 'TimelineCommunity'
};
function ContentItemTypeFromJSON(json) {
    return ContentItemTypeFromJSONTyped(json, false);
}
exports.ContentItemTypeFromJSON = ContentItemTypeFromJSON;
function ContentItemTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ContentItemTypeFromJSONTyped = ContentItemTypeFromJSONTyped;
function ContentItemTypeToJSON(value) {
    return value;
}
exports.ContentItemTypeToJSON = ContentItemTypeToJSON;
