// twitter-openapi-typescript-generated/dist/models/CreateTweetResponseResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTweetResponseResultToJSON = exports.CreateTweetResponseResultFromJSONTyped = exports.CreateTweetResponseResultFromJSON = exports.instanceOfCreateTweetResponseResult = void 0;
var CreateTweet_1 = require("./CreateTweet");
/**
 * Check if a given object implements the CreateTweetResponseResult interface.
 */
function instanceOfCreateTweetResponseResult(value) {
    var isInstance = true;
    isInstance = isInstance && "tweetResults" in value;
    return isInstance;
}
exports.instanceOfCreateTweetResponseResult = instanceOfCreateTweetResponseResult;
function CreateTweetResponseResultFromJSON(json) {
    return CreateTweetResponseResultFromJSONTyped(json, false);
}
exports.CreateTweetResponseResultFromJSON = CreateTweetResponseResultFromJSON;
function CreateTweetResponseResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tweetResults': (0, CreateTweet_1.CreateTweetFromJSON)(json['tweet_results']),
    };
}
exports.CreateTweetResponseResultFromJSONTyped = CreateTweetResponseResultFromJSONTyped;
function CreateTweetResponseResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'tweet_results': (0, CreateTweet_1.CreateTweetToJSON)(value.tweetResults),
    };
}
exports.CreateTweetResponseResultToJSON = CreateTweetResponseResultToJSON;
