// twitter-openapi-typescript-generated/dist/models/CreateTweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTweetResponseDataToJSON = exports.CreateTweetResponseDataFromJSONTyped = exports.CreateTweetResponseDataFromJSON = exports.instanceOfCreateTweetResponseData = void 0;
var CreateTweetResponseResult_1 = require("./CreateTweetResponseResult");
/**
 * Check if a given object implements the CreateTweetResponseData interface.
 */
function instanceOfCreateTweetResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "createTweet" in value;
    return isInstance;
}
exports.instanceOfCreateTweetResponseData = instanceOfCreateTweetResponseData;
function CreateTweetResponseDataFromJSON(json) {
    return CreateTweetResponseDataFromJSONTyped(json, false);
}
exports.CreateTweetResponseDataFromJSON = CreateTweetResponseDataFromJSON;
function CreateTweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'createTweet': (0, CreateTweetResponseResult_1.CreateTweetResponseResultFromJSON)(json['create_tweet']),
    };
}
exports.CreateTweetResponseDataFromJSONTyped = CreateTweetResponseDataFromJSONTyped;
function CreateTweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'create_tweet': (0, CreateTweetResponseResult_1.CreateTweetResponseResultToJSON)(value.createTweet),
    };
}
exports.CreateTweetResponseDataToJSON = CreateTweetResponseDataToJSON;
