// twitter-openapi-typescript-generated/dist/models/CreateRetweetResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRetweetResponseToJSON = exports.CreateRetweetResponseFromJSONTyped = exports.CreateRetweetResponseFromJSON = exports.instanceOfCreateRetweetResponse = void 0;
var CreateRetweetResponseData_1 = require("./CreateRetweetResponseData");
/**
 * Check if a given object implements the CreateRetweetResponse interface.
 */
function instanceOfCreateRetweetResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfCreateRetweetResponse = instanceOfCreateRetweetResponse;
function CreateRetweetResponseFromJSON(json) {
    return CreateRetweetResponseFromJSONTyped(json, false);
}
exports.CreateRetweetResponseFromJSON = CreateRetweetResponseFromJSON;
function CreateRetweetResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, CreateRetweetResponseData_1.CreateRetweetResponseDataFromJSON)(json['data']),
    };
}
exports.CreateRetweetResponseFromJSONTyped = CreateRetweetResponseFromJSONTyped;
function CreateRetweetResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, CreateRetweetResponseData_1.CreateRetweetResponseDataToJSON)(value.data),
    };
}
exports.CreateRetweetResponseToJSON = CreateRetweetResponseToJSON;
