// twitter-openapi-typescript-generated/dist/models/CreateRetweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRetweetToJSON = exports.CreateRetweetFromJSONTyped = exports.CreateRetweetFromJSON = exports.instanceOfCreateRetweet = void 0;
var Retweet_1 = require("./Retweet");
/**
 * Check if a given object implements the CreateRetweet interface.
 */
function instanceOfCreateRetweet(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfCreateRetweet = instanceOfCreateRetweet;
function CreateRetweetFromJSON(json) {
    return CreateRetweetFromJSONTyped(json, false);
}
exports.CreateRetweetFromJSON = CreateRetweetFromJSON;
function CreateRetweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, Retweet_1.RetweetFromJSON)(json['result']),
    };
}
exports.CreateRetweetFromJSONTyped = CreateRetweetFromJSONTyped;
function CreateRetweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, Retweet_1.RetweetToJSON)(value.result),
    };
}
exports.CreateRetweetToJSON = CreateRetweetToJSON;
