// twitter-openapi-typescript-generated/dist/models/CreateTweetResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTweetResponseToJSON = exports.CreateTweetResponseFromJSONTyped = exports.CreateTweetResponseFromJSON = exports.instanceOfCreateTweetResponse = void 0;
var CreateTweetResponseData_1 = require("./CreateTweetResponseData");
/**
 * Check if a given object implements the CreateTweetResponse interface.
 */
function instanceOfCreateTweetResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfCreateTweetResponse = instanceOfCreateTweetResponse;
function CreateTweetResponseFromJSON(json) {
    return CreateTweetResponseFromJSONTyped(json, false);
}
exports.CreateTweetResponseFromJSON = CreateTweetResponseFromJSON;
function CreateTweetResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, CreateTweetResponseData_1.CreateTweetResponseDataFromJSON)(json['data']),
    };
}
exports.CreateTweetResponseFromJSONTyped = CreateTweetResponseFromJSONTyped;
function CreateTweetResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, CreateTweetResponseData_1.CreateTweetResponseDataToJSON)(value.data),
    };
}
exports.CreateTweetResponseToJSON = CreateTweetResponseToJSON;
