// twitter-openapi-typescript-generated/dist/models/CreateRetweetResponseResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRetweetResponseResultToJSON = exports.CreateRetweetResponseResultFromJSONTyped = exports.CreateRetweetResponseResultFromJSON = exports.instanceOfCreateRetweetResponseResult = void 0;
var CreateRetweet_1 = require("./CreateRetweet");
/**
 * Check if a given object implements the CreateRetweetResponseResult interface.
 */
function instanceOfCreateRetweetResponseResult(value) {
    var isInstance = true;
    isInstance = isInstance && "retweetResults" in value;
    return isInstance;
}
exports.instanceOfCreateRetweetResponseResult = instanceOfCreateRetweetResponseResult;
function CreateRetweetResponseResultFromJSON(json) {
    return CreateRetweetResponseResultFromJSONTyped(json, false);
}
exports.CreateRetweetResponseResultFromJSON = CreateRetweetResponseResultFromJSON;
function CreateRetweetResponseResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'retweetResults': (0, CreateRetweet_1.CreateRetweetFromJSON)(json['retweet_results']),
    };
}
exports.CreateRetweetResponseResultFromJSONTyped = CreateRetweetResponseResultFromJSONTyped;
function CreateRetweetResponseResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'retweet_results': (0, CreateRetweet_1.CreateRetweetToJSON)(value.retweetResults),
    };
}
exports.CreateRetweetResponseResultToJSON = CreateRetweetResponseResultToJSON;
