// twitter-openapi-typescript-generated/dist/models/CreateTweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTweetToJSON = exports.CreateTweetFromJSONTyped = exports.CreateTweetFromJSON = exports.instanceOfCreateTweet = void 0;
var Tweet_1 = require("./Tweet");
/**
 * Check if a given object implements the CreateTweet interface.
 */
function instanceOfCreateTweet(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfCreateTweet = instanceOfCreateTweet;
function CreateTweetFromJSON(json) {
    return CreateTweetFromJSONTyped(json, false);
}
exports.CreateTweetFromJSON = CreateTweetFromJSON;
function CreateTweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, Tweet_1.TweetFromJSON)(json['result']),
    };
}
exports.CreateTweetFromJSONTyped = CreateTweetFromJSONTyped;
function CreateTweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, Tweet_1.TweetToJSON)(value.result),
    };
}
exports.CreateTweetToJSON = CreateTweetToJSON;
