// twitter-openapi-typescript-generated/dist/models/CreateRetweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRetweetResponseDataToJSON = exports.CreateRetweetResponseDataFromJSONTyped = exports.CreateRetweetResponseDataFromJSON = exports.instanceOfCreateRetweetResponseData = void 0;
var CreateRetweetResponseResult_1 = require("./CreateRetweetResponseResult");
/**
 * Check if a given object implements the CreateRetweetResponseData interface.
 */
function instanceOfCreateRetweetResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "createRetweet" in value;
    return isInstance;
}
exports.instanceOfCreateRetweetResponseData = instanceOfCreateRetweetResponseData;
function CreateRetweetResponseDataFromJSON(json) {
    return CreateRetweetResponseDataFromJSONTyped(json, false);
}
exports.CreateRetweetResponseDataFromJSON = CreateRetweetResponseDataFromJSON;
function CreateRetweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'createRetweet': (0, CreateRetweetResponseResult_1.CreateRetweetResponseResultFromJSON)(json['create_retweet']),
    };
}
exports.CreateRetweetResponseDataFromJSONTyped = CreateRetweetResponseDataFromJSONTyped;
function CreateRetweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'create_retweet': (0, CreateRetweetResponseResult_1.CreateRetweetResponseResultToJSON)(value.createRetweet),
    };
}
exports.CreateRetweetResponseDataToJSON = CreateRetweetResponseDataToJSON;
