// twitter-openapi-typescript-generated/dist/models/CtaClientEventInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CtaClientEventInfoToJSON = exports.CtaClientEventInfoFromJSONTyped = exports.CtaClientEventInfoFromJSON = exports.instanceOfCtaClientEventInfo = exports.CtaClientEventInfoActionEnum = void 0;
/**
 * @export
 */
exports.CtaClientEventInfoActionEnum = {
    PrimaryCta: 'primary_cta'
};
/**
 * Check if a given object implements the CtaClientEventInfo interface.
 */
function instanceOfCtaClientEventInfo(value) {
    var isInstance = true;
    isInstance = isInstance && "action" in value;
    return isInstance;
}
exports.instanceOfCtaClientEventInfo = instanceOfCtaClientEventInfo;
function CtaClientEventInfoFromJSON(json) {
    return CtaClientEventInfoFromJSONTyped(json, false);
}
exports.CtaClientEventInfoFromJSON = CtaClientEventInfoFromJSON;
function CtaClientEventInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'action': json['action'],
    };
}
exports.CtaClientEventInfoFromJSONTyped = CtaClientEventInfoFromJSONTyped;
function CtaClientEventInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'action': value.action,
    };
}
exports.CtaClientEventInfoToJSON = CtaClientEventInfoToJSON;
