// twitter-openapi-typescript-generated/dist/models/CursorType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CursorTypeToJSON = exports.CursorTypeFromJSONTyped = exports.CursorTypeFromJSON = exports.CursorType = void 0;
/**
 *
 * @export
 */
exports.CursorType = {
    Top: 'Top',
    Bottom: 'Bottom',
    ShowMore: 'ShowMore',
    ShowMoreThreads: 'ShowMoreThreads',
    Gap: 'Gap'
};
function CursorTypeFromJSON(json) {
    return CursorTypeFromJSONTyped(json, false);
}
exports.CursorTypeFromJSON = CursorTypeFromJSON;
function CursorTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.CursorTypeFromJSONTyped = CursorTypeFromJSONTyped;
function CursorTypeToJSON(value) {
    return value;
}
exports.CursorTypeToJSON = CursorTypeToJSON;
