// twitter-openapi-typescript-generated/dist/models/DeleteTweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteTweetResponseDataToJSON = exports.DeleteTweetResponseDataFromJSONTyped = exports.DeleteTweetResponseDataFromJSON = exports.instanceOfDeleteTweetResponseData = void 0;
var runtime_1 = require("../runtime");
var DeleteTweetResponseResult_1 = require("./DeleteTweetResponseResult");
/**
 * Check if a given object implements the DeleteTweetResponseData interface.
 */
function instanceOfDeleteTweetResponseData(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfDeleteTweetResponseData = instanceOfDeleteTweetResponseData;
function DeleteTweetResponseDataFromJSON(json) {
    return DeleteTweetResponseDataFromJSONTyped(json, false);
}
exports.DeleteTweetResponseDataFromJSON = DeleteTweetResponseDataFromJSON;
function DeleteTweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deleteRetweet': !(0, runtime_1.exists)(json, 'delete_retweet') ? undefined : (0, DeleteTweetResponseResult_1.DeleteTweetResponseResultFromJSON)(json['delete_retweet']),
    };
}
exports.DeleteTweetResponseDataFromJSONTyped = DeleteTweetResponseDataFromJSONTyped;
function DeleteTweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'delete_retweet': (0, DeleteTweetResponseResult_1.DeleteTweetResponseResultToJSON)(value.deleteRetweet),
    };
}
exports.DeleteTweetResponseDataToJSON = DeleteTweetResponseDataToJSON;
