// twitter-openapi-typescript-generated/dist/models/DeleteTweetResponseResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteTweetResponseResultToJSON = exports.DeleteTweetResponseResultFromJSONTyped = exports.DeleteTweetResponseResultFromJSON = exports.instanceOfDeleteTweetResponseResult = void 0;
/**
 * Check if a given object implements the DeleteTweetResponseResult interface.
 */
function instanceOfDeleteTweetResponseResult(value) {
    var isInstance = true;
    isInstance = isInstance && "tweetResults" in value;
    return isInstance;
}
exports.instanceOfDeleteTweetResponseResult = instanceOfDeleteTweetResponseResult;
function DeleteTweetResponseResultFromJSON(json) {
    return DeleteTweetResponseResultFromJSONTyped(json, false);
}
exports.DeleteTweetResponseResultFromJSON = DeleteTweetResponseResultFromJSON;
function DeleteTweetResponseResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tweetResults': json['tweet_results'],
    };
}
exports.DeleteTweetResponseResultFromJSONTyped = DeleteTweetResponseResultFromJSONTyped;
function DeleteTweetResponseResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'tweet_results': value.tweetResults,
    };
}
exports.DeleteTweetResponseResultToJSON = DeleteTweetResponseResultToJSON;
