// twitter-openapi-typescript-generated/dist/models/DeleteRetweetResponseResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRetweetResponseResultToJSON = exports.DeleteRetweetResponseResultFromJSONTyped = exports.DeleteRetweetResponseResultFromJSON = exports.instanceOfDeleteRetweetResponseResult = void 0;
var runtime_1 = require("../runtime");
var DeleteRetweet_1 = require("./DeleteRetweet");
/**
 * Check if a given object implements the DeleteRetweetResponseResult interface.
 */
function instanceOfDeleteRetweetResponseResult(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfDeleteRetweetResponseResult = instanceOfDeleteRetweetResponseResult;
function DeleteRetweetResponseResultFromJSON(json) {
    return DeleteRetweetResponseResultFromJSONTyped(json, false);
}
exports.DeleteRetweetResponseResultFromJSON = DeleteRetweetResponseResultFromJSON;
function DeleteRetweetResponseResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'retweetResults': !(0, runtime_1.exists)(json, 'retweet_results') ? undefined : (0, DeleteRetweet_1.DeleteRetweetFromJSON)(json['retweet_results']),
    };
}
exports.DeleteRetweetResponseResultFromJSONTyped = DeleteRetweetResponseResultFromJSONTyped;
function DeleteRetweetResponseResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'retweet_results': (0, DeleteRetweet_1.DeleteRetweetToJSON)(value.retweetResults),
    };
}
exports.DeleteRetweetResponseResultToJSON = DeleteRetweetResponseResultToJSON;
