// twitter-openapi-typescript-generated/dist/models/DeleteRetweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRetweetToJSON = exports.DeleteRetweetFromJSONTyped = exports.DeleteRetweetFromJSON = exports.instanceOfDeleteRetweet = void 0;
var Retweet_1 = require("./Retweet");
/**
 * Check if a given object implements the DeleteRetweet interface.
 */
function instanceOfDeleteRetweet(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfDeleteRetweet = instanceOfDeleteRetweet;
function DeleteRetweetFromJSON(json) {
    return DeleteRetweetFromJSONTyped(json, false);
}
exports.DeleteRetweetFromJSON = DeleteRetweetFromJSON;
function DeleteRetweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (json['result'].map(Retweet_1.RetweetFromJSON)),
    };
}
exports.DeleteRetweetFromJSONTyped = DeleteRetweetFromJSONTyped;
function DeleteRetweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (value.result.map(Retweet_1.RetweetToJSON)),
    };
}
exports.DeleteRetweetToJSON = DeleteRetweetToJSON;
