// twitter-openapi-typescript-generated/dist/models/DeleteTweetResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteTweetResponseToJSON = exports.DeleteTweetResponseFromJSONTyped = exports.DeleteTweetResponseFromJSON = exports.instanceOfDeleteTweetResponse = void 0;
var DeleteTweetResponseData_1 = require("./DeleteTweetResponseData");
/**
 * Check if a given object implements the DeleteTweetResponse interface.
 */
function instanceOfDeleteTweetResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfDeleteTweetResponse = instanceOfDeleteTweetResponse;
function DeleteTweetResponseFromJSON(json) {
    return DeleteTweetResponseFromJSONTyped(json, false);
}
exports.DeleteTweetResponseFromJSON = DeleteTweetResponseFromJSON;
function DeleteTweetResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, DeleteTweetResponseData_1.DeleteTweetResponseDataFromJSON)(json['data']),
    };
}
exports.DeleteTweetResponseFromJSONTyped = DeleteTweetResponseFromJSONTyped;
function DeleteTweetResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, DeleteTweetResponseData_1.DeleteTweetResponseDataToJSON)(value.data),
    };
}
exports.DeleteTweetResponseToJSON = DeleteTweetResponseToJSON;
