// twitter-openapi-typescript-generated/dist/models/DeleteRetweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRetweetResponseDataToJSON = exports.DeleteRetweetResponseDataFromJSONTyped = exports.DeleteRetweetResponseDataFromJSON = exports.instanceOfDeleteRetweetResponseData = void 0;
var runtime_1 = require("../runtime");
var CreateRetweetResponseResult_1 = require("./CreateRetweetResponseResult");
/**
 * Check if a given object implements the DeleteRetweetResponseData interface.
 */
function instanceOfDeleteRetweetResponseData(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfDeleteRetweetResponseData = instanceOfDeleteRetweetResponseData;
function DeleteRetweetResponseDataFromJSON(json) {
    return DeleteRetweetResponseDataFromJSONTyped(json, false);
}
exports.DeleteRetweetResponseDataFromJSON = DeleteRetweetResponseDataFromJSON;
function DeleteRetweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'createRetweet': !(0, runtime_1.exists)(json, 'create_retweet') ? undefined : (0, CreateRetweetResponseResult_1.CreateRetweetResponseResultFromJSON)(json['create_retweet']),
    };
}
exports.DeleteRetweetResponseDataFromJSONTyped = DeleteRetweetResponseDataFromJSONTyped;
function DeleteRetweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'create_retweet': (0, CreateRetweetResponseResult_1.CreateRetweetResponseResultToJSON)(value.createRetweet),
    };
}
exports.DeleteRetweetResponseDataToJSON = DeleteRetweetResponseDataToJSON;
