// twitter-openapi-typescript-generated/dist/models/DeleteRetweetResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteRetweetResponseToJSON = exports.DeleteRetweetResponseFromJSONTyped = exports.DeleteRetweetResponseFromJSON = exports.instanceOfDeleteRetweetResponse = void 0;
var DeleteRetweetResponseData_1 = require("./DeleteRetweetResponseData");
/**
 * Check if a given object implements the DeleteRetweetResponse interface.
 */
function instanceOfDeleteRetweetResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfDeleteRetweetResponse = instanceOfDeleteRetweetResponse;
function DeleteRetweetResponseFromJSON(json) {
    return DeleteRetweetResponseFromJSONTyped(json, false);
}
exports.DeleteRetweetResponseFromJSON = DeleteRetweetResponseFromJSON;
function DeleteRetweetResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, DeleteRetweetResponseData_1.DeleteRetweetResponseDataFromJSON)(json['data']),
    };
}
exports.DeleteRetweetResponseFromJSONTyped = DeleteRetweetResponseFromJSONTyped;
function DeleteRetweetResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, DeleteRetweetResponseData_1.DeleteRetweetResponseDataToJSON)(value.data),
    };
}
exports.DeleteRetweetResponseToJSON = DeleteRetweetResponseToJSON;
