// twitter-openapi-typescript-generated/dist/models/Errors.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorsToJSON = exports.ErrorsFromJSONTyped = exports.ErrorsFromJSON = exports.instanceOfErrors = void 0;
/**
 * Check if a given object implements the Errors interface.
 */
function instanceOfErrors(value) {
    var isInstance = true;
    isInstance = isInstance && "errors" in value;
    return isInstance;
}
exports.instanceOfErrors = instanceOfErrors;
function ErrorsFromJSON(json) {
    return ErrorsFromJSONTyped(json, false);
}
exports.ErrorsFromJSON = ErrorsFromJSON;
function ErrorsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errors': json['errors'],
    };
}
exports.ErrorsFromJSONTyped = ErrorsFromJSONTyped;
function ErrorsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'errors': value.errors,
    };
}
exports.ErrorsToJSON = ErrorsToJSON;
