// twitter-openapi-typescript-generated/dist/models/ErrorExtensions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorExtensionsToJSON = exports.ErrorExtensionsFromJSONTyped = exports.ErrorExtensionsFromJSON = exports.instanceOfErrorExtensions = void 0;
var Tracing_1 = require("./Tracing");
/**
 * Check if a given object implements the ErrorExtensions interface.
 */
function instanceOfErrorExtensions(value) {
    var isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "kind" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "retryAfter" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "tracing" in value;
    return isInstance;
}
exports.instanceOfErrorExtensions = instanceOfErrorExtensions;
function ErrorExtensionsFromJSON(json) {
    return ErrorExtensionsFromJSONTyped(json, false);
}
exports.ErrorExtensionsFromJSON = ErrorExtensionsFromJSON;
function ErrorExtensionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'kind': json['kind'],
        'name': json['name'],
        'retryAfter': json['retry_after'],
        'source': json['source'],
        'tracing': (0, Tracing_1.TracingFromJSON)(json['tracing']),
    };
}
exports.ErrorExtensionsFromJSONTyped = ErrorExtensionsFromJSONTyped;
function ErrorExtensionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'kind': value.kind,
        'name': value.name,
        'retry_after': value.retryAfter,
        'source': value.source,
        'tracing': (0, Tracing_1.TracingToJSON)(value.tracing),
    };
}
exports.ErrorExtensionsToJSON = ErrorExtensionsToJSON;
