// twitter-openapi-typescript-generated/dist/models/ExtendedEntities.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtendedEntitiesToJSON = exports.ExtendedEntitiesFromJSONTyped = exports.ExtendedEntitiesFromJSON = exports.instanceOfExtendedEntities = void 0;
var MediaExtended_1 = require("./MediaExtended");
/**
 * Check if a given object implements the ExtendedEntities interface.
 */
function instanceOfExtendedEntities(value) {
    var isInstance = true;
    isInstance = isInstance && "media" in value;
    return isInstance;
}
exports.instanceOfExtendedEntities = instanceOfExtendedEntities;
function ExtendedEntitiesFromJSON(json) {
    return ExtendedEntitiesFromJSONTyped(json, false);
}
exports.ExtendedEntitiesFromJSON = ExtendedEntitiesFromJSON;
function ExtendedEntitiesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'media': (json['media'].map(MediaExtended_1.MediaExtendedFromJSON)),
    };
}
exports.ExtendedEntitiesFromJSONTyped = ExtendedEntitiesFromJSONTyped;
function ExtendedEntitiesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'media': (value.media.map(MediaExtended_1.MediaExtendedToJSON)),
    };
}
exports.ExtendedEntitiesToJSON = ExtendedEntitiesToJSON;
