// twitter-openapi-typescript-generated/dist/models/Extensions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionsToJSON = exports.ExtensionsFromJSONTyped = exports.ExtensionsFromJSON = exports.instanceOfExtensions = void 0;
var Tracing_1 = require("./Tracing");
/**
 * Check if a given object implements the Extensions interface.
 */
function instanceOfExtensions(value) {
    var isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "kind" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "source" in value;
    isInstance = isInstance && "tracing" in value;
    return isInstance;
}
exports.instanceOfExtensions = instanceOfExtensions;
function ExtensionsFromJSON(json) {
    return ExtensionsFromJSONTyped(json, false);
}
exports.ExtensionsFromJSON = ExtensionsFromJSON;
function ExtensionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'kind': json['kind'],
        'name': json['name'],
        'source': json['source'],
        'tracing': (0, Tracing_1.TracingFromJSON)(json['tracing']),
    };
}
exports.ExtensionsFromJSONTyped = ExtensionsFromJSONTyped;
function ExtensionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'kind': value.kind,
        'name': value.name,
        'source': value.source,
        'tracing': (0, Tracing_1.TracingToJSON)(value.tracing),
    };
}
exports.ExtensionsToJSON = ExtensionsToJSON;
