// twitter-openapi-typescript-generated/dist/models/ExtMediaAvailability.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtMediaAvailabilityToJSON = exports.ExtMediaAvailabilityFromJSONTyped = exports.ExtMediaAvailabilityFromJSON = exports.instanceOfExtMediaAvailability = exports.ExtMediaAvailabilityStatusEnum = void 0;
var runtime_1 = require("../runtime");
/**
 * @export
 */
exports.ExtMediaAvailabilityStatusEnum = {
    Available: 'Available',
    Unavailable: 'Unavailable'
};
/**
 * Check if a given object implements the ExtMediaAvailability interface.
 */
function instanceOfExtMediaAvailability(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfExtMediaAvailability = instanceOfExtMediaAvailability;
function ExtMediaAvailabilityFromJSON(json) {
    return ExtMediaAvailabilityFromJSONTyped(json, false);
}
exports.ExtMediaAvailabilityFromJSON = ExtMediaAvailabilityFromJSON;
function ExtMediaAvailabilityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'reason': !(0, runtime_1.exists)(json, 'reason') ? undefined : json['reason'],
        'status': !(0, runtime_1.exists)(json, 'status') ? undefined : json['status'],
    };
}
exports.ExtMediaAvailabilityFromJSONTyped = ExtMediaAvailabilityFromJSONTyped;
function ExtMediaAvailabilityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'reason': value.reason,
        'status': value.status,
    };
}
exports.ExtMediaAvailabilityToJSON = ExtMediaAvailabilityToJSON;
