// twitter-openapi-typescript-generated/dist/models/FavoriteTweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FavoriteTweetToJSON = exports.FavoriteTweetFromJSONTyped = exports.FavoriteTweetFromJSON = exports.instanceOfFavoriteTweet = void 0;
/**
 * Check if a given object implements the FavoriteTweet interface.
 */
function instanceOfFavoriteTweet(value) {
    var isInstance = true;
    isInstance = isInstance && "favoriteTweet" in value;
    return isInstance;
}
exports.instanceOfFavoriteTweet = instanceOfFavoriteTweet;
function FavoriteTweetFromJSON(json) {
    return FavoriteTweetFromJSONTyped(json, false);
}
exports.FavoriteTweetFromJSON = FavoriteTweetFromJSON;
function FavoriteTweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'favoriteTweet': json['favorite_tweet'],
    };
}
exports.FavoriteTweetFromJSONTyped = FavoriteTweetFromJSONTyped;
function FavoriteTweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'favorite_tweet': value.favoriteTweet,
    };
}
exports.FavoriteTweetToJSON = FavoriteTweetToJSON;
