// twitter-openapi-typescript-generated/dist/models/FavoriteTweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FavoriteTweetResponseDataToJSON = exports.FavoriteTweetResponseDataFromJSONTyped = exports.FavoriteTweetResponseDataFromJSON = exports.instanceOfFavoriteTweetResponseData = void 0;
var FavoriteTweet_1 = require("./FavoriteTweet");
/**
 * Check if a given object implements the FavoriteTweetResponseData interface.
 */
function instanceOfFavoriteTweetResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfFavoriteTweetResponseData = instanceOfFavoriteTweetResponseData;
function FavoriteTweetResponseDataFromJSON(json) {
    return FavoriteTweetResponseDataFromJSONTyped(json, false);
}
exports.FavoriteTweetResponseDataFromJSON = FavoriteTweetResponseDataFromJSON;
function FavoriteTweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, FavoriteTweet_1.FavoriteTweetFromJSON)(json['data']),
    };
}
exports.FavoriteTweetResponseDataFromJSONTyped = FavoriteTweetResponseDataFromJSONTyped;
function FavoriteTweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, FavoriteTweet_1.FavoriteTweetToJSON)(value.data),
    };
}
exports.FavoriteTweetResponseDataToJSON = FavoriteTweetResponseDataToJSON;
