// twitter-openapi-typescript-generated/dist/models/FollowTimeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowTimelineToJSON = exports.FollowTimelineFromJSONTyped = exports.FollowTimelineFromJSON = exports.instanceOfFollowTimeline = void 0;
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the FollowTimeline interface.
 */
function instanceOfFollowTimeline(value) {
    var isInstance = true;
    isInstance = isInstance && "timeline" in value;
    return isInstance;
}
exports.instanceOfFollowTimeline = instanceOfFollowTimeline;
function FollowTimelineFromJSON(json) {
    return FollowTimelineFromJSONTyped(json, false);
}
exports.FollowTimelineFromJSON = FollowTimelineFromJSON;
function FollowTimelineFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'timeline': (0, Timeline_1.TimelineFromJSON)(json['timeline']),
    };
}
exports.FollowTimelineFromJSONTyped = FollowTimelineFromJSONTyped;
function FollowTimelineToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'timeline': (0, Timeline_1.TimelineToJSON)(value.timeline),
    };
}
exports.FollowTimelineToJSON = FollowTimelineToJSON;
