// twitter-openapi-typescript-generated/dist/models/FollowResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowResponseToJSON = exports.FollowResponseFromJSONTyped = exports.FollowResponseFromJSON = exports.instanceOfFollowResponse = void 0;
var FollowResponseData_1 = require("./FollowResponseData");
/**
 * Check if a given object implements the FollowResponse interface.
 */
function instanceOfFollowResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfFollowResponse = instanceOfFollowResponse;
function FollowResponseFromJSON(json) {
    return FollowResponseFromJSONTyped(json, false);
}
exports.FollowResponseFromJSON = FollowResponseFromJSON;
function FollowResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, FollowResponseData_1.FollowResponseDataFromJSON)(json['data']),
    };
}
exports.FollowResponseFromJSONTyped = FollowResponseFromJSONTyped;
function FollowResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, FollowResponseData_1.FollowResponseDataToJSON)(value.data),
    };
}
exports.FollowResponseToJSON = FollowResponseToJSON;
