// twitter-openapi-typescript-generated/dist/models/FollowResponseUser.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowResponseUserToJSON = exports.FollowResponseUserFromJSONTyped = exports.FollowResponseUserFromJSON = exports.instanceOfFollowResponseUser = void 0;
var FollowResponseResult_1 = require("./FollowResponseResult");
/**
 * Check if a given object implements the FollowResponseUser interface.
 */
function instanceOfFollowResponseUser(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfFollowResponseUser = instanceOfFollowResponseUser;
function FollowResponseUserFromJSON(json) {
    return FollowResponseUserFromJSONTyped(json, false);
}
exports.FollowResponseUserFromJSON = FollowResponseUserFromJSON;
function FollowResponseUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, FollowResponseResult_1.FollowResponseResultFromJSON)(json['result']),
    };
}
exports.FollowResponseUserFromJSONTyped = FollowResponseUserFromJSONTyped;
function FollowResponseUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, FollowResponseResult_1.FollowResponseResultToJSON)(value.result),
    };
}
exports.FollowResponseUserToJSON = FollowResponseUserToJSON;
