// twitter-openapi-typescript-generated/dist/models/FollowResponseResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowResponseResultToJSON = exports.FollowResponseResultFromJSONTyped = exports.FollowResponseResultFromJSON = exports.instanceOfFollowResponseResult = void 0;
var FollowTimeline_1 = require("./FollowTimeline");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the FollowResponseResult interface.
 */
function instanceOfFollowResponseResult(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "timeline" in value;
    return isInstance;
}
exports.instanceOfFollowResponseResult = instanceOfFollowResponseResult;
function FollowResponseResultFromJSON(json) {
    return FollowResponseResultFromJSONTyped(json, false);
}
exports.FollowResponseResultFromJSON = FollowResponseResultFromJSON;
function FollowResponseResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'timeline': (0, FollowTimeline_1.FollowTimelineFromJSON)(json['timeline']),
    };
}
exports.FollowResponseResultFromJSONTyped = FollowResponseResultFromJSONTyped;
function FollowResponseResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'timeline': (0, FollowTimeline_1.FollowTimelineToJSON)(value.timeline),
    };
}
exports.FollowResponseResultToJSON = FollowResponseResultToJSON;
