// twitter-openapi-typescript-generated/dist/models/FollowResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowResponseDataToJSON = exports.FollowResponseDataFromJSONTyped = exports.FollowResponseDataFromJSON = exports.instanceOfFollowResponseData = void 0;
var FollowResponseUser_1 = require("./FollowResponseUser");
/**
 * Check if a given object implements the FollowResponseData interface.
 */
function instanceOfFollowResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfFollowResponseData = instanceOfFollowResponseData;
function FollowResponseDataFromJSON(json) {
    return FollowResponseDataFromJSONTyped(json, false);
}
exports.FollowResponseDataFromJSON = FollowResponseDataFromJSON;
function FollowResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, FollowResponseUser_1.FollowResponseUserFromJSON)(json['user']),
    };
}
exports.FollowResponseDataFromJSONTyped = FollowResponseDataFromJSONTyped;
function FollowResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, FollowResponseUser_1.FollowResponseUserToJSON)(value.user),
    };
}
exports.FollowResponseDataToJSON = FollowResponseDataToJSON;
