// twitter-openapi-typescript-generated/dist/models/HomeTimelineResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HomeTimelineResponseDataToJSON = exports.HomeTimelineResponseDataFromJSONTyped = exports.HomeTimelineResponseDataFromJSON = exports.instanceOfHomeTimelineResponseData = void 0;
var HomeTimelineHome_1 = require("./HomeTimelineHome");
/**
 * Check if a given object implements the HomeTimelineResponseData interface.
 */
function instanceOfHomeTimelineResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "home" in value;
    return isInstance;
}
exports.instanceOfHomeTimelineResponseData = instanceOfHomeTimelineResponseData;
function HomeTimelineResponseDataFromJSON(json) {
    return HomeTimelineResponseDataFromJSONTyped(json, false);
}
exports.HomeTimelineResponseDataFromJSON = HomeTimelineResponseDataFromJSON;
function HomeTimelineResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'home': (0, HomeTimelineHome_1.HomeTimelineHomeFromJSON)(json['home']),
    };
}
exports.HomeTimelineResponseDataFromJSONTyped = HomeTimelineResponseDataFromJSONTyped;
function HomeTimelineResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'home': (0, HomeTimelineHome_1.HomeTimelineHomeToJSON)(value.home),
    };
}
exports.HomeTimelineResponseDataToJSON = HomeTimelineResponseDataToJSON;
