// twitter-openapi-typescript-generated/dist/models/HomeTimelineHome.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HomeTimelineHomeToJSON = exports.HomeTimelineHomeFromJSONTyped = exports.HomeTimelineHomeFromJSON = exports.instanceOfHomeTimelineHome = void 0;
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the HomeTimelineHome interface.
 */
function instanceOfHomeTimelineHome(value) {
    var isInstance = true;
    isInstance = isInstance && "homeTimelineUrt" in value;
    return isInstance;
}
exports.instanceOfHomeTimelineHome = instanceOfHomeTimelineHome;
function HomeTimelineHomeFromJSON(json) {
    return HomeTimelineHomeFromJSONTyped(json, false);
}
exports.HomeTimelineHomeFromJSON = HomeTimelineHomeFromJSON;
function HomeTimelineHomeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'homeTimelineUrt': (0, Timeline_1.TimelineFromJSON)(json['home_timeline_urt']),
    };
}
exports.HomeTimelineHomeFromJSONTyped = HomeTimelineHomeFromJSONTyped;
function HomeTimelineHomeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'home_timeline_urt': (0, Timeline_1.TimelineToJSON)(value.homeTimelineUrt),
    };
}
exports.HomeTimelineHomeToJSON = HomeTimelineHomeToJSON;
