// twitter-openapi-typescript-generated/dist/models/InstructionType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstructionTypeToJSON = exports.InstructionTypeFromJSONTyped = exports.InstructionTypeFromJSON = exports.InstructionType = void 0;
/**
 *
 * @export
 */
exports.InstructionType = {
    TimelineAddEntries: 'TimelineAddEntries',
    TimelineAddToModule: 'TimelineAddToModule',
    TimelineClearCache: 'TimelineClearCache',
    TimelinePinEntry: 'TimelinePinEntry',
    TimelineReplaceEntry: 'TimelineReplaceEntry',
    TimelineShowAlert: 'TimelineShowAlert',
    TimelineTerminateTimeline: 'TimelineTerminateTimeline',
    TimelineShowCover: 'TimelineShowCover'
};
function InstructionTypeFromJSON(json) {
    return InstructionTypeFromJSONTyped(json, false);
}
exports.InstructionTypeFromJSON = InstructionTypeFromJSON;
function InstructionTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.InstructionTypeFromJSONTyped = InstructionTypeFromJSONTyped;
function InstructionTypeToJSON(value) {
    return value;
}
exports.InstructionTypeToJSON = InstructionTypeToJSON;
