// twitter-openapi-typescript-generated/dist/models/ItemResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemResultToJSON = exports.ItemResultFromJSONTyped = exports.ItemResultFromJSON = exports.instanceOfItemResult = void 0;
var runtime_1 = require("../runtime");
var TweetUnion_1 = require("./TweetUnion");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the ItemResult interface.
 */
function instanceOfItemResult(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfItemResult = instanceOfItemResult;
function ItemResultFromJSON(json) {
    return ItemResultFromJSONTyped(json, false);
}
exports.ItemResultFromJSON = ItemResultFromJSON;
function ItemResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': !(0, runtime_1.exists)(json, '__typename') ? undefined : (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'result': !(0, runtime_1.exists)(json, 'result') ? undefined : (0, TweetUnion_1.TweetUnionFromJSON)(json['result']),
    };
}
exports.ItemResultFromJSONTyped = ItemResultFromJSONTyped;
function ItemResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'result': (0, TweetUnion_1.TweetUnionToJSON)(value.result),
    };
}
exports.ItemResultToJSON = ItemResultToJSON;
