// twitter-openapi-typescript-generated/dist/models/ListLatestTweetsTimelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLatestTweetsTimelineResponseToJSON = exports.ListLatestTweetsTimelineResponseFromJSONTyped = exports.ListLatestTweetsTimelineResponseFromJSON = exports.instanceOfListLatestTweetsTimelineResponse = void 0;
var ListTweetsTimelineData_1 = require("./ListTweetsTimelineData");
/**
 * Check if a given object implements the ListLatestTweetsTimelineResponse interface.
 */
function instanceOfListLatestTweetsTimelineResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfListLatestTweetsTimelineResponse = instanceOfListLatestTweetsTimelineResponse;
function ListLatestTweetsTimelineResponseFromJSON(json) {
    return ListLatestTweetsTimelineResponseFromJSONTyped(json, false);
}
exports.ListLatestTweetsTimelineResponseFromJSON = ListLatestTweetsTimelineResponseFromJSON;
function ListLatestTweetsTimelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, ListTweetsTimelineData_1.ListTweetsTimelineDataFromJSON)(json['data']),
    };
}
exports.ListLatestTweetsTimelineResponseFromJSONTyped = ListLatestTweetsTimelineResponseFromJSONTyped;
function ListLatestTweetsTimelineResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, ListTweetsTimelineData_1.ListTweetsTimelineDataToJSON)(value.data),
    };
}
exports.ListLatestTweetsTimelineResponseToJSON = ListLatestTweetsTimelineResponseToJSON;
