// twitter-openapi-typescript-generated/dist/models/ListTweetsTimelineList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTweetsTimelineListToJSON = exports.ListTweetsTimelineListFromJSONTyped = exports.ListTweetsTimelineListFromJSON = exports.instanceOfListTweetsTimelineList = void 0;
var ListTweetsTimeline_1 = require("./ListTweetsTimeline");
/**
 * Check if a given object implements the ListTweetsTimelineList interface.
 */
function instanceOfListTweetsTimelineList(value) {
    var isInstance = true;
    isInstance = isInstance && "tweetsTimeline" in value;
    return isInstance;
}
exports.instanceOfListTweetsTimelineList = instanceOfListTweetsTimelineList;
function ListTweetsTimelineListFromJSON(json) {
    return ListTweetsTimelineListFromJSONTyped(json, false);
}
exports.ListTweetsTimelineListFromJSON = ListTweetsTimelineListFromJSON;
function ListTweetsTimelineListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tweetsTimeline': (0, ListTweetsTimeline_1.ListTweetsTimelineFromJSON)(json['tweets_timeline']),
    };
}
exports.ListTweetsTimelineListFromJSONTyped = ListTweetsTimelineListFromJSONTyped;
function ListTweetsTimelineListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'tweets_timeline': (0, ListTweetsTimeline_1.ListTweetsTimelineToJSON)(value.tweetsTimeline),
    };
}
exports.ListTweetsTimelineListToJSON = ListTweetsTimelineListToJSON;
