// twitter-openapi-typescript-generated/dist/models/ListTweetsTimeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTweetsTimelineToJSON = exports.ListTweetsTimelineFromJSONTyped = exports.ListTweetsTimelineFromJSON = exports.instanceOfListTweetsTimeline = void 0;
var runtime_1 = require("../runtime");
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the ListTweetsTimeline interface.
 */
function instanceOfListTweetsTimeline(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfListTweetsTimeline = instanceOfListTweetsTimeline;
function ListTweetsTimelineFromJSON(json) {
    return ListTweetsTimelineFromJSONTyped(json, false);
}
exports.ListTweetsTimelineFromJSON = ListTweetsTimelineFromJSON;
function ListTweetsTimelineFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'timeline': !(0, runtime_1.exists)(json, 'timeline') ? undefined : (0, Timeline_1.TimelineFromJSON)(json['timeline']),
    };
}
exports.ListTweetsTimelineFromJSONTyped = ListTweetsTimelineFromJSONTyped;
function ListTweetsTimelineToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'timeline': (0, Timeline_1.TimelineToJSON)(value.timeline),
    };
}
exports.ListTweetsTimelineToJSON = ListTweetsTimelineToJSON;
