// twitter-openapi-typescript-generated/dist/models/ListTweetsTimelineData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTweetsTimelineDataToJSON = exports.ListTweetsTimelineDataFromJSONTyped = exports.ListTweetsTimelineDataFromJSON = exports.instanceOfListTweetsTimelineData = void 0;
var ListTweetsTimelineList_1 = require("./ListTweetsTimelineList");
/**
 * Check if a given object implements the ListTweetsTimelineData interface.
 */
function instanceOfListTweetsTimelineData(value) {
    var isInstance = true;
    isInstance = isInstance && "list" in value;
    return isInstance;
}
exports.instanceOfListTweetsTimelineData = instanceOfListTweetsTimelineData;
function ListTweetsTimelineDataFromJSON(json) {
    return ListTweetsTimelineDataFromJSONTyped(json, false);
}
exports.ListTweetsTimelineDataFromJSON = ListTweetsTimelineDataFromJSON;
function ListTweetsTimelineDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'list': (0, ListTweetsTimelineList_1.ListTweetsTimelineListFromJSON)(json['list']),
    };
}
exports.ListTweetsTimelineDataFromJSONTyped = ListTweetsTimelineDataFromJSONTyped;
function ListTweetsTimelineDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'list': (0, ListTweetsTimelineList_1.ListTweetsTimelineListToJSON)(value.list),
    };
}
exports.ListTweetsTimelineDataToJSON = ListTweetsTimelineDataToJSON;
