// twitter-openapi-typescript-generated/dist/models/Location.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationToJSON = exports.LocationFromJSONTyped = exports.LocationFromJSON = exports.instanceOfLocation = void 0;
/**
 * Check if a given object implements the Location interface.
 */
function instanceOfLocation(value) {
    var isInstance = true;
    isInstance = isInstance && "column" in value;
    isInstance = isInstance && "line" in value;
    return isInstance;
}
exports.instanceOfLocation = instanceOfLocation;
function LocationFromJSON(json) {
    return LocationFromJSONTyped(json, false);
}
exports.LocationFromJSON = LocationFromJSON;
function LocationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'column': json['column'],
        'line': json['line'],
    };
}
exports.LocationFromJSONTyped = LocationFromJSONTyped;
function LocationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'column': value.column,
        'line': value.line,
    };
}
exports.LocationToJSON = LocationToJSON;
