// twitter-openapi-typescript-generated/dist/models/MediaSize.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaSizeToJSON = exports.MediaSizeFromJSONTyped = exports.MediaSizeFromJSON = exports.instanceOfMediaSize = exports.MediaSizeResizeEnum = void 0;
/**
 * @export
 */
exports.MediaSizeResizeEnum = {
    Crop: 'crop',
    Fit: 'fit'
};
/**
 * Check if a given object implements the MediaSize interface.
 */
function instanceOfMediaSize(value) {
    var isInstance = true;
    isInstance = isInstance && "h" in value;
    isInstance = isInstance && "resize" in value;
    isInstance = isInstance && "w" in value;
    return isInstance;
}
exports.instanceOfMediaSize = instanceOfMediaSize;
function MediaSizeFromJSON(json) {
    return MediaSizeFromJSONTyped(json, false);
}
exports.MediaSizeFromJSON = MediaSizeFromJSON;
function MediaSizeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'h': json['h'],
        'resize': json['resize'],
        'w': json['w'],
    };
}
exports.MediaSizeFromJSONTyped = MediaSizeFromJSONTyped;
function MediaSizeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'h': value.h,
        'resize': value.resize,
        'w': value.w,
    };
}
exports.MediaSizeToJSON = MediaSizeToJSON;
