// twitter-openapi-typescript-generated/dist/models/MediaSizes.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaSizesToJSON = exports.MediaSizesFromJSONTyped = exports.MediaSizesFromJSON = exports.instanceOfMediaSizes = void 0;
var MediaSize_1 = require("./MediaSize");
/**
 * Check if a given object implements the MediaSizes interface.
 */
function instanceOfMediaSizes(value) {
    var isInstance = true;
    isInstance = isInstance && "large" in value;
    isInstance = isInstance && "medium" in value;
    isInstance = isInstance && "small" in value;
    isInstance = isInstance && "thumb" in value;
    return isInstance;
}
exports.instanceOfMediaSizes = instanceOfMediaSizes;
function MediaSizesFromJSON(json) {
    return MediaSizesFromJSONTyped(json, false);
}
exports.MediaSizesFromJSON = MediaSizesFromJSON;
function MediaSizesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'large': (0, MediaSize_1.MediaSizeFromJSON)(json['large']),
        'medium': (0, MediaSize_1.MediaSizeFromJSON)(json['medium']),
        'small': (0, MediaSize_1.MediaSizeFromJSON)(json['small']),
        'thumb': (0, MediaSize_1.MediaSizeFromJSON)(json['thumb']),
    };
}
exports.MediaSizesFromJSONTyped = MediaSizesFromJSONTyped;
function MediaSizesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'large': (0, MediaSize_1.MediaSizeToJSON)(value.large),
        'medium': (0, MediaSize_1.MediaSizeToJSON)(value.medium),
        'small': (0, MediaSize_1.MediaSizeToJSON)(value.small),
        'thumb': (0, MediaSize_1.MediaSizeToJSON)(value.thumb),
    };
}
exports.MediaSizesToJSON = MediaSizesToJSON;
