// twitter-openapi-typescript-generated/dist/models/MediaOriginalInfoFocusRect.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaOriginalInfoFocusRectToJSON = exports.MediaOriginalInfoFocusRectFromJSONTyped = exports.MediaOriginalInfoFocusRectFromJSON = exports.instanceOfMediaOriginalInfoFocusRect = void 0;
/**
 * Check if a given object implements the MediaOriginalInfoFocusRect interface.
 */
function instanceOfMediaOriginalInfoFocusRect(value) {
    var isInstance = true;
    isInstance = isInstance && "h" in value;
    isInstance = isInstance && "w" in value;
    isInstance = isInstance && "x" in value;
    isInstance = isInstance && "y" in value;
    return isInstance;
}
exports.instanceOfMediaOriginalInfoFocusRect = instanceOfMediaOriginalInfoFocusRect;
function MediaOriginalInfoFocusRectFromJSON(json) {
    return MediaOriginalInfoFocusRectFromJSONTyped(json, false);
}
exports.MediaOriginalInfoFocusRectFromJSON = MediaOriginalInfoFocusRectFromJSON;
function MediaOriginalInfoFocusRectFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'h': json['h'],
        'w': json['w'],
        'x': json['x'],
        'y': json['y'],
    };
}
exports.MediaOriginalInfoFocusRectFromJSONTyped = MediaOriginalInfoFocusRectFromJSONTyped;
function MediaOriginalInfoFocusRectToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'h': value.h,
        'w': value.w,
        'x': value.x,
        'y': value.y,
    };
}
exports.MediaOriginalInfoFocusRectToJSON = MediaOriginalInfoFocusRectToJSON;
