// twitter-openapi-typescript-generated/dist/models/Media.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaToJSON = exports.MediaFromJSONTyped = exports.MediaFromJSON = exports.instanceOfMedia = exports.MediaTypeEnum = void 0;
var runtime_1 = require("../runtime");
var MediaOriginalInfo_1 = require("./MediaOriginalInfo");
var MediaSizes_1 = require("./MediaSizes");
/**
 * @export
 */
exports.MediaTypeEnum = {
    Photo: 'photo',
    Video: 'video',
    AnimatedGif: 'animated_gif'
};
/**
 * Check if a given object implements the Media interface.
 */
function instanceOfMedia(value) {
    var isInstance = true;
    isInstance = isInstance && "displayUrl" in value;
    isInstance = isInstance && "expandedUrl" in value;
    isInstance = isInstance && "idStr" in value;
    isInstance = isInstance && "indices" in value;
    isInstance = isInstance && "mediaUrlHttps" in value;
    isInstance = isInstance && "originalInfo" in value;
    isInstance = isInstance && "sizes" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfMedia = instanceOfMedia;
function MediaFromJSON(json) {
    return MediaFromJSONTyped(json, false);
}
exports.MediaFromJSON = MediaFromJSON;
function MediaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'displayUrl': json['display_url'],
        'expandedUrl': json['expanded_url'],
        'features': !(0, runtime_1.exists)(json, 'features') ? undefined : json['features'],
        'idStr': json['id_str'],
        'indices': json['indices'],
        'mediaUrlHttps': json['media_url_https'],
        'originalInfo': (0, MediaOriginalInfo_1.MediaOriginalInfoFromJSON)(json['original_info']),
        'sizes': (0, MediaSizes_1.MediaSizesFromJSON)(json['sizes']),
        'type': json['type'],
        'url': json['url'],
    };
}
exports.MediaFromJSONTyped = MediaFromJSONTyped;
function MediaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'display_url': value.displayUrl,
        'expanded_url': value.expandedUrl,
        'features': value.features,
        'id_str': value.idStr,
        'indices': value.indices,
        'media_url_https': value.mediaUrlHttps,
        'original_info': (0, MediaOriginalInfo_1.MediaOriginalInfoToJSON)(value.originalInfo),
        'sizes': (0, MediaSizes_1.MediaSizesToJSON)(value.sizes),
        'type': value.type,
        'url': value.url,
    };
}
exports.MediaToJSON = MediaToJSON;
