// twitter-openapi-typescript-generated/dist/models/MediaOriginalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaOriginalInfoToJSON = exports.MediaOriginalInfoFromJSONTyped = exports.MediaOriginalInfoFromJSON = exports.instanceOfMediaOriginalInfo = void 0;
var runtime_1 = require("../runtime");
var MediaOriginalInfoFocusRect_1 = require("./MediaOriginalInfoFocusRect");
/**
 * Check if a given object implements the MediaOriginalInfo interface.
 */
function instanceOfMediaOriginalInfo(value) {
    var isInstance = true;
    isInstance = isInstance && "height" in value;
    isInstance = isInstance && "width" in value;
    return isInstance;
}
exports.instanceOfMediaOriginalInfo = instanceOfMediaOriginalInfo;
function MediaOriginalInfoFromJSON(json) {
    return MediaOriginalInfoFromJSONTyped(json, false);
}
exports.MediaOriginalInfoFromJSON = MediaOriginalInfoFromJSON;
function MediaOriginalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'focusRects': !(0, runtime_1.exists)(json, 'focus_rects') ? undefined : (json['focus_rects'].map(MediaOriginalInfoFocusRect_1.MediaOriginalInfoFocusRectFromJSON)),
        'height': json['height'],
        'width': json['width'],
    };
}
exports.MediaOriginalInfoFromJSONTyped = MediaOriginalInfoFromJSONTyped;
function MediaOriginalInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'focus_rects': value.focusRects === undefined ? undefined : (value.focusRects.map(MediaOriginalInfoFocusRect_1.MediaOriginalInfoFocusRectToJSON)),
        'height': value.height,
        'width': value.width,
    };
}
exports.MediaOriginalInfoToJSON = MediaOriginalInfoToJSON;
