// twitter-openapi-typescript-generated/dist/models/MediaStats.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaStatsToJSON = exports.MediaStatsFromJSONTyped = exports.MediaStatsFromJSON = exports.instanceOfMediaStats = void 0;
/**
 * Check if a given object implements the MediaStats interface.
 */
function instanceOfMediaStats(value) {
    var isInstance = true;
    isInstance = isInstance && "viewCount" in value;
    return isInstance;
}
exports.instanceOfMediaStats = instanceOfMediaStats;
function MediaStatsFromJSON(json) {
    return MediaStatsFromJSONTyped(json, false);
}
exports.MediaStatsFromJSON = MediaStatsFromJSON;
function MediaStatsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'viewCount': json['viewCount'],
    };
}
exports.MediaStatsFromJSONTyped = MediaStatsFromJSONTyped;
function MediaStatsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'viewCount': value.viewCount,
    };
}
exports.MediaStatsToJSON = MediaStatsToJSON;
