// twitter-openapi-typescript-generated/dist/models/ModuleItem.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModuleItemToJSON = exports.ModuleItemFromJSONTyped = exports.ModuleItemFromJSON = exports.instanceOfModuleItem = void 0;
var ModuleEntry_1 = require("./ModuleEntry");
/**
 * Check if a given object implements the ModuleItem interface.
 */
function instanceOfModuleItem(value) {
    var isInstance = true;
    isInstance = isInstance && "entryId" in value;
    isInstance = isInstance && "item" in value;
    return isInstance;
}
exports.instanceOfModuleItem = instanceOfModuleItem;
function ModuleItemFromJSON(json) {
    return ModuleItemFromJSONTyped(json, false);
}
exports.ModuleItemFromJSON = ModuleItemFromJSON;
function ModuleItemFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entryId': json['entryId'],
        'item': (0, ModuleEntry_1.ModuleEntryFromJSON)(json['item']),
    };
}
exports.ModuleItemFromJSONTyped = ModuleItemFromJSONTyped;
function ModuleItemToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entryId': value.entryId,
        'item': (0, ModuleEntry_1.ModuleEntryToJSON)(value.item),
    };
}
exports.ModuleItemToJSON = ModuleItemToJSON;
