// twitter-openapi-typescript-generated/dist/models/ModuleEntry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModuleEntryToJSON = exports.ModuleEntryFromJSONTyped = exports.ModuleEntryFromJSON = exports.instanceOfModuleEntry = void 0;
var ClientEventInfo_1 = require("./ClientEventInfo");
var ItemContentUnion_1 = require("./ItemContentUnion");
/**
 * Check if a given object implements the ModuleEntry interface.
 */
function instanceOfModuleEntry(value) {
    var isInstance = true;
    isInstance = isInstance && "clientEventInfo" in value;
    isInstance = isInstance && "itemContent" in value;
    return isInstance;
}
exports.instanceOfModuleEntry = instanceOfModuleEntry;
function ModuleEntryFromJSON(json) {
    return ModuleEntryFromJSONTyped(json, false);
}
exports.ModuleEntryFromJSON = ModuleEntryFromJSON;
function ModuleEntryFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientEventInfo': (0, ClientEventInfo_1.ClientEventInfoFromJSON)(json['clientEventInfo']),
        'itemContent': (0, ItemContentUnion_1.ItemContentUnionFromJSON)(json['itemContent']),
    };
}
exports.ModuleEntryFromJSONTyped = ModuleEntryFromJSONTyped;
function ModuleEntryToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientEventInfo': (0, ClientEventInfo_1.ClientEventInfoToJSON)(value.clientEventInfo),
        'itemContent': (0, ItemContentUnion_1.ItemContentUnionToJSON)(value.itemContent),
    };
}
exports.ModuleEntryToJSON = ModuleEntryToJSON;
