// twitter-openapi-typescript-generated/dist/models/NoteTweetResultRichText.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetResultRichTextToJSON = exports.NoteTweetResultRichTextFromJSONTyped = exports.NoteTweetResultRichTextFromJSON = exports.instanceOfNoteTweetResultRichText = void 0;
var NoteTweetResultRichTextTag_1 = require("./NoteTweetResultRichTextTag");
/**
 * Check if a given object implements the NoteTweetResultRichText interface.
 */
function instanceOfNoteTweetResultRichText(value) {
    var isInstance = true;
    isInstance = isInstance && "richtextTags" in value;
    return isInstance;
}
exports.instanceOfNoteTweetResultRichText = instanceOfNoteTweetResultRichText;
function NoteTweetResultRichTextFromJSON(json) {
    return NoteTweetResultRichTextFromJSONTyped(json, false);
}
exports.NoteTweetResultRichTextFromJSON = NoteTweetResultRichTextFromJSON;
function NoteTweetResultRichTextFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'richtextTags': (json['richtext_tags'].map(NoteTweetResultRichTextTag_1.NoteTweetResultRichTextTagFromJSON)),
    };
}
exports.NoteTweetResultRichTextFromJSONTyped = NoteTweetResultRichTextFromJSONTyped;
function NoteTweetResultRichTextToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'richtext_tags': (value.richtextTags.map(NoteTweetResultRichTextTag_1.NoteTweetResultRichTextTagToJSON)),
    };
}
exports.NoteTweetResultRichTextToJSON = NoteTweetResultRichTextToJSON;
