// twitter-openapi-typescript-generated/dist/models/NoteTweetResultMedia.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetResultMediaToJSON = exports.NoteTweetResultMediaFromJSONTyped = exports.NoteTweetResultMediaFromJSON = exports.instanceOfNoteTweetResultMedia = void 0;
var NoteTweetResultMediaInlineMedia_1 = require("./NoteTweetResultMediaInlineMedia");
/**
 * Check if a given object implements the NoteTweetResultMedia interface.
 */
function instanceOfNoteTweetResultMedia(value) {
    var isInstance = true;
    isInstance = isInstance && "inlineMedia" in value;
    return isInstance;
}
exports.instanceOfNoteTweetResultMedia = instanceOfNoteTweetResultMedia;
function NoteTweetResultMediaFromJSON(json) {
    return NoteTweetResultMediaFromJSONTyped(json, false);
}
exports.NoteTweetResultMediaFromJSON = NoteTweetResultMediaFromJSON;
function NoteTweetResultMediaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'inlineMedia': (json['inline_media'].map(NoteTweetResultMediaInlineMedia_1.NoteTweetResultMediaInlineMediaFromJSON)),
    };
}
exports.NoteTweetResultMediaFromJSONTyped = NoteTweetResultMediaFromJSONTyped;
function NoteTweetResultMediaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'inline_media': (value.inlineMedia.map(NoteTweetResultMediaInlineMedia_1.NoteTweetResultMediaInlineMediaToJSON)),
    };
}
exports.NoteTweetResultMediaToJSON = NoteTweetResultMediaToJSON;
