// twitter-openapi-typescript-generated/dist/models/NoteTweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetToJSON = exports.NoteTweetFromJSONTyped = exports.NoteTweetFromJSON = exports.instanceOfNoteTweet = void 0;
var NoteTweetResult_1 = require("./NoteTweetResult");
/**
 * Check if a given object implements the NoteTweet interface.
 */
function instanceOfNoteTweet(value) {
    var isInstance = true;
    isInstance = isInstance && "isExpandable" in value;
    isInstance = isInstance && "noteTweetResults" in value;
    return isInstance;
}
exports.instanceOfNoteTweet = instanceOfNoteTweet;
function NoteTweetFromJSON(json) {
    return NoteTweetFromJSONTyped(json, false);
}
exports.NoteTweetFromJSON = NoteTweetFromJSON;
function NoteTweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'isExpandable': json['is_expandable'],
        'noteTweetResults': (0, NoteTweetResult_1.NoteTweetResultFromJSON)(json['note_tweet_results']),
    };
}
exports.NoteTweetFromJSONTyped = NoteTweetFromJSONTyped;
function NoteTweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'is_expandable': value.isExpandable,
        'note_tweet_results': (0, NoteTweetResult_1.NoteTweetResultToJSON)(value.noteTweetResults),
    };
}
exports.NoteTweetToJSON = NoteTweetToJSON;
