// twitter-openapi-typescript-generated/dist/models/NoteTweetResultRichTextTag.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetResultRichTextTagToJSON = exports.NoteTweetResultRichTextTagFromJSONTyped = exports.NoteTweetResultRichTextTagFromJSON = exports.instanceOfNoteTweetResultRichTextTag = exports.NoteTweetResultRichTextTagRichtextTypesEnum = void 0;
/**
 * @export
 */
exports.NoteTweetResultRichTextTagRichtextTypesEnum = {
    Bold: 'Bold',
    Italic: 'Italic'
};
/**
 * Check if a given object implements the NoteTweetResultRichTextTag interface.
 */
function instanceOfNoteTweetResultRichTextTag(value) {
    var isInstance = true;
    isInstance = isInstance && "fromIndex" in value;
    isInstance = isInstance && "richtextTypes" in value;
    isInstance = isInstance && "toIndex" in value;
    return isInstance;
}
exports.instanceOfNoteTweetResultRichTextTag = instanceOfNoteTweetResultRichTextTag;
function NoteTweetResultRichTextTagFromJSON(json) {
    return NoteTweetResultRichTextTagFromJSONTyped(json, false);
}
exports.NoteTweetResultRichTextTagFromJSON = NoteTweetResultRichTextTagFromJSON;
function NoteTweetResultRichTextTagFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fromIndex': json['from_index'],
        'richtextTypes': json['richtext_types'],
        'toIndex': json['to_index'],
    };
}
exports.NoteTweetResultRichTextTagFromJSONTyped = NoteTweetResultRichTextTagFromJSONTyped;
function NoteTweetResultRichTextTagToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'from_index': value.fromIndex,
        'richtext_types': value.richtextTypes,
        'to_index': value.toIndex,
    };
}
exports.NoteTweetResultRichTextTagToJSON = NoteTweetResultRichTextTagToJSON;
