// twitter-openapi-typescript-generated/dist/models/NoteTweetResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetResultToJSON = exports.NoteTweetResultFromJSONTyped = exports.NoteTweetResultFromJSON = exports.instanceOfNoteTweetResult = void 0;
var NoteTweetResultData_1 = require("./NoteTweetResultData");
/**
 * Check if a given object implements the NoteTweetResult interface.
 */
function instanceOfNoteTweetResult(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfNoteTweetResult = instanceOfNoteTweetResult;
function NoteTweetResultFromJSON(json) {
    return NoteTweetResultFromJSONTyped(json, false);
}
exports.NoteTweetResultFromJSON = NoteTweetResultFromJSON;
function NoteTweetResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, NoteTweetResultData_1.NoteTweetResultDataFromJSON)(json['result']),
    };
}
exports.NoteTweetResultFromJSONTyped = NoteTweetResultFromJSONTyped;
function NoteTweetResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, NoteTweetResultData_1.NoteTweetResultDataToJSON)(value.result),
    };
}
exports.NoteTweetResultToJSON = NoteTweetResultToJSON;
