// twitter-openapi-typescript-generated/dist/models/NoteTweetResultMediaInlineMedia.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NoteTweetResultMediaInlineMediaToJSON = exports.NoteTweetResultMediaInlineMediaFromJSONTyped = exports.NoteTweetResultMediaInlineMediaFromJSON = exports.instanceOfNoteTweetResultMediaInlineMedia = void 0;
/**
 * Check if a given object implements the NoteTweetResultMediaInlineMedia interface.
 */
function instanceOfNoteTweetResultMediaInlineMedia(value) {
    var isInstance = true;
    isInstance = isInstance && "index" in value;
    isInstance = isInstance && "mediaId" in value;
    return isInstance;
}
exports.instanceOfNoteTweetResultMediaInlineMedia = instanceOfNoteTweetResultMediaInlineMedia;
function NoteTweetResultMediaInlineMediaFromJSON(json) {
    return NoteTweetResultMediaInlineMediaFromJSONTyped(json, false);
}
exports.NoteTweetResultMediaInlineMediaFromJSON = NoteTweetResultMediaInlineMediaFromJSON;
function NoteTweetResultMediaInlineMediaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'index': json['index'],
        'mediaId': json['media_id'],
    };
}
exports.NoteTweetResultMediaInlineMediaFromJSONTyped = NoteTweetResultMediaInlineMediaFromJSONTyped;
function NoteTweetResultMediaInlineMediaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'index': value.index,
        'media_id': value.mediaId,
    };
}
exports.NoteTweetResultMediaInlineMediaToJSON = NoteTweetResultMediaInlineMediaToJSON;
