// twitter-openapi-typescript-generated/dist/models/OneFactorLoginEligibility.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OneFactorLoginEligibilityToJSON = exports.OneFactorLoginEligibilityFromJSONTyped = exports.OneFactorLoginEligibilityFromJSON = exports.instanceOfOneFactorLoginEligibility = void 0;
/**
 * Check if a given object implements the OneFactorLoginEligibility interface.
 */
function instanceOfOneFactorLoginEligibility(value) {
    var isInstance = true;
    isInstance = isInstance && "fetchStatus" in value;
    return isInstance;
}
exports.instanceOfOneFactorLoginEligibility = instanceOfOneFactorLoginEligibility;
function OneFactorLoginEligibilityFromJSON(json) {
    return OneFactorLoginEligibilityFromJSONTyped(json, false);
}
exports.OneFactorLoginEligibilityFromJSON = OneFactorLoginEligibilityFromJSON;
function OneFactorLoginEligibilityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fetchStatus': json['fetchStatus'],
    };
}
exports.OneFactorLoginEligibilityFromJSONTyped = OneFactorLoginEligibilityFromJSONTyped;
function OneFactorLoginEligibilityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fetchStatus': value.fetchStatus,
    };
}
exports.OneFactorLoginEligibilityToJSON = OneFactorLoginEligibilityToJSON;
