// twitter-openapi-typescript-generated/dist/models/OtherResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtherResponseToJSON = exports.OtherResponseFromJSONTyped = exports.OtherResponseFromJSON = exports.instanceOfOtherResponse = void 0;
var runtime_1 = require("../runtime");
var Session_1 = require("./Session");
/**
 * Check if a given object implements the OtherResponse interface.
 */
function instanceOfOtherResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOtherResponse = instanceOfOtherResponse;
function OtherResponseFromJSON(json) {
    return OtherResponseFromJSONTyped(json, false);
}
exports.OtherResponseFromJSON = OtherResponseFromJSON;
function OtherResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'session': !(0, runtime_1.exists)(json, 'Session') ? undefined : (0, Session_1.SessionFromJSON)(json['Session']),
    };
}
exports.OtherResponseFromJSONTyped = OtherResponseFromJSONTyped;
function OtherResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'Session': (0, Session_1.SessionToJSON)(value.session),
    };
}
exports.OtherResponseToJSON = OtherResponseToJSON;
