// twitter-openapi-typescript-generated/dist/models/PostCreateRetweetRequestVariables.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateRetweetRequestVariablesToJSON = exports.PostCreateRetweetRequestVariablesFromJSONTyped = exports.PostCreateRetweetRequestVariablesFromJSON = exports.instanceOfPostCreateRetweetRequestVariables = void 0;
/**
 * Check if a given object implements the PostCreateRetweetRequestVariables interface.
 */
function instanceOfPostCreateRetweetRequestVariables(value) {
    var isInstance = true;
    isInstance = isInstance && "darkRequest" in value;
    isInstance = isInstance && "tweetId" in value;
    return isInstance;
}
exports.instanceOfPostCreateRetweetRequestVariables = instanceOfPostCreateRetweetRequestVariables;
function PostCreateRetweetRequestVariablesFromJSON(json) {
    return PostCreateRetweetRequestVariablesFromJSONTyped(json, false);
}
exports.PostCreateRetweetRequestVariablesFromJSON = PostCreateRetweetRequestVariablesFromJSON;
function PostCreateRetweetRequestVariablesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'darkRequest': json['dark_request'],
        'tweetId': json['tweet_id'],
    };
}
exports.PostCreateRetweetRequestVariablesFromJSONTyped = PostCreateRetweetRequestVariablesFromJSONTyped;
function PostCreateRetweetRequestVariablesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'dark_request': value.darkRequest,
        'tweet_id': value.tweetId,
    };
}
exports.PostCreateRetweetRequestVariablesToJSON = PostCreateRetweetRequestVariablesToJSON;
