// twitter-openapi-typescript-generated/dist/models/PostDeleteRetweetRequestVariables.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostDeleteRetweetRequestVariablesToJSON = exports.PostDeleteRetweetRequestVariablesFromJSONTyped = exports.PostDeleteRetweetRequestVariablesFromJSON = exports.instanceOfPostDeleteRetweetRequestVariables = void 0;
/**
 * Check if a given object implements the PostDeleteRetweetRequestVariables interface.
 */
function instanceOfPostDeleteRetweetRequestVariables(value) {
    var isInstance = true;
    isInstance = isInstance && "darkRequest" in value;
    isInstance = isInstance && "sourceTweetId" in value;
    return isInstance;
}
exports.instanceOfPostDeleteRetweetRequestVariables = instanceOfPostDeleteRetweetRequestVariables;
function PostDeleteRetweetRequestVariablesFromJSON(json) {
    return PostDeleteRetweetRequestVariablesFromJSONTyped(json, false);
}
exports.PostDeleteRetweetRequestVariablesFromJSON = PostDeleteRetweetRequestVariablesFromJSON;
function PostDeleteRetweetRequestVariablesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'darkRequest': json['dark_request'],
        'sourceTweetId': json['source_tweet_id'],
    };
}
exports.PostDeleteRetweetRequestVariablesFromJSONTyped = PostDeleteRetweetRequestVariablesFromJSONTyped;
function PostDeleteRetweetRequestVariablesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'dark_request': value.darkRequest,
        'source_tweet_id': value.sourceTweetId,
    };
}
exports.PostDeleteRetweetRequestVariablesToJSON = PostDeleteRetweetRequestVariablesToJSON;
