// twitter-openapi-typescript-generated/dist/models/PostCreateRetweetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateRetweetRequestToJSON = exports.PostCreateRetweetRequestFromJSONTyped = exports.PostCreateRetweetRequestFromJSON = exports.instanceOfPostCreateRetweetRequest = void 0;
var PostCreateRetweetRequestVariables_1 = require("./PostCreateRetweetRequestVariables");
/**
 * Check if a given object implements the PostCreateRetweetRequest interface.
 */
function instanceOfPostCreateRetweetRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "queryId" in value;
    isInstance = isInstance && "variables" in value;
    return isInstance;
}
exports.instanceOfPostCreateRetweetRequest = instanceOfPostCreateRetweetRequest;
function PostCreateRetweetRequestFromJSON(json) {
    return PostCreateRetweetRequestFromJSONTyped(json, false);
}
exports.PostCreateRetweetRequestFromJSON = PostCreateRetweetRequestFromJSON;
function PostCreateRetweetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'queryId': json['queryId'],
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesFromJSON)(json['variables']),
    };
}
exports.PostCreateRetweetRequestFromJSONTyped = PostCreateRetweetRequestFromJSONTyped;
function PostCreateRetweetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'queryId': value.queryId,
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesToJSON)(value.variables),
    };
}
exports.PostCreateRetweetRequestToJSON = PostCreateRetweetRequestToJSON;
