// twitter-openapi-typescript-generated/dist/models/PostUnfavoriteTweetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostUnfavoriteTweetRequestToJSON = exports.PostUnfavoriteTweetRequestFromJSONTyped = exports.PostUnfavoriteTweetRequestFromJSON = exports.instanceOfPostUnfavoriteTweetRequest = void 0;
var PostCreateRetweetRequestVariables_1 = require("./PostCreateRetweetRequestVariables");
/**
 * Check if a given object implements the PostUnfavoriteTweetRequest interface.
 */
function instanceOfPostUnfavoriteTweetRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "queryId" in value;
    isInstance = isInstance && "variables" in value;
    return isInstance;
}
exports.instanceOfPostUnfavoriteTweetRequest = instanceOfPostUnfavoriteTweetRequest;
function PostUnfavoriteTweetRequestFromJSON(json) {
    return PostUnfavoriteTweetRequestFromJSONTyped(json, false);
}
exports.PostUnfavoriteTweetRequestFromJSON = PostUnfavoriteTweetRequestFromJSON;
function PostUnfavoriteTweetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'queryId': json['queryId'],
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesFromJSON)(json['variables']),
    };
}
exports.PostUnfavoriteTweetRequestFromJSONTyped = PostUnfavoriteTweetRequestFromJSONTyped;
function PostUnfavoriteTweetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'queryId': value.queryId,
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesToJSON)(value.variables),
    };
}
exports.PostUnfavoriteTweetRequestToJSON = PostUnfavoriteTweetRequestToJSON;
