// twitter-openapi-typescript-generated/dist/models/PostCreateTweetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateTweetRequestToJSON = exports.PostCreateTweetRequestFromJSONTyped = exports.PostCreateTweetRequestFromJSON = exports.instanceOfPostCreateTweetRequest = void 0;
var PostCreateTweetRequestFeatures_1 = require("./PostCreateTweetRequestFeatures");
var PostCreateTweetRequestVariables_1 = require("./PostCreateTweetRequestVariables");
/**
 * Check if a given object implements the PostCreateTweetRequest interface.
 */
function instanceOfPostCreateTweetRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "features" in value;
    isInstance = isInstance && "queryId" in value;
    isInstance = isInstance && "variables" in value;
    return isInstance;
}
exports.instanceOfPostCreateTweetRequest = instanceOfPostCreateTweetRequest;
function PostCreateTweetRequestFromJSON(json) {
    return PostCreateTweetRequestFromJSONTyped(json, false);
}
exports.PostCreateTweetRequestFromJSON = PostCreateTweetRequestFromJSON;
function PostCreateTweetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'features': (0, PostCreateTweetRequestFeatures_1.PostCreateTweetRequestFeaturesFromJSON)(json['features']),
        'queryId': json['queryId'],
        'variables': (0, PostCreateTweetRequestVariables_1.PostCreateTweetRequestVariablesFromJSON)(json['variables']),
    };
}
exports.PostCreateTweetRequestFromJSONTyped = PostCreateTweetRequestFromJSONTyped;
function PostCreateTweetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'features': (0, PostCreateTweetRequestFeatures_1.PostCreateTweetRequestFeaturesToJSON)(value.features),
        'queryId': value.queryId,
        'variables': (0, PostCreateTweetRequestVariables_1.PostCreateTweetRequestVariablesToJSON)(value.variables),
    };
}
exports.PostCreateTweetRequestToJSON = PostCreateTweetRequestToJSON;
