// twitter-openapi-typescript-generated/dist/models/PostCreateTweetRequestVariablesReply.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateTweetRequestVariablesReplyToJSON = exports.PostCreateTweetRequestVariablesReplyFromJSONTyped = exports.PostCreateTweetRequestVariablesReplyFromJSON = exports.instanceOfPostCreateTweetRequestVariablesReply = void 0;
/**
 * Check if a given object implements the PostCreateTweetRequestVariablesReply interface.
 */
function instanceOfPostCreateTweetRequestVariablesReply(value) {
    var isInstance = true;
    isInstance = isInstance && "excludeReplyUserIds" in value;
    isInstance = isInstance && "inReplyToTweetId" in value;
    return isInstance;
}
exports.instanceOfPostCreateTweetRequestVariablesReply = instanceOfPostCreateTweetRequestVariablesReply;
function PostCreateTweetRequestVariablesReplyFromJSON(json) {
    return PostCreateTweetRequestVariablesReplyFromJSONTyped(json, false);
}
exports.PostCreateTweetRequestVariablesReplyFromJSON = PostCreateTweetRequestVariablesReplyFromJSON;
function PostCreateTweetRequestVariablesReplyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'excludeReplyUserIds': json['exclude_reply_user_ids'],
        'inReplyToTweetId': json['in_reply_to_tweet_id'],
    };
}
exports.PostCreateTweetRequestVariablesReplyFromJSONTyped = PostCreateTweetRequestVariablesReplyFromJSONTyped;
function PostCreateTweetRequestVariablesReplyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'exclude_reply_user_ids': value.excludeReplyUserIds,
        'in_reply_to_tweet_id': value.inReplyToTweetId,
    };
}
exports.PostCreateTweetRequestVariablesReplyToJSON = PostCreateTweetRequestVariablesReplyToJSON;
