// twitter-openapi-typescript-generated/dist/models/PostCreateTweetRequestVariablesMedia.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateTweetRequestVariablesMediaToJSON = exports.PostCreateTweetRequestVariablesMediaFromJSONTyped = exports.PostCreateTweetRequestVariablesMediaFromJSON = exports.instanceOfPostCreateTweetRequestVariablesMedia = void 0;
var PostCreateTweetRequestVariablesMediaMediaEntitiesInner_1 = require("./PostCreateTweetRequestVariablesMediaMediaEntitiesInner");
/**
 * Check if a given object implements the PostCreateTweetRequestVariablesMedia interface.
 */
function instanceOfPostCreateTweetRequestVariablesMedia(value) {
    var isInstance = true;
    isInstance = isInstance && "mediaEntities" in value;
    isInstance = isInstance && "possiblySensitive" in value;
    return isInstance;
}
exports.instanceOfPostCreateTweetRequestVariablesMedia = instanceOfPostCreateTweetRequestVariablesMedia;
function PostCreateTweetRequestVariablesMediaFromJSON(json) {
    return PostCreateTweetRequestVariablesMediaFromJSONTyped(json, false);
}
exports.PostCreateTweetRequestVariablesMediaFromJSON = PostCreateTweetRequestVariablesMediaFromJSON;
function PostCreateTweetRequestVariablesMediaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'mediaEntities': (json['media_entities'].map(PostCreateTweetRequestVariablesMediaMediaEntitiesInner_1.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSON)),
        'possiblySensitive': json['possibly_sensitive'],
    };
}
exports.PostCreateTweetRequestVariablesMediaFromJSONTyped = PostCreateTweetRequestVariablesMediaFromJSONTyped;
function PostCreateTweetRequestVariablesMediaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'media_entities': (value.mediaEntities.map(PostCreateTweetRequestVariablesMediaMediaEntitiesInner_1.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerToJSON)),
        'possibly_sensitive': value.possiblySensitive,
    };
}
exports.PostCreateTweetRequestVariablesMediaToJSON = PostCreateTweetRequestVariablesMediaToJSON;
