// twitter-openapi-typescript-generated/dist/models/PostDeleteTweetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostDeleteTweetRequestToJSON = exports.PostDeleteTweetRequestFromJSONTyped = exports.PostDeleteTweetRequestFromJSON = exports.instanceOfPostDeleteTweetRequest = void 0;
var PostCreateRetweetRequestVariables_1 = require("./PostCreateRetweetRequestVariables");
/**
 * Check if a given object implements the PostDeleteTweetRequest interface.
 */
function instanceOfPostDeleteTweetRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "queryId" in value;
    isInstance = isInstance && "variables" in value;
    return isInstance;
}
exports.instanceOfPostDeleteTweetRequest = instanceOfPostDeleteTweetRequest;
function PostDeleteTweetRequestFromJSON(json) {
    return PostDeleteTweetRequestFromJSONTyped(json, false);
}
exports.PostDeleteTweetRequestFromJSON = PostDeleteTweetRequestFromJSON;
function PostDeleteTweetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'queryId': json['queryId'],
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesFromJSON)(json['variables']),
    };
}
exports.PostDeleteTweetRequestFromJSONTyped = PostDeleteTweetRequestFromJSONTyped;
function PostDeleteTweetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'queryId': value.queryId,
        'variables': (0, PostCreateRetweetRequestVariables_1.PostCreateRetweetRequestVariablesToJSON)(value.variables),
    };
}
exports.PostDeleteTweetRequestToJSON = PostDeleteTweetRequestToJSON;
