// twitter-openapi-typescript-generated/dist/models/PostCreateTweetRequestVariablesMediaMediaEntitiesInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerToJSON = exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSONTyped = exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSON = exports.instanceOfPostCreateTweetRequestVariablesMediaMediaEntitiesInner = void 0;
/**
 * Check if a given object implements the PostCreateTweetRequestVariablesMediaMediaEntitiesInner interface.
 */
function instanceOfPostCreateTweetRequestVariablesMediaMediaEntitiesInner(value) {
    var isInstance = true;
    isInstance = isInstance && "mediaId" in value;
    isInstance = isInstance && "taggedUsers" in value;
    return isInstance;
}
exports.instanceOfPostCreateTweetRequestVariablesMediaMediaEntitiesInner = instanceOfPostCreateTweetRequestVariablesMediaMediaEntitiesInner;
function PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSON(json) {
    return PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSONTyped(json, false);
}
exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSON = PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSON;
function PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'mediaId': json['media_id'],
        'taggedUsers': json['tagged_users'],
    };
}
exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSONTyped = PostCreateTweetRequestVariablesMediaMediaEntitiesInnerFromJSONTyped;
function PostCreateTweetRequestVariablesMediaMediaEntitiesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'media_id': value.mediaId,
        'tagged_users': value.taggedUsers,
    };
}
exports.PostCreateTweetRequestVariablesMediaMediaEntitiesInnerToJSON = PostCreateTweetRequestVariablesMediaMediaEntitiesInnerToJSON;
