// twitter-openapi-typescript-generated/dist/models/PostDeleteRetweetRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostDeleteRetweetRequestToJSON = exports.PostDeleteRetweetRequestFromJSONTyped = exports.PostDeleteRetweetRequestFromJSON = exports.instanceOfPostDeleteRetweetRequest = void 0;
var PostDeleteRetweetRequestVariables_1 = require("./PostDeleteRetweetRequestVariables");
/**
 * Check if a given object implements the PostDeleteRetweetRequest interface.
 */
function instanceOfPostDeleteRetweetRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "queryId" in value;
    isInstance = isInstance && "variables" in value;
    return isInstance;
}
exports.instanceOfPostDeleteRetweetRequest = instanceOfPostDeleteRetweetRequest;
function PostDeleteRetweetRequestFromJSON(json) {
    return PostDeleteRetweetRequestFromJSONTyped(json, false);
}
exports.PostDeleteRetweetRequestFromJSON = PostDeleteRetweetRequestFromJSON;
function PostDeleteRetweetRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'queryId': json['queryId'],
        'variables': (0, PostDeleteRetweetRequestVariables_1.PostDeleteRetweetRequestVariablesFromJSON)(json['variables']),
    };
}
exports.PostDeleteRetweetRequestFromJSONTyped = PostDeleteRetweetRequestFromJSONTyped;
function PostDeleteRetweetRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'queryId': value.queryId,
        'variables': (0, PostDeleteRetweetRequestVariables_1.PostDeleteRetweetRequestVariablesToJSON)(value.variables),
    };
}
exports.PostDeleteRetweetRequestToJSON = PostDeleteRetweetRequestToJSON;
