// twitter-openapi-typescript-generated/dist/models/ProfileResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileResponseDataToJSON = exports.ProfileResponseDataFromJSONTyped = exports.ProfileResponseDataFromJSON = exports.instanceOfProfileResponseData = void 0;
var UserResultByScreenName_1 = require("./UserResultByScreenName");
/**
 * Check if a given object implements the ProfileResponseData interface.
 */
function instanceOfProfileResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "userResultByScreenName" in value;
    return isInstance;
}
exports.instanceOfProfileResponseData = instanceOfProfileResponseData;
function ProfileResponseDataFromJSON(json) {
    return ProfileResponseDataFromJSONTyped(json, false);
}
exports.ProfileResponseDataFromJSON = ProfileResponseDataFromJSON;
function ProfileResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'userResultByScreenName': (0, UserResultByScreenName_1.UserResultByScreenNameFromJSON)(json['user_result_by_screen_name']),
    };
}
exports.ProfileResponseDataFromJSONTyped = ProfileResponseDataFromJSONTyped;
function ProfileResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user_result_by_screen_name': (0, UserResultByScreenName_1.UserResultByScreenNameToJSON)(value.userResultByScreenName),
    };
}
exports.ProfileResponseDataToJSON = ProfileResponseDataToJSON;
