// twitter-openapi-typescript-generated/dist/models/ProfileResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileResponseToJSON = exports.ProfileResponseFromJSONTyped = exports.ProfileResponseFromJSON = exports.instanceOfProfileResponse = void 0;
var ProfileResponseData_1 = require("./ProfileResponseData");
/**
 * Check if a given object implements the ProfileResponse interface.
 */
function instanceOfProfileResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfProfileResponse = instanceOfProfileResponse;
function ProfileResponseFromJSON(json) {
    return ProfileResponseFromJSONTyped(json, false);
}
exports.ProfileResponseFromJSON = ProfileResponseFromJSON;
function ProfileResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, ProfileResponseData_1.ProfileResponseDataFromJSON)(json['data']),
    };
}
exports.ProfileResponseFromJSONTyped = ProfileResponseFromJSONTyped;
function ProfileResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, ProfileResponseData_1.ProfileResponseDataToJSON)(value.data),
    };
}
exports.ProfileResponseToJSON = ProfileResponseToJSON;
