// twitter-openapi-typescript-generated/dist/models/RetweetLegacy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetweetLegacyToJSON = exports.RetweetLegacyFromJSONTyped = exports.RetweetLegacyFromJSON = exports.instanceOfRetweetLegacy = void 0;
/**
 * Check if a given object implements the RetweetLegacy interface.
 */
function instanceOfRetweetLegacy(value) {
    var isInstance = true;
    isInstance = isInstance && "fullText" in value;
    return isInstance;
}
exports.instanceOfRetweetLegacy = instanceOfRetweetLegacy;
function RetweetLegacyFromJSON(json) {
    return RetweetLegacyFromJSONTyped(json, false);
}
exports.RetweetLegacyFromJSON = RetweetLegacyFromJSON;
function RetweetLegacyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fullText': json['full_text'],
    };
}
exports.RetweetLegacyFromJSONTyped = RetweetLegacyFromJSONTyped;
function RetweetLegacyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'full_text': value.fullText,
    };
}
exports.RetweetLegacyToJSON = RetweetLegacyToJSON;
