// twitter-openapi-typescript-generated/dist/models/Retweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetweetToJSON = exports.RetweetFromJSONTyped = exports.RetweetFromJSON = exports.instanceOfRetweet = void 0;
var RetweetLegacy_1 = require("./RetweetLegacy");
/**
 * Check if a given object implements the Retweet interface.
 */
function instanceOfRetweet(value) {
    var isInstance = true;
    isInstance = isInstance && "legacy" in value;
    isInstance = isInstance && "restId" in value;
    return isInstance;
}
exports.instanceOfRetweet = instanceOfRetweet;
function RetweetFromJSON(json) {
    return RetweetFromJSONTyped(json, false);
}
exports.RetweetFromJSON = RetweetFromJSON;
function RetweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'legacy': (0, RetweetLegacy_1.RetweetLegacyFromJSON)(json['legacy']),
        'restId': json['rest_id'],
    };
}
exports.RetweetFromJSONTyped = RetweetFromJSONTyped;
function RetweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'legacy': (0, RetweetLegacy_1.RetweetLegacyToJSON)(value.legacy),
        'rest_id': value.restId,
    };
}
exports.RetweetToJSON = RetweetToJSON;
