// twitter-openapi-typescript-generated/dist/models/SearchTimelineData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTimelineDataToJSON = exports.SearchTimelineDataFromJSONTyped = exports.SearchTimelineDataFromJSON = exports.instanceOfSearchTimelineData = void 0;
var SearchByRawQuery_1 = require("./SearchByRawQuery");
/**
 * Check if a given object implements the SearchTimelineData interface.
 */
function instanceOfSearchTimelineData(value) {
    var isInstance = true;
    isInstance = isInstance && "searchByRawQuery" in value;
    return isInstance;
}
exports.instanceOfSearchTimelineData = instanceOfSearchTimelineData;
function SearchTimelineDataFromJSON(json) {
    return SearchTimelineDataFromJSONTyped(json, false);
}
exports.SearchTimelineDataFromJSON = SearchTimelineDataFromJSON;
function SearchTimelineDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'searchByRawQuery': (0, SearchByRawQuery_1.SearchByRawQueryFromJSON)(json['search_by_raw_query']),
    };
}
exports.SearchTimelineDataFromJSONTyped = SearchTimelineDataFromJSONTyped;
function SearchTimelineDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'search_by_raw_query': (0, SearchByRawQuery_1.SearchByRawQueryToJSON)(value.searchByRawQuery),
    };
}
exports.SearchTimelineDataToJSON = SearchTimelineDataToJSON;
