// twitter-openapi-typescript-generated/dist/models/SearchTimelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTimelineResponseToJSON = exports.SearchTimelineResponseFromJSONTyped = exports.SearchTimelineResponseFromJSON = exports.instanceOfSearchTimelineResponse = void 0;
var SearchTimelineData_1 = require("./SearchTimelineData");
/**
 * Check if a given object implements the SearchTimelineResponse interface.
 */
function instanceOfSearchTimelineResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfSearchTimelineResponse = instanceOfSearchTimelineResponse;
function SearchTimelineResponseFromJSON(json) {
    return SearchTimelineResponseFromJSONTyped(json, false);
}
exports.SearchTimelineResponseFromJSON = SearchTimelineResponseFromJSON;
function SearchTimelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, SearchTimelineData_1.SearchTimelineDataFromJSON)(json['data']),
    };
}
exports.SearchTimelineResponseFromJSONTyped = SearchTimelineResponseFromJSONTyped;
function SearchTimelineResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, SearchTimelineData_1.SearchTimelineDataToJSON)(value.data),
    };
}
exports.SearchTimelineResponseToJSON = SearchTimelineResponseToJSON;
