// twitter-openapi-typescript-generated/dist/models/SearchByRawQuery.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchByRawQueryToJSON = exports.SearchByRawQueryFromJSONTyped = exports.SearchByRawQueryFromJSON = exports.instanceOfSearchByRawQuery = void 0;
var SearchTimeline_1 = require("./SearchTimeline");
/**
 * Check if a given object implements the SearchByRawQuery interface.
 */
function instanceOfSearchByRawQuery(value) {
    var isInstance = true;
    isInstance = isInstance && "searchTimeline" in value;
    return isInstance;
}
exports.instanceOfSearchByRawQuery = instanceOfSearchByRawQuery;
function SearchByRawQueryFromJSON(json) {
    return SearchByRawQueryFromJSONTyped(json, false);
}
exports.SearchByRawQueryFromJSON = SearchByRawQueryFromJSON;
function SearchByRawQueryFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'searchTimeline': (0, SearchTimeline_1.SearchTimelineFromJSON)(json['search_timeline']),
    };
}
exports.SearchByRawQueryFromJSONTyped = SearchByRawQueryFromJSONTyped;
function SearchByRawQueryToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'search_timeline': (0, SearchTimeline_1.SearchTimelineToJSON)(value.searchTimeline),
    };
}
exports.SearchByRawQueryToJSON = SearchByRawQueryToJSON;
