// twitter-openapi-typescript-generated/dist/models/SelfThread.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SelfThreadToJSON = exports.SelfThreadFromJSONTyped = exports.SelfThreadFromJSON = exports.instanceOfSelfThread = void 0;
/**
 * Check if a given object implements the SelfThread interface.
 */
function instanceOfSelfThread(value) {
    var isInstance = true;
    isInstance = isInstance && "idStr" in value;
    return isInstance;
}
exports.instanceOfSelfThread = instanceOfSelfThread;
function SelfThreadFromJSON(json) {
    return SelfThreadFromJSONTyped(json, false);
}
exports.SelfThreadFromJSON = SelfThreadFromJSON;
function SelfThreadFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'idStr': json['id_str'],
    };
}
exports.SelfThreadFromJSONTyped = SelfThreadFromJSONTyped;
function SelfThreadToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id_str': value.idStr,
    };
}
exports.SelfThreadToJSON = SelfThreadToJSON;
