// twitter-openapi-typescript-generated/dist/models/SocialContext.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocialContextToJSON = exports.SocialContextFromJSONTyped = exports.SocialContextFromJSON = exports.instanceOfSocialContext = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the SocialContext interface.
 */
function instanceOfSocialContext(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfSocialContext = instanceOfSocialContext;
function SocialContextFromJSON(json) {
    return SocialContextFromJSONTyped(json, false);
}
exports.SocialContextFromJSON = SocialContextFromJSON;
function SocialContextFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'contextType': !(0, runtime_1.exists)(json, 'contextType') ? undefined : json['contextType'],
        'text': !(0, runtime_1.exists)(json, 'text') ? undefined : json['text'],
        'type': !(0, runtime_1.exists)(json, 'type') ? undefined : json['type'],
    };
}
exports.SocialContextFromJSONTyped = SocialContextFromJSONTyped;
function SocialContextToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'contextType': value.contextType,
        'text': value.text,
        'type': value.type,
    };
}
exports.SocialContextToJSON = SocialContextToJSON;
