// twitter-openapi-typescript-generated/dist/models/Text.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextToJSON = exports.TextFromJSONTyped = exports.TextFromJSON = exports.instanceOfText = void 0;
var TextEntity_1 = require("./TextEntity");
/**
 * Check if a given object implements the Text interface.
 */
function instanceOfText(value) {
    var isInstance = true;
    isInstance = isInstance && "entities" in value;
    isInstance = isInstance && "text" in value;
    return isInstance;
}
exports.instanceOfText = instanceOfText;
function TextFromJSON(json) {
    return TextFromJSONTyped(json, false);
}
exports.TextFromJSON = TextFromJSON;
function TextFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entities': (json['entities'].map(TextEntity_1.TextEntityFromJSON)),
        'text': json['text'],
    };
}
exports.TextFromJSONTyped = TextFromJSONTyped;
function TextToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entities': (value.entities.map(TextEntity_1.TextEntityToJSON)),
        'text': value.text,
    };
}
exports.TextToJSON = TextToJSON;
