// twitter-openapi-typescript-generated/dist/models/TextEntityRef.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextEntityRefToJSON = exports.TextEntityRefFromJSONTyped = exports.TextEntityRefFromJSON = exports.instanceOfTextEntityRef = exports.TextEntityRefUrlTypeEnum = exports.TextEntityRefTypeEnum = void 0;
/**
 * @export
 */
exports.TextEntityRefTypeEnum = {
    TimelineUrl: 'TimelineUrl'
};
/**
 * @export
 */
exports.TextEntityRefUrlTypeEnum = {
    ExternalUrl: 'ExternalUrl'
};
/**
 * Check if a given object implements the TextEntityRef interface.
 */
function instanceOfTextEntityRef(value) {
    var isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "urlType" in value;
    return isInstance;
}
exports.instanceOfTextEntityRef = instanceOfTextEntityRef;
function TextEntityRefFromJSON(json) {
    return TextEntityRefFromJSONTyped(json, false);
}
exports.TextEntityRefFromJSON = TextEntityRefFromJSON;
function TextEntityRefFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'url': json['url'],
        'urlType': json['urlType'],
    };
}
exports.TextEntityRefFromJSONTyped = TextEntityRefFromJSONTyped;
function TextEntityRefToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'url': value.url,
        'urlType': value.urlType,
    };
}
exports.TextEntityRefToJSON = TextEntityRefToJSON;
