// twitter-openapi-typescript-generated/dist/models/TextEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextEntityToJSON = exports.TextEntityFromJSONTyped = exports.TextEntityFromJSON = exports.instanceOfTextEntity = void 0;
var TextEntityRef_1 = require("./TextEntityRef");
/**
 * Check if a given object implements the TextEntity interface.
 */
function instanceOfTextEntity(value) {
    var isInstance = true;
    isInstance = isInstance && "fromIndex" in value;
    isInstance = isInstance && "ref" in value;
    isInstance = isInstance && "toIndex" in value;
    return isInstance;
}
exports.instanceOfTextEntity = instanceOfTextEntity;
function TextEntityFromJSON(json) {
    return TextEntityFromJSONTyped(json, false);
}
exports.TextEntityFromJSON = TextEntityFromJSON;
function TextEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fromIndex': json['fromIndex'],
        'ref': (0, TextEntityRef_1.TextEntityRefFromJSON)(json['ref']),
        'toIndex': json['toIndex'],
    };
}
exports.TextEntityFromJSONTyped = TextEntityFromJSONTyped;
function TextEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fromIndex': value.fromIndex,
        'ref': (0, TextEntityRef_1.TextEntityRefToJSON)(value.ref),
        'toIndex': value.toIndex,
    };
}
exports.TextEntityToJSON = TextEntityToJSON;
