// twitter-openapi-typescript-generated/dist/models/TimelineCoverBehavior.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineCoverBehaviorToJSON = exports.TimelineCoverBehaviorFromJSONTyped = exports.TimelineCoverBehaviorFromJSON = exports.instanceOfTimelineCoverBehavior = exports.TimelineCoverBehaviorTypeEnum = void 0;
/**
 * @export
 */
exports.TimelineCoverBehaviorTypeEnum = {
    TimelineCoverBehaviorDismiss: 'TimelineCoverBehaviorDismiss'
};
/**
 * Check if a given object implements the TimelineCoverBehavior interface.
 */
function instanceOfTimelineCoverBehavior(value) {
    var isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineCoverBehavior = instanceOfTimelineCoverBehavior;
function TimelineCoverBehaviorFromJSON(json) {
    return TimelineCoverBehaviorFromJSONTyped(json, false);
}
exports.TimelineCoverBehaviorFromJSON = TimelineCoverBehaviorFromJSON;
function TimelineCoverBehaviorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
    };
}
exports.TimelineCoverBehaviorFromJSONTyped = TimelineCoverBehaviorFromJSONTyped;
function TimelineCoverBehaviorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
    };
}
exports.TimelineCoverBehaviorToJSON = TimelineCoverBehaviorToJSON;
