// twitter-openapi-typescript-generated/dist/models/TimelineAddEntry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineAddEntryToJSON = exports.TimelineAddEntryFromJSONTyped = exports.TimelineAddEntryFromJSON = exports.instanceOfTimelineAddEntry = void 0;
var ContentUnion_1 = require("./ContentUnion");
/**
 * Check if a given object implements the TimelineAddEntry interface.
 */
function instanceOfTimelineAddEntry(value) {
    var isInstance = true;
    isInstance = isInstance && "content" in value;
    isInstance = isInstance && "entryId" in value;
    isInstance = isInstance && "sortIndex" in value;
    return isInstance;
}
exports.instanceOfTimelineAddEntry = instanceOfTimelineAddEntry;
function TimelineAddEntryFromJSON(json) {
    return TimelineAddEntryFromJSONTyped(json, false);
}
exports.TimelineAddEntryFromJSON = TimelineAddEntryFromJSON;
function TimelineAddEntryFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'content': (0, ContentUnion_1.ContentUnionFromJSON)(json['content']),
        'entryId': json['entryId'],
        'sortIndex': json['sortIndex'],
    };
}
exports.TimelineAddEntryFromJSONTyped = TimelineAddEntryFromJSONTyped;
function TimelineAddEntryToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'content': (0, ContentUnion_1.ContentUnionToJSON)(value.content),
        'entryId': value.entryId,
        'sortIndex': value.sortIndex,
    };
}
exports.TimelineAddEntryToJSON = TimelineAddEntryToJSON;
