// twitter-openapi-typescript-generated/dist/models/TimelineClearCache.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineClearCacheToJSON = exports.TimelineClearCacheFromJSONTyped = exports.TimelineClearCacheFromJSON = exports.instanceOfTimelineClearCache = void 0;
var InstructionType_1 = require("./InstructionType");
/**
 * Check if a given object implements the TimelineClearCache interface.
 */
function instanceOfTimelineClearCache(value) {
    var isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineClearCache = instanceOfTimelineClearCache;
function TimelineClearCacheFromJSON(json) {
    return TimelineClearCacheFromJSONTyped(json, false);
}
exports.TimelineClearCacheFromJSON = TimelineClearCacheFromJSON;
function TimelineClearCacheFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineClearCacheFromJSONTyped = TimelineClearCacheFromJSONTyped;
function TimelineClearCacheToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineClearCacheToJSON = TimelineClearCacheToJSON;
