// twitter-openapi-typescript-generated/dist/models/TimelineReplaceEntry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineReplaceEntryToJSON = exports.TimelineReplaceEntryFromJSONTyped = exports.TimelineReplaceEntryFromJSON = exports.instanceOfTimelineReplaceEntry = void 0;
var InstructionType_1 = require("./InstructionType");
var TimelineAddEntry_1 = require("./TimelineAddEntry");
/**
 * Check if a given object implements the TimelineReplaceEntry interface.
 */
function instanceOfTimelineReplaceEntry(value) {
    var isInstance = true;
    isInstance = isInstance && "entry" in value;
    isInstance = isInstance && "entryIdToReplace" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineReplaceEntry = instanceOfTimelineReplaceEntry;
function TimelineReplaceEntryFromJSON(json) {
    return TimelineReplaceEntryFromJSONTyped(json, false);
}
exports.TimelineReplaceEntryFromJSON = TimelineReplaceEntryFromJSON;
function TimelineReplaceEntryFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entry': (0, TimelineAddEntry_1.TimelineAddEntryFromJSON)(json['entry']),
        'entryIdToReplace': json['entry_id_to_replace'],
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineReplaceEntryFromJSONTyped = TimelineReplaceEntryFromJSONTyped;
function TimelineReplaceEntryToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entry': (0, TimelineAddEntry_1.TimelineAddEntryToJSON)(value.entry),
        'entry_id_to_replace': value.entryIdToReplace,
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineReplaceEntryToJSON = TimelineReplaceEntryToJSON;
