// twitter-openapi-typescript-generated/dist/models/TimelineShowCover.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineShowCoverToJSON = exports.TimelineShowCoverFromJSONTyped = exports.TimelineShowCoverFromJSON = exports.instanceOfTimelineShowCover = void 0;
var ClientEventInfo_1 = require("./ClientEventInfo");
var InstructionType_1 = require("./InstructionType");
var TimelineHalfCover_1 = require("./TimelineHalfCover");
/**
 * Check if a given object implements the TimelineShowCover interface.
 */
function instanceOfTimelineShowCover(value) {
    var isInstance = true;
    isInstance = isInstance && "clientEventInfo" in value;
    isInstance = isInstance && "cover" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineShowCover = instanceOfTimelineShowCover;
function TimelineShowCoverFromJSON(json) {
    return TimelineShowCoverFromJSONTyped(json, false);
}
exports.TimelineShowCoverFromJSON = TimelineShowCoverFromJSON;
function TimelineShowCoverFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientEventInfo': (0, ClientEventInfo_1.ClientEventInfoFromJSON)(json['clientEventInfo']),
        'cover': (0, TimelineHalfCover_1.TimelineHalfCoverFromJSON)(json['cover']),
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineShowCoverFromJSONTyped = TimelineShowCoverFromJSONTyped;
function TimelineShowCoverToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientEventInfo': (0, ClientEventInfo_1.ClientEventInfoToJSON)(value.clientEventInfo),
        'cover': (0, TimelineHalfCover_1.TimelineHalfCoverToJSON)(value.cover),
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineShowCoverToJSON = TimelineShowCoverToJSON;
