// twitter-openapi-typescript-generated/dist/models/TimelineV2.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineV2ToJSON = exports.TimelineV2FromJSONTyped = exports.TimelineV2FromJSON = exports.instanceOfTimelineV2 = void 0;
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the TimelineV2 interface.
 */
function instanceOfTimelineV2(value) {
    var isInstance = true;
    isInstance = isInstance && "timeline" in value;
    return isInstance;
}
exports.instanceOfTimelineV2 = instanceOfTimelineV2;
function TimelineV2FromJSON(json) {
    return TimelineV2FromJSONTyped(json, false);
}
exports.TimelineV2FromJSON = TimelineV2FromJSON;
function TimelineV2FromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'timeline': (0, Timeline_1.TimelineFromJSON)(json['timeline']),
    };
}
exports.TimelineV2FromJSONTyped = TimelineV2FromJSONTyped;
function TimelineV2ToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'timeline': (0, Timeline_1.TimelineToJSON)(value.timeline),
    };
}
exports.TimelineV2ToJSON = TimelineV2ToJSON;
