// twitter-openapi-typescript-generated/dist/models/TimelinePrompt.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelinePromptToJSON = exports.TimelinePromptFromJSONTyped = exports.TimelinePromptFromJSON = exports.instanceOfTimelinePrompt = void 0;
var runtime_1 = require("../runtime");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the TimelinePrompt interface.
 */
function instanceOfTimelinePrompt(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTimelinePrompt = instanceOfTimelinePrompt;
function TimelinePromptFromJSON(json) {
    return TimelinePromptFromJSONTyped(json, false);
}
exports.TimelinePromptFromJSON = TimelinePromptFromJSON;
function TimelinePromptFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return __assign(__assign({}, json), { 'typename': !(0, runtime_1.exists)(json, '__typename') ? undefined : (0, TypeName_1.TypeNameFromJSON)(json['__typename']) });
}
exports.TimelinePromptFromJSONTyped = TimelinePromptFromJSONTyped;
function TimelinePromptToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return __assign(__assign({}, value), { '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename) });
}
exports.TimelinePromptToJSON = TimelinePromptToJSON;
