// twitter-openapi-typescript-generated/dist/models/TimelineResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineResponseToJSON = exports.TimelineResponseFromJSONTyped = exports.TimelineResponseFromJSON = exports.instanceOfTimelineResponse = void 0;
var HomeTimelineResponseData_1 = require("./HomeTimelineResponseData");
/**
 * Check if a given object implements the TimelineResponse interface.
 */
function instanceOfTimelineResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfTimelineResponse = instanceOfTimelineResponse;
function TimelineResponseFromJSON(json) {
    return TimelineResponseFromJSONTyped(json, false);
}
exports.TimelineResponseFromJSON = TimelineResponseFromJSON;
function TimelineResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, HomeTimelineResponseData_1.HomeTimelineResponseDataFromJSON)(json['data']),
    };
}
exports.TimelineResponseFromJSONTyped = TimelineResponseFromJSONTyped;
function TimelineResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, HomeTimelineResponseData_1.HomeTimelineResponseDataToJSON)(value.data),
    };
}
exports.TimelineResponseToJSON = TimelineResponseToJSON;
