// twitter-openapi-typescript-generated/dist/models/TimelineTweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineTweetToJSON = exports.TimelineTweetFromJSONTyped = exports.TimelineTweetFromJSON = exports.instanceOfTimelineTweet = void 0;
var runtime_1 = require("../runtime");
var ContentItemType_1 = require("./ContentItemType");
var ItemResult_1 = require("./ItemResult");
var SocialContext_1 = require("./SocialContext");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the TimelineTweet interface.
 */
function instanceOfTimelineTweet(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "itemType" in value;
    isInstance = isInstance && "tweetDisplayType" in value;
    isInstance = isInstance && "tweetResults" in value;
    return isInstance;
}
exports.instanceOfTimelineTweet = instanceOfTimelineTweet;
function TimelineTweetFromJSON(json) {
    return TimelineTweetFromJSONTyped(json, false);
}
exports.TimelineTweetFromJSON = TimelineTweetFromJSON;
function TimelineTweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'socialContext': !(0, runtime_1.exists)(json, 'SocialContext') ? undefined : (0, SocialContext_1.SocialContextFromJSON)(json['SocialContext']),
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'itemType': (0, ContentItemType_1.ContentItemTypeFromJSON)(json['itemType']),
        'promotedMetadata': !(0, runtime_1.exists)(json, 'promotedMetadata') ? undefined : json['promotedMetadata'],
        'tweetDisplayType': json['tweetDisplayType'],
        'tweetResults': (0, ItemResult_1.ItemResultFromJSON)(json['tweet_results']),
    };
}
exports.TimelineTweetFromJSONTyped = TimelineTweetFromJSONTyped;
function TimelineTweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'SocialContext': (0, SocialContext_1.SocialContextToJSON)(value.socialContext),
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'itemType': (0, ContentItemType_1.ContentItemTypeToJSON)(value.itemType),
        'promotedMetadata': value.promotedMetadata,
        'tweetDisplayType': value.tweetDisplayType,
        'tweet_results': (0, ItemResult_1.ItemResultToJSON)(value.tweetResults),
    };
}
exports.TimelineTweetToJSON = TimelineTweetToJSON;
