// twitter-openapi-typescript-generated/dist/models/TimelineTimelineCursor.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineTimelineCursorToJSON = exports.TimelineTimelineCursorFromJSONTyped = exports.TimelineTimelineCursorFromJSON = exports.instanceOfTimelineTimelineCursor = void 0;
var runtime_1 = require("../runtime");
var ContentEntryType_1 = require("./ContentEntryType");
var CursorType_1 = require("./CursorType");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the TimelineTimelineCursor interface.
 */
function instanceOfTimelineTimelineCursor(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "cursorType" in value;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
exports.instanceOfTimelineTimelineCursor = instanceOfTimelineTimelineCursor;
function TimelineTimelineCursorFromJSON(json) {
    return TimelineTimelineCursorFromJSONTyped(json, false);
}
exports.TimelineTimelineCursorFromJSON = TimelineTimelineCursorFromJSON;
function TimelineTimelineCursorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'cursorType': (0, CursorType_1.CursorTypeFromJSON)(json['cursorType']),
        'entryType': !(0, runtime_1.exists)(json, 'entryType') ? undefined : (0, ContentEntryType_1.ContentEntryTypeFromJSON)(json['entryType']),
        'itemType': !(0, runtime_1.exists)(json, 'itemType') ? undefined : (0, ContentEntryType_1.ContentEntryTypeFromJSON)(json['itemType']),
        'value': json['value'],
    };
}
exports.TimelineTimelineCursorFromJSONTyped = TimelineTimelineCursorFromJSONTyped;
function TimelineTimelineCursorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'cursorType': (0, CursorType_1.CursorTypeToJSON)(value.cursorType),
        'entryType': (0, ContentEntryType_1.ContentEntryTypeToJSON)(value.entryType),
        'itemType': (0, ContentEntryType_1.ContentEntryTypeToJSON)(value.itemType),
        'value': value.value,
    };
}
exports.TimelineTimelineCursorToJSON = TimelineTimelineCursorToJSON;
