// twitter-openapi-typescript-generated/dist/models/Timeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineToJSON = exports.TimelineFromJSONTyped = exports.TimelineFromJSON = exports.instanceOfTimeline = void 0;
var runtime_1 = require("../runtime");
var InstructionUnion_1 = require("./InstructionUnion");
/**
 * Check if a given object implements the Timeline interface.
 */
function instanceOfTimeline(value) {
    var isInstance = true;
    isInstance = isInstance && "instructions" in value;
    return isInstance;
}
exports.instanceOfTimeline = instanceOfTimeline;
function TimelineFromJSON(json) {
    return TimelineFromJSONTyped(json, false);
}
exports.TimelineFromJSON = TimelineFromJSON;
function TimelineFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'instructions': (json['instructions'].map(InstructionUnion_1.InstructionUnionFromJSON)),
        'metadata': !(0, runtime_1.exists)(json, 'metadata') ? undefined : json['metadata'],
        'responseObjects': !(0, runtime_1.exists)(json, 'responseObjects') ? undefined : json['responseObjects'],
    };
}
exports.TimelineFromJSONTyped = TimelineFromJSONTyped;
function TimelineToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'instructions': (value.instructions.map(InstructionUnion_1.InstructionUnionToJSON)),
        'metadata': value.metadata,
        'responseObjects': value.responseObjects,
    };
}
exports.TimelineToJSON = TimelineToJSON;
