// twitter-openapi-typescript-generated/dist/models/TimelineTerminateTimeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineTerminateTimelineToJSON = exports.TimelineTerminateTimelineFromJSONTyped = exports.TimelineTerminateTimelineFromJSON = exports.instanceOfTimelineTerminateTimeline = exports.TimelineTerminateTimelineDirectionEnum = void 0;
var InstructionType_1 = require("./InstructionType");
/**
 * @export
 */
exports.TimelineTerminateTimelineDirectionEnum = {
    Top: 'Top',
    Bottom: 'Bottom'
};
/**
 * Check if a given object implements the TimelineTerminateTimeline interface.
 */
function instanceOfTimelineTerminateTimeline(value) {
    var isInstance = true;
    isInstance = isInstance && "direction" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineTerminateTimeline = instanceOfTimelineTerminateTimeline;
function TimelineTerminateTimelineFromJSON(json) {
    return TimelineTerminateTimelineFromJSONTyped(json, false);
}
exports.TimelineTerminateTimelineFromJSON = TimelineTerminateTimelineFromJSON;
function TimelineTerminateTimelineFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'direction': json['direction'],
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineTerminateTimelineFromJSONTyped = TimelineTerminateTimelineFromJSONTyped;
function TimelineTerminateTimelineToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'direction': value.direction,
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineTerminateTimelineToJSON = TimelineTerminateTimelineToJSON;
