// twitter-openapi-typescript-generated/dist/models/TimelineAddToModule.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineAddToModuleToJSON = exports.TimelineAddToModuleFromJSONTyped = exports.TimelineAddToModuleFromJSON = exports.instanceOfTimelineAddToModule = void 0;
var runtime_1 = require("../runtime");
var InstructionType_1 = require("./InstructionType");
var ModuleItem_1 = require("./ModuleItem");
/**
 * Check if a given object implements the TimelineAddToModule interface.
 */
function instanceOfTimelineAddToModule(value) {
    var isInstance = true;
    isInstance = isInstance && "moduleEntryId" in value;
    isInstance = isInstance && "moduleItems" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineAddToModule = instanceOfTimelineAddToModule;
function TimelineAddToModuleFromJSON(json) {
    return TimelineAddToModuleFromJSONTyped(json, false);
}
exports.TimelineAddToModuleFromJSON = TimelineAddToModuleFromJSON;
function TimelineAddToModuleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'moduleEntryId': json['moduleEntryId'],
        'moduleItems': (json['moduleItems'].map(ModuleItem_1.ModuleItemFromJSON)),
        'prepend': !(0, runtime_1.exists)(json, 'prepend') ? undefined : json['prepend'],
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineAddToModuleFromJSONTyped = TimelineAddToModuleFromJSONTyped;
function TimelineAddToModuleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'moduleEntryId': value.moduleEntryId,
        'moduleItems': (value.moduleItems.map(ModuleItem_1.ModuleItemToJSON)),
        'prepend': value.prepend,
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineAddToModuleToJSON = TimelineAddToModuleToJSON;
