// twitter-openapi-typescript-generated/dist/models/TimelineTimelineItem.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineTimelineItemToJSON = exports.TimelineTimelineItemFromJSONTyped = exports.TimelineTimelineItemFromJSON = exports.instanceOfTimelineTimelineItem = void 0;
var runtime_1 = require("../runtime");
var ClientEventInfo_1 = require("./ClientEventInfo");
var ContentEntryType_1 = require("./ContentEntryType");
var ItemContentUnion_1 = require("./ItemContentUnion");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the TimelineTimelineItem interface.
 */
function instanceOfTimelineTimelineItem(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "entryType" in value;
    isInstance = isInstance && "itemContent" in value;
    return isInstance;
}
exports.instanceOfTimelineTimelineItem = instanceOfTimelineTimelineItem;
function TimelineTimelineItemFromJSON(json) {
    return TimelineTimelineItemFromJSONTyped(json, false);
}
exports.TimelineTimelineItemFromJSON = TimelineTimelineItemFromJSON;
function TimelineTimelineItemFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'clientEventInfo': !(0, runtime_1.exists)(json, 'clientEventInfo') ? undefined : (0, ClientEventInfo_1.ClientEventInfoFromJSON)(json['clientEventInfo']),
        'entryType': (0, ContentEntryType_1.ContentEntryTypeFromJSON)(json['entryType']),
        'feedbackInfo': !(0, runtime_1.exists)(json, 'feedbackInfo') ? undefined : json['feedbackInfo'],
        'itemContent': (0, ItemContentUnion_1.ItemContentUnionFromJSON)(json['itemContent']),
    };
}
exports.TimelineTimelineItemFromJSONTyped = TimelineTimelineItemFromJSONTyped;
function TimelineTimelineItemToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'clientEventInfo': (0, ClientEventInfo_1.ClientEventInfoToJSON)(value.clientEventInfo),
        'entryType': (0, ContentEntryType_1.ContentEntryTypeToJSON)(value.entryType),
        'feedbackInfo': value.feedbackInfo,
        'itemContent': (0, ItemContentUnion_1.ItemContentUnionToJSON)(value.itemContent),
    };
}
exports.TimelineTimelineItemToJSON = TimelineTimelineItemToJSON;
