// twitter-openapi-typescript-generated/dist/models/TimelineAddEntries.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineAddEntriesToJSON = exports.TimelineAddEntriesFromJSONTyped = exports.TimelineAddEntriesFromJSON = exports.instanceOfTimelineAddEntries = void 0;
var InstructionType_1 = require("./InstructionType");
var TimelineAddEntry_1 = require("./TimelineAddEntry");
/**
 * Check if a given object implements the TimelineAddEntries interface.
 */
function instanceOfTimelineAddEntries(value) {
    var isInstance = true;
    isInstance = isInstance && "entries" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelineAddEntries = instanceOfTimelineAddEntries;
function TimelineAddEntriesFromJSON(json) {
    return TimelineAddEntriesFromJSONTyped(json, false);
}
exports.TimelineAddEntriesFromJSON = TimelineAddEntriesFromJSON;
function TimelineAddEntriesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entries': (json['entries'].map(TimelineAddEntry_1.TimelineAddEntryFromJSON)),
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelineAddEntriesFromJSONTyped = TimelineAddEntriesFromJSONTyped;
function TimelineAddEntriesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entries': (value.entries.map(TimelineAddEntry_1.TimelineAddEntryToJSON)),
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelineAddEntriesToJSON = TimelineAddEntriesToJSON;
