// twitter-openapi-typescript-generated/dist/models/TimelineTimelineModule.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineTimelineModuleToJSON = exports.TimelineTimelineModuleFromJSONTyped = exports.TimelineTimelineModuleFromJSON = exports.instanceOfTimelineTimelineModule = exports.TimelineTimelineModuleDisplayTypeEnum = void 0;
var runtime_1 = require("../runtime");
var ContentEntryType_1 = require("./ContentEntryType");
var ModuleItem_1 = require("./ModuleItem");
var TypeName_1 = require("./TypeName");
/**
 * @export
 */
exports.TimelineTimelineModuleDisplayTypeEnum = {
    Vertical: 'Vertical',
    VerticalConversation: 'VerticalConversation',
    Carousel: 'Carousel'
};
/**
 * Check if a given object implements the TimelineTimelineModule interface.
 */
function instanceOfTimelineTimelineModule(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "clientEventInfo" in value;
    isInstance = isInstance && "displayType" in value;
    isInstance = isInstance && "entryType" in value;
    return isInstance;
}
exports.instanceOfTimelineTimelineModule = instanceOfTimelineTimelineModule;
function TimelineTimelineModuleFromJSON(json) {
    return TimelineTimelineModuleFromJSONTyped(json, false);
}
exports.TimelineTimelineModuleFromJSON = TimelineTimelineModuleFromJSON;
function TimelineTimelineModuleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'clientEventInfo': json['clientEventInfo'],
        'displayType': json['displayType'],
        'entryType': (0, ContentEntryType_1.ContentEntryTypeFromJSON)(json['entryType']),
        'footer': !(0, runtime_1.exists)(json, 'footer') ? undefined : json['footer'],
        'header': !(0, runtime_1.exists)(json, 'header') ? undefined : json['header'],
        'items': !(0, runtime_1.exists)(json, 'items') ? undefined : (json['items'].map(ModuleItem_1.ModuleItemFromJSON)),
    };
}
exports.TimelineTimelineModuleFromJSONTyped = TimelineTimelineModuleFromJSONTyped;
function TimelineTimelineModuleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'clientEventInfo': value.clientEventInfo,
        'displayType': value.displayType,
        'entryType': (0, ContentEntryType_1.ContentEntryTypeToJSON)(value.entryType),
        'footer': value.footer,
        'header': value.header,
        'items': value.items === undefined ? undefined : (value.items.map(ModuleItem_1.ModuleItemToJSON)),
    };
}
exports.TimelineTimelineModuleToJSON = TimelineTimelineModuleToJSON;
