// twitter-openapi-typescript-generated/dist/models/TimelinePinEntry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelinePinEntryToJSON = exports.TimelinePinEntryFromJSONTyped = exports.TimelinePinEntryFromJSON = exports.instanceOfTimelinePinEntry = void 0;
var InstructionType_1 = require("./InstructionType");
var TimelineAddEntry_1 = require("./TimelineAddEntry");
/**
 * Check if a given object implements the TimelinePinEntry interface.
 */
function instanceOfTimelinePinEntry(value) {
    var isInstance = true;
    isInstance = isInstance && "entry" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTimelinePinEntry = instanceOfTimelinePinEntry;
function TimelinePinEntryFromJSON(json) {
    return TimelinePinEntryFromJSONTyped(json, false);
}
exports.TimelinePinEntryFromJSON = TimelinePinEntryFromJSON;
function TimelinePinEntryFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entry': (0, TimelineAddEntry_1.TimelineAddEntryFromJSON)(json['entry']),
        'type': (0, InstructionType_1.InstructionTypeFromJSON)(json['type']),
    };
}
exports.TimelinePinEntryFromJSONTyped = TimelinePinEntryFromJSONTyped;
function TimelinePinEntryToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entry': (0, TimelineAddEntry_1.TimelineAddEntryToJSON)(value.entry),
        'type': (0, InstructionType_1.InstructionTypeToJSON)(value.type),
    };
}
exports.TimelinePinEntryToJSON = TimelinePinEntryToJSON;
