// twitter-openapi-typescript-generated/dist/models/TimelineShowAlertRichText.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineShowAlertRichTextToJSON = exports.TimelineShowAlertRichTextFromJSONTyped = exports.TimelineShowAlertRichTextFromJSON = exports.instanceOfTimelineShowAlertRichText = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TimelineShowAlertRichText interface.
 */
function instanceOfTimelineShowAlertRichText(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTimelineShowAlertRichText = instanceOfTimelineShowAlertRichText;
function TimelineShowAlertRichTextFromJSON(json) {
    return TimelineShowAlertRichTextFromJSONTyped(json, false);
}
exports.TimelineShowAlertRichTextFromJSON = TimelineShowAlertRichTextFromJSON;
function TimelineShowAlertRichTextFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'entities': !(0, runtime_1.exists)(json, 'entities') ? undefined : json['entities'],
        'text': !(0, runtime_1.exists)(json, 'text') ? undefined : json['text'],
    };
}
exports.TimelineShowAlertRichTextFromJSONTyped = TimelineShowAlertRichTextFromJSONTyped;
function TimelineShowAlertRichTextToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'entities': value.entities,
        'text': value.text,
    };
}
exports.TimelineShowAlertRichTextToJSON = TimelineShowAlertRichTextToJSON;
