// twitter-openapi-typescript-generated/dist/models/Tracing.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TracingToJSON = exports.TracingFromJSONTyped = exports.TracingFromJSON = exports.instanceOfTracing = void 0;
/**
 * Check if a given object implements the Tracing interface.
 */
function instanceOfTracing(value) {
    var isInstance = true;
    isInstance = isInstance && "traceId" in value;
    return isInstance;
}
exports.instanceOfTracing = instanceOfTracing;
function TracingFromJSON(json) {
    return TracingFromJSONTyped(json, false);
}
exports.TracingFromJSON = TracingFromJSON;
function TracingFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'traceId': json['trace_id'],
    };
}
exports.TracingFromJSONTyped = TracingFromJSONTyped;
function TracingToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'trace_id': value.traceId,
    };
}
exports.TracingToJSON = TracingToJSON;
