// twitter-openapi-typescript-generated/dist/models/TweetFavoritersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetFavoritersResponseToJSON = exports.TweetFavoritersResponseFromJSONTyped = exports.TweetFavoritersResponseFromJSON = exports.instanceOfTweetFavoritersResponse = void 0;
var TweetFavoritersResponseData_1 = require("./TweetFavoritersResponseData");
/**
 * Check if a given object implements the TweetFavoritersResponse interface.
 */
function instanceOfTweetFavoritersResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfTweetFavoritersResponse = instanceOfTweetFavoritersResponse;
function TweetFavoritersResponseFromJSON(json) {
    return TweetFavoritersResponseFromJSONTyped(json, false);
}
exports.TweetFavoritersResponseFromJSON = TweetFavoritersResponseFromJSON;
function TweetFavoritersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, TweetFavoritersResponseData_1.TweetFavoritersResponseDataFromJSON)(json['data']),
    };
}
exports.TweetFavoritersResponseFromJSONTyped = TweetFavoritersResponseFromJSONTyped;
function TweetFavoritersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, TweetFavoritersResponseData_1.TweetFavoritersResponseDataToJSON)(value.data),
    };
}
exports.TweetFavoritersResponseToJSON = TweetFavoritersResponseToJSON;
