// twitter-openapi-typescript-generated/dist/models/TweetDetailResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetDetailResponseDataToJSON = exports.TweetDetailResponseDataFromJSONTyped = exports.TweetDetailResponseDataFromJSON = exports.instanceOfTweetDetailResponseData = void 0;
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the TweetDetailResponseData interface.
 */
function instanceOfTweetDetailResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "threadedConversationWithInjectionsV2" in value;
    return isInstance;
}
exports.instanceOfTweetDetailResponseData = instanceOfTweetDetailResponseData;
function TweetDetailResponseDataFromJSON(json) {
    return TweetDetailResponseDataFromJSONTyped(json, false);
}
exports.TweetDetailResponseDataFromJSON = TweetDetailResponseDataFromJSON;
function TweetDetailResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'threadedConversationWithInjectionsV2': (0, Timeline_1.TimelineFromJSON)(json['threaded_conversation_with_injections_v2']),
    };
}
exports.TweetDetailResponseDataFromJSONTyped = TweetDetailResponseDataFromJSONTyped;
function TweetDetailResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'threaded_conversation_with_injections_v2': (0, Timeline_1.TimelineToJSON)(value.threadedConversationWithInjectionsV2),
    };
}
exports.TweetDetailResponseDataToJSON = TweetDetailResponseDataToJSON;
