// twitter-openapi-typescript-generated/dist/models/TweetEditPrespective.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetEditPrespectiveToJSON = exports.TweetEditPrespectiveFromJSONTyped = exports.TweetEditPrespectiveFromJSON = exports.instanceOfTweetEditPrespective = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TweetEditPrespective interface.
 */
function instanceOfTweetEditPrespective(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTweetEditPrespective = instanceOfTweetEditPrespective;
function TweetEditPrespectiveFromJSON(json) {
    return TweetEditPrespectiveFromJSONTyped(json, false);
}
exports.TweetEditPrespectiveFromJSON = TweetEditPrespectiveFromJSON;
function TweetEditPrespectiveFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'favorited': !(0, runtime_1.exists)(json, 'favorited') ? undefined : json['favorited'],
        'retweeted': !(0, runtime_1.exists)(json, 'retweeted') ? undefined : json['retweeted'],
    };
}
exports.TweetEditPrespectiveFromJSONTyped = TweetEditPrespectiveFromJSONTyped;
function TweetEditPrespectiveToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'favorited': value.favorited,
        'retweeted': value.retweeted,
    };
}
exports.TweetEditPrespectiveToJSON = TweetEditPrespectiveToJSON;
