// twitter-openapi-typescript-generated/dist/models/TweetCardLegacy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetCardLegacyToJSON = exports.TweetCardLegacyFromJSONTyped = exports.TweetCardLegacyFromJSON = exports.instanceOfTweetCardLegacy = void 0;
var TweetCardLegacyBindingValue_1 = require("./TweetCardLegacyBindingValue");
/**
 * Check if a given object implements the TweetCardLegacy interface.
 */
function instanceOfTweetCardLegacy(value) {
    var isInstance = true;
    isInstance = isInstance && "bindingValues" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfTweetCardLegacy = instanceOfTweetCardLegacy;
function TweetCardLegacyFromJSON(json) {
    return TweetCardLegacyFromJSONTyped(json, false);
}
exports.TweetCardLegacyFromJSON = TweetCardLegacyFromJSON;
function TweetCardLegacyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'bindingValues': (json['binding_values'].map(TweetCardLegacyBindingValue_1.TweetCardLegacyBindingValueFromJSON)),
        'name': json['name'],
        'url': json['url'],
    };
}
exports.TweetCardLegacyFromJSONTyped = TweetCardLegacyFromJSONTyped;
function TweetCardLegacyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'binding_values': (value.bindingValues.map(TweetCardLegacyBindingValue_1.TweetCardLegacyBindingValueToJSON)),
        'name': value.name,
        'url': value.url,
    };
}
exports.TweetCardLegacyToJSON = TweetCardLegacyToJSON;
