// twitter-openapi-typescript-generated/dist/models/TweetRetweetersResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetRetweetersResponseDataToJSON = exports.TweetRetweetersResponseDataFromJSONTyped = exports.TweetRetweetersResponseDataFromJSON = exports.instanceOfTweetRetweetersResponseData = void 0;
var TimelineV2_1 = require("./TimelineV2");
/**
 * Check if a given object implements the TweetRetweetersResponseData interface.
 */
function instanceOfTweetRetweetersResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "retweetersTimeline" in value;
    return isInstance;
}
exports.instanceOfTweetRetweetersResponseData = instanceOfTweetRetweetersResponseData;
function TweetRetweetersResponseDataFromJSON(json) {
    return TweetRetweetersResponseDataFromJSONTyped(json, false);
}
exports.TweetRetweetersResponseDataFromJSON = TweetRetweetersResponseDataFromJSON;
function TweetRetweetersResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'retweetersTimeline': (0, TimelineV2_1.TimelineV2FromJSON)(json['retweeters_timeline']),
    };
}
exports.TweetRetweetersResponseDataFromJSONTyped = TweetRetweetersResponseDataFromJSONTyped;
function TweetRetweetersResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'retweeters_timeline': (0, TimelineV2_1.TimelineV2ToJSON)(value.retweetersTimeline),
    };
}
exports.TweetRetweetersResponseDataToJSON = TweetRetweetersResponseDataToJSON;
