// twitter-openapi-typescript-generated/dist/models/TweetEditControlInitial.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetEditControlInitialToJSON = exports.TweetEditControlInitialFromJSONTyped = exports.TweetEditControlInitialFromJSON = exports.instanceOfTweetEditControlInitial = void 0;
/**
 * Check if a given object implements the TweetEditControlInitial interface.
 */
function instanceOfTweetEditControlInitial(value) {
    var isInstance = true;
    isInstance = isInstance && "editTweetIds" in value;
    isInstance = isInstance && "editableUntilMsecs" in value;
    isInstance = isInstance && "editsRemaining" in value;
    isInstance = isInstance && "isEditEligible" in value;
    return isInstance;
}
exports.instanceOfTweetEditControlInitial = instanceOfTweetEditControlInitial;
function TweetEditControlInitialFromJSON(json) {
    return TweetEditControlInitialFromJSONTyped(json, false);
}
exports.TweetEditControlInitialFromJSON = TweetEditControlInitialFromJSON;
function TweetEditControlInitialFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'editTweetIds': json['edit_tweet_ids'],
        'editableUntilMsecs': json['editable_until_msecs'],
        'editsRemaining': json['edits_remaining'],
        'isEditEligible': json['is_edit_eligible'],
    };
}
exports.TweetEditControlInitialFromJSONTyped = TweetEditControlInitialFromJSONTyped;
function TweetEditControlInitialToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'edit_tweet_ids': value.editTweetIds,
        'editable_until_msecs': value.editableUntilMsecs,
        'edits_remaining': value.editsRemaining,
        'is_edit_eligible': value.isEditEligible,
    };
}
exports.TweetEditControlInitialToJSON = TweetEditControlInitialToJSON;
