// twitter-openapi-typescript-generated/dist/models/TweetRetweetersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetRetweetersResponseToJSON = exports.TweetRetweetersResponseFromJSONTyped = exports.TweetRetweetersResponseFromJSON = exports.instanceOfTweetRetweetersResponse = void 0;
var TweetRetweetersResponseData_1 = require("./TweetRetweetersResponseData");
/**
 * Check if a given object implements the TweetRetweetersResponse interface.
 */
function instanceOfTweetRetweetersResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfTweetRetweetersResponse = instanceOfTweetRetweetersResponse;
function TweetRetweetersResponseFromJSON(json) {
    return TweetRetweetersResponseFromJSONTyped(json, false);
}
exports.TweetRetweetersResponseFromJSON = TweetRetweetersResponseFromJSON;
function TweetRetweetersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, TweetRetweetersResponseData_1.TweetRetweetersResponseDataFromJSON)(json['data']),
    };
}
exports.TweetRetweetersResponseFromJSONTyped = TweetRetweetersResponseFromJSONTyped;
function TweetRetweetersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, TweetRetweetersResponseData_1.TweetRetweetersResponseDataToJSON)(value.data),
    };
}
exports.TweetRetweetersResponseToJSON = TweetRetweetersResponseToJSON;
