// twitter-openapi-typescript-generated/dist/models/TweetCard.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetCardToJSON = exports.TweetCardFromJSONTyped = exports.TweetCardFromJSON = exports.instanceOfTweetCard = void 0;
var runtime_1 = require("../runtime");
var TweetCardLegacy_1 = require("./TweetCardLegacy");
/**
 * Check if a given object implements the TweetCard interface.
 */
function instanceOfTweetCard(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTweetCard = instanceOfTweetCard;
function TweetCardFromJSON(json) {
    return TweetCardFromJSONTyped(json, false);
}
exports.TweetCardFromJSON = TweetCardFromJSON;
function TweetCardFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'legacy': !(0, runtime_1.exists)(json, 'legacy') ? undefined : (0, TweetCardLegacy_1.TweetCardLegacyFromJSON)(json['legacy']),
        'restId': !(0, runtime_1.exists)(json, 'rest_id') ? undefined : json['rest_id'],
    };
}
exports.TweetCardFromJSONTyped = TweetCardFromJSONTyped;
function TweetCardToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'legacy': (0, TweetCardLegacy_1.TweetCardLegacyToJSON)(value.legacy),
        'rest_id': value.restId,
    };
}
exports.TweetCardToJSON = TweetCardToJSON;
