// twitter-openapi-typescript-generated/dist/models/TweetWithVisibilityResults.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetWithVisibilityResultsToJSON = exports.TweetWithVisibilityResultsFromJSONTyped = exports.TweetWithVisibilityResultsFromJSON = exports.instanceOfTweetWithVisibilityResults = void 0;
var Tweet_1 = require("./Tweet");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the TweetWithVisibilityResults interface.
 */
function instanceOfTweetWithVisibilityResults(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "tweet" in value;
    return isInstance;
}
exports.instanceOfTweetWithVisibilityResults = instanceOfTweetWithVisibilityResults;
function TweetWithVisibilityResultsFromJSON(json) {
    return TweetWithVisibilityResultsFromJSONTyped(json, false);
}
exports.TweetWithVisibilityResultsFromJSON = TweetWithVisibilityResultsFromJSON;
function TweetWithVisibilityResultsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'tweet': (0, Tweet_1.TweetFromJSON)(json['tweet']),
    };
}
exports.TweetWithVisibilityResultsFromJSONTyped = TweetWithVisibilityResultsFromJSONTyped;
function TweetWithVisibilityResultsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'tweet': (0, Tweet_1.TweetToJSON)(value.tweet),
    };
}
exports.TweetWithVisibilityResultsToJSON = TweetWithVisibilityResultsToJSON;
