// twitter-openapi-typescript-generated/dist/models/TweetFavoritersResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetFavoritersResponseDataToJSON = exports.TweetFavoritersResponseDataFromJSONTyped = exports.TweetFavoritersResponseDataFromJSON = exports.instanceOfTweetFavoritersResponseData = void 0;
var TimelineV2_1 = require("./TimelineV2");
/**
 * Check if a given object implements the TweetFavoritersResponseData interface.
 */
function instanceOfTweetFavoritersResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "favoritersTimeline" in value;
    return isInstance;
}
exports.instanceOfTweetFavoritersResponseData = instanceOfTweetFavoritersResponseData;
function TweetFavoritersResponseDataFromJSON(json) {
    return TweetFavoritersResponseDataFromJSONTyped(json, false);
}
exports.TweetFavoritersResponseDataFromJSON = TweetFavoritersResponseDataFromJSON;
function TweetFavoritersResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'favoritersTimeline': (0, TimelineV2_1.TimelineV2FromJSON)(json['favoriters_timeline']),
    };
}
exports.TweetFavoritersResponseDataFromJSONTyped = TweetFavoritersResponseDataFromJSONTyped;
function TweetFavoritersResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'favoriters_timeline': (0, TimelineV2_1.TimelineV2ToJSON)(value.favoritersTimeline),
    };
}
exports.TweetFavoritersResponseDataToJSON = TweetFavoritersResponseDataToJSON;
