// twitter-openapi-typescript-generated/dist/models/TweetView.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetViewToJSON = exports.TweetViewFromJSONTyped = exports.TweetViewFromJSON = exports.instanceOfTweetView = exports.TweetViewStateEnum = void 0;
var runtime_1 = require("../runtime");
/**
 * @export
 */
exports.TweetViewStateEnum = {
    EnabledWithCount: 'EnabledWithCount'
};
/**
 * Check if a given object implements the TweetView interface.
 */
function instanceOfTweetView(value) {
    var isInstance = true;
    isInstance = isInstance && "state" in value;
    return isInstance;
}
exports.instanceOfTweetView = instanceOfTweetView;
function TweetViewFromJSON(json) {
    return TweetViewFromJSONTyped(json, false);
}
exports.TweetViewFromJSON = TweetViewFromJSON;
function TweetViewFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'count': !(0, runtime_1.exists)(json, 'count') ? undefined : json['count'],
        'state': json['state'],
    };
}
exports.TweetViewFromJSONTyped = TweetViewFromJSONTyped;
function TweetViewToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'count': value.count,
        'state': value.state,
    };
}
exports.TweetViewToJSON = TweetViewToJSON;
