// twitter-openapi-typescript-generated/dist/models/TweetCardLegacyBindingValueData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetCardLegacyBindingValueDataToJSON = exports.TweetCardLegacyBindingValueDataFromJSONTyped = exports.TweetCardLegacyBindingValueDataFromJSON = exports.instanceOfTweetCardLegacyBindingValueData = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TweetCardLegacyBindingValueData interface.
 */
function instanceOfTweetCardLegacyBindingValueData(value) {
    var isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfTweetCardLegacyBindingValueData = instanceOfTweetCardLegacyBindingValueData;
function TweetCardLegacyBindingValueDataFromJSON(json) {
    return TweetCardLegacyBindingValueDataFromJSONTyped(json, false);
}
exports.TweetCardLegacyBindingValueDataFromJSON = TweetCardLegacyBindingValueDataFromJSON;
function TweetCardLegacyBindingValueDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'booleanValue': !(0, runtime_1.exists)(json, 'boolean_value') ? undefined : json['boolean_value'],
        'scribeKey': !(0, runtime_1.exists)(json, 'scribe_key') ? undefined : json['scribe_key'],
        'stringValue': !(0, runtime_1.exists)(json, 'string_value') ? undefined : json['string_value'],
        'type': json['type'],
    };
}
exports.TweetCardLegacyBindingValueDataFromJSONTyped = TweetCardLegacyBindingValueDataFromJSONTyped;
function TweetCardLegacyBindingValueDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'boolean_value': value.booleanValue,
        'scribe_key': value.scribeKey,
        'string_value': value.stringValue,
        'type': value.type,
    };
}
exports.TweetCardLegacyBindingValueDataToJSON = TweetCardLegacyBindingValueDataToJSON;
