// twitter-openapi-typescript-generated/dist/models/TweetCardLegacyBindingValue.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetCardLegacyBindingValueToJSON = exports.TweetCardLegacyBindingValueFromJSONTyped = exports.TweetCardLegacyBindingValueFromJSON = exports.instanceOfTweetCardLegacyBindingValue = void 0;
var TweetCardLegacyBindingValueData_1 = require("./TweetCardLegacyBindingValueData");
/**
 * Check if a given object implements the TweetCardLegacyBindingValue interface.
 */
function instanceOfTweetCardLegacyBindingValue(value) {
    var isInstance = true;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
exports.instanceOfTweetCardLegacyBindingValue = instanceOfTweetCardLegacyBindingValue;
function TweetCardLegacyBindingValueFromJSON(json) {
    return TweetCardLegacyBindingValueFromJSONTyped(json, false);
}
exports.TweetCardLegacyBindingValueFromJSON = TweetCardLegacyBindingValueFromJSON;
function TweetCardLegacyBindingValueFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': json['key'],
        'value': (0, TweetCardLegacyBindingValueData_1.TweetCardLegacyBindingValueDataFromJSON)(json['value']),
    };
}
exports.TweetCardLegacyBindingValueFromJSONTyped = TweetCardLegacyBindingValueFromJSONTyped;
function TweetCardLegacyBindingValueToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
        'value': (0, TweetCardLegacyBindingValueData_1.TweetCardLegacyBindingValueDataToJSON)(value.value),
    };
}
exports.TweetCardLegacyBindingValueToJSON = TweetCardLegacyBindingValueToJSON;
