// twitter-openapi-typescript-generated/dist/models/TweetDetailResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetDetailResponseToJSON = exports.TweetDetailResponseFromJSONTyped = exports.TweetDetailResponseFromJSON = exports.instanceOfTweetDetailResponse = void 0;
var TweetDetailResponseData_1 = require("./TweetDetailResponseData");
/**
 * Check if a given object implements the TweetDetailResponse interface.
 */
function instanceOfTweetDetailResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfTweetDetailResponse = instanceOfTweetDetailResponse;
function TweetDetailResponseFromJSON(json) {
    return TweetDetailResponseFromJSONTyped(json, false);
}
exports.TweetDetailResponseFromJSON = TweetDetailResponseFromJSON;
function TweetDetailResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, TweetDetailResponseData_1.TweetDetailResponseDataFromJSON)(json['data']),
    };
}
exports.TweetDetailResponseFromJSONTyped = TweetDetailResponseFromJSONTyped;
function TweetDetailResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, TweetDetailResponseData_1.TweetDetailResponseDataToJSON)(value.data),
    };
}
exports.TweetDetailResponseToJSON = TweetDetailResponseToJSON;
