// twitter-openapi-typescript-generated/dist/models/TweetResultByRestIdData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetResultByRestIdDataToJSON = exports.TweetResultByRestIdDataFromJSONTyped = exports.TweetResultByRestIdDataFromJSON = exports.instanceOfTweetResultByRestIdData = void 0;
var ItemResult_1 = require("./ItemResult");
/**
 * Check if a given object implements the TweetResultByRestIdData interface.
 */
function instanceOfTweetResultByRestIdData(value) {
    var isInstance = true;
    isInstance = isInstance && "tweetResult" in value;
    return isInstance;
}
exports.instanceOfTweetResultByRestIdData = instanceOfTweetResultByRestIdData;
function TweetResultByRestIdDataFromJSON(json) {
    return TweetResultByRestIdDataFromJSONTyped(json, false);
}
exports.TweetResultByRestIdDataFromJSON = TweetResultByRestIdDataFromJSON;
function TweetResultByRestIdDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'tweetResult': (0, ItemResult_1.ItemResultFromJSON)(json['tweetResult']),
    };
}
exports.TweetResultByRestIdDataFromJSONTyped = TweetResultByRestIdDataFromJSONTyped;
function TweetResultByRestIdDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'tweetResult': (0, ItemResult_1.ItemResultToJSON)(value.tweetResult),
    };
}
exports.TweetResultByRestIdDataToJSON = TweetResultByRestIdDataToJSON;
