// twitter-openapi-typescript-generated/dist/models/TweetResultByRestIdResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetResultByRestIdResponseToJSON = exports.TweetResultByRestIdResponseFromJSONTyped = exports.TweetResultByRestIdResponseFromJSON = exports.instanceOfTweetResultByRestIdResponse = void 0;
var TweetResultByRestIdData_1 = require("./TweetResultByRestIdData");
/**
 * Check if a given object implements the TweetResultByRestIdResponse interface.
 */
function instanceOfTweetResultByRestIdResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfTweetResultByRestIdResponse = instanceOfTweetResultByRestIdResponse;
function TweetResultByRestIdResponseFromJSON(json) {
    return TweetResultByRestIdResponseFromJSONTyped(json, false);
}
exports.TweetResultByRestIdResponseFromJSON = TweetResultByRestIdResponseFromJSON;
function TweetResultByRestIdResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, TweetResultByRestIdData_1.TweetResultByRestIdDataFromJSON)(json['data']),
    };
}
exports.TweetResultByRestIdResponseFromJSONTyped = TweetResultByRestIdResponseFromJSONTyped;
function TweetResultByRestIdResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, TweetResultByRestIdData_1.TweetResultByRestIdDataToJSON)(value.data),
    };
}
exports.TweetResultByRestIdResponseToJSON = TweetResultByRestIdResponseToJSON;
