// twitter-openapi-typescript-generated/dist/models/TweetEditControl.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TweetEditControlToJSON = exports.TweetEditControlFromJSONTyped = exports.TweetEditControlFromJSON = exports.instanceOfTweetEditControl = void 0;
var runtime_1 = require("../runtime");
var TweetEditControlInitial_1 = require("./TweetEditControlInitial");
/**
 * Check if a given object implements the TweetEditControl interface.
 */
function instanceOfTweetEditControl(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTweetEditControl = instanceOfTweetEditControl;
function TweetEditControlFromJSON(json) {
    return TweetEditControlFromJSONTyped(json, false);
}
exports.TweetEditControlFromJSON = TweetEditControlFromJSON;
function TweetEditControlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'editControlInitial': !(0, runtime_1.exists)(json, 'edit_control_initial') ? undefined : (0, TweetEditControlInitial_1.TweetEditControlInitialFromJSON)(json['edit_control_initial']),
        'editTweetIds': !(0, runtime_1.exists)(json, 'edit_tweet_ids') ? undefined : json['edit_tweet_ids'],
        'editableUntilMsecs': !(0, runtime_1.exists)(json, 'editable_until_msecs') ? undefined : json['editable_until_msecs'],
        'editsRemaining': !(0, runtime_1.exists)(json, 'edits_remaining') ? undefined : json['edits_remaining'],
        'initialTweetId': !(0, runtime_1.exists)(json, 'initial_tweet_id') ? undefined : json['initial_tweet_id'],
        'isEditEligible': !(0, runtime_1.exists)(json, 'is_edit_eligible') ? undefined : json['is_edit_eligible'],
    };
}
exports.TweetEditControlFromJSONTyped = TweetEditControlFromJSONTyped;
function TweetEditControlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'edit_control_initial': (0, TweetEditControlInitial_1.TweetEditControlInitialToJSON)(value.editControlInitial),
        'edit_tweet_ids': value.editTweetIds,
        'editable_until_msecs': value.editableUntilMsecs,
        'edits_remaining': value.editsRemaining,
        'initial_tweet_id': value.initialTweetId,
        'is_edit_eligible': value.isEditEligible,
    };
}
exports.TweetEditControlToJSON = TweetEditControlToJSON;
