// twitter-openapi-typescript-generated/dist/models/TypeName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypeNameToJSON = exports.TypeNameFromJSONTyped = exports.TypeNameFromJSON = exports.TypeName = void 0;
/**
 *
 * @export
 */
exports.TypeName = {
    TimelineTweet: 'TimelineTweet',
    TimelineTimelineItem: 'TimelineTimelineItem',
    TimelineUser: 'TimelineUser',
    TimelineTimelineCursor: 'TimelineTimelineCursor',
    TweetWithVisibilityResults: 'TweetWithVisibilityResults',
    TimelineTimelineModule: 'TimelineTimelineModule',
    TweetTombstone: 'TweetTombstone',
    TimelinePrompt: 'TimelinePrompt',
    TimelineMessagePrompt: 'TimelineMessagePrompt',
    TimelineCommunity: 'TimelineCommunity',
    TweetUnavailable: 'TweetUnavailable',
    Tweet: 'Tweet',
    User: 'User',
    UserUnavailable: 'UserUnavailable'
};
function TypeNameFromJSON(json) {
    return TypeNameFromJSONTyped(json, false);
}
exports.TypeNameFromJSON = TypeNameFromJSON;
function TypeNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.TypeNameFromJSONTyped = TypeNameFromJSONTyped;
function TypeNameToJSON(value) {
    return value;
}
exports.TypeNameToJSON = TypeNameToJSON;
