// twitter-openapi-typescript-generated/dist/models/UnfavoriteTweet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnfavoriteTweetToJSON = exports.UnfavoriteTweetFromJSONTyped = exports.UnfavoriteTweetFromJSON = exports.instanceOfUnfavoriteTweet = void 0;
/**
 * Check if a given object implements the UnfavoriteTweet interface.
 */
function instanceOfUnfavoriteTweet(value) {
    var isInstance = true;
    isInstance = isInstance && "unfavoriteTweet" in value;
    return isInstance;
}
exports.instanceOfUnfavoriteTweet = instanceOfUnfavoriteTweet;
function UnfavoriteTweetFromJSON(json) {
    return UnfavoriteTweetFromJSONTyped(json, false);
}
exports.UnfavoriteTweetFromJSON = UnfavoriteTweetFromJSON;
function UnfavoriteTweetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'unfavoriteTweet': json['unfavorite_tweet'],
    };
}
exports.UnfavoriteTweetFromJSONTyped = UnfavoriteTweetFromJSONTyped;
function UnfavoriteTweetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'unfavorite_tweet': value.unfavoriteTweet,
    };
}
exports.UnfavoriteTweetToJSON = UnfavoriteTweetToJSON;
