// twitter-openapi-typescript-generated/dist/models/UnfavoriteTweetResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnfavoriteTweetResponseDataToJSON = exports.UnfavoriteTweetResponseDataFromJSONTyped = exports.UnfavoriteTweetResponseDataFromJSON = exports.instanceOfUnfavoriteTweetResponseData = void 0;
var UnfavoriteTweet_1 = require("./UnfavoriteTweet");
/**
 * Check if a given object implements the UnfavoriteTweetResponseData interface.
 */
function instanceOfUnfavoriteTweetResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfUnfavoriteTweetResponseData = instanceOfUnfavoriteTweetResponseData;
function UnfavoriteTweetResponseDataFromJSON(json) {
    return UnfavoriteTweetResponseDataFromJSONTyped(json, false);
}
exports.UnfavoriteTweetResponseDataFromJSON = UnfavoriteTweetResponseDataFromJSON;
function UnfavoriteTweetResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, UnfavoriteTweet_1.UnfavoriteTweetFromJSON)(json['data']),
    };
}
exports.UnfavoriteTweetResponseDataFromJSONTyped = UnfavoriteTweetResponseDataFromJSONTyped;
function UnfavoriteTweetResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, UnfavoriteTweet_1.UnfavoriteTweetToJSON)(value.data),
    };
}
exports.UnfavoriteTweetResponseDataToJSON = UnfavoriteTweetResponseDataToJSON;
