// twitter-openapi-typescript-generated/dist/models/Url.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlToJSON = exports.UrlFromJSONTyped = exports.UrlFromJSON = exports.instanceOfUrl = void 0;
/**
 * Check if a given object implements the Url interface.
 */
function instanceOfUrl(value) {
    var isInstance = true;
    isInstance = isInstance && "displayUrl" in value;
    isInstance = isInstance && "expandedUrl" in value;
    isInstance = isInstance && "indices" in value;
    isInstance = isInstance && "url" in value;
    return isInstance;
}
exports.instanceOfUrl = instanceOfUrl;
function UrlFromJSON(json) {
    return UrlFromJSONTyped(json, false);
}
exports.UrlFromJSON = UrlFromJSON;
function UrlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'displayUrl': json['display_url'],
        'expandedUrl': json['expanded_url'],
        'indices': json['indices'],
        'url': json['url'],
    };
}
exports.UrlFromJSONTyped = UrlFromJSONTyped;
function UrlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'display_url': value.displayUrl,
        'expanded_url': value.expandedUrl,
        'indices': value.indices,
        'url': value.url,
    };
}
exports.UrlToJSON = UrlToJSON;
