// twitter-openapi-typescript-generated/dist/models/UserResultCore.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResultCoreToJSON = exports.UserResultCoreFromJSONTyped = exports.UserResultCoreFromJSON = exports.instanceOfUserResultCore = void 0;
var UserResults_1 = require("./UserResults");
/**
 * Check if a given object implements the UserResultCore interface.
 */
function instanceOfUserResultCore(value) {
    var isInstance = true;
    isInstance = isInstance && "userResults" in value;
    return isInstance;
}
exports.instanceOfUserResultCore = instanceOfUserResultCore;
function UserResultCoreFromJSON(json) {
    return UserResultCoreFromJSONTyped(json, false);
}
exports.UserResultCoreFromJSON = UserResultCoreFromJSON;
function UserResultCoreFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'userResults': (0, UserResults_1.UserResultsFromJSON)(json['user_results']),
    };
}
exports.UserResultCoreFromJSONTyped = UserResultCoreFromJSONTyped;
function UserResultCoreToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user_results': (0, UserResults_1.UserResultsToJSON)(value.userResults),
    };
}
exports.UserResultCoreToJSON = UserResultCoreToJSON;
