// twitter-openapi-typescript-generated/dist/models/UserTweetsData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTweetsDataToJSON = exports.UserTweetsDataFromJSONTyped = exports.UserTweetsDataFromJSON = exports.instanceOfUserTweetsData = void 0;
var UserTweetsUser_1 = require("./UserTweetsUser");
/**
 * Check if a given object implements the UserTweetsData interface.
 */
function instanceOfUserTweetsData(value) {
    var isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfUserTweetsData = instanceOfUserTweetsData;
function UserTweetsDataFromJSON(json) {
    return UserTweetsDataFromJSONTyped(json, false);
}
exports.UserTweetsDataFromJSON = UserTweetsDataFromJSON;
function UserTweetsDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, UserTweetsUser_1.UserTweetsUserFromJSON)(json['user']),
    };
}
exports.UserTweetsDataFromJSONTyped = UserTweetsDataFromJSONTyped;
function UserTweetsDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, UserTweetsUser_1.UserTweetsUserToJSON)(value.user),
    };
}
exports.UserTweetsDataToJSON = UserTweetsDataToJSON;
