// twitter-openapi-typescript-generated/dist/models/UserResultByScreenName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResultByScreenNameToJSON = exports.UserResultByScreenNameFromJSONTyped = exports.UserResultByScreenNameFromJSON = exports.instanceOfUserResultByScreenName = void 0;
var UserResultByScreenNameResult_1 = require("./UserResultByScreenNameResult");
/**
 * Check if a given object implements the UserResultByScreenName interface.
 */
function instanceOfUserResultByScreenName(value) {
    var isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfUserResultByScreenName = instanceOfUserResultByScreenName;
function UserResultByScreenNameFromJSON(json) {
    return UserResultByScreenNameFromJSONTyped(json, false);
}
exports.UserResultByScreenNameFromJSON = UserResultByScreenNameFromJSON;
function UserResultByScreenNameFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'result': (0, UserResultByScreenNameResult_1.UserResultByScreenNameResultFromJSON)(json['result']),
    };
}
exports.UserResultByScreenNameFromJSONTyped = UserResultByScreenNameFromJSONTyped;
function UserResultByScreenNameToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'result': (0, UserResultByScreenNameResult_1.UserResultByScreenNameResultToJSON)(value.result),
    };
}
exports.UserResultByScreenNameToJSON = UserResultByScreenNameToJSON;
