// twitter-openapi-typescript-generated/dist/models/UserLegacy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLegacyToJSON = exports.UserLegacyFromJSONTyped = exports.UserLegacyFromJSON = exports.instanceOfUserLegacy = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserLegacy interface.
 */
function instanceOfUserLegacy(value) {
    var isInstance = true;
    isInstance = isInstance && "blockedBy" in value;
    isInstance = isInstance && "blocking" in value;
    isInstance = isInstance && "canDm" in value;
    isInstance = isInstance && "canMediaTag" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "defaultProfile" in value;
    isInstance = isInstance && "defaultProfileImage" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "entities" in value;
    isInstance = isInstance && "fastFollowersCount" in value;
    isInstance = isInstance && "favouritesCount" in value;
    isInstance = isInstance && "followersCount" in value;
    isInstance = isInstance && "friendsCount" in value;
    isInstance = isInstance && "hasCustomTimelines" in value;
    isInstance = isInstance && "isTranslator" in value;
    isInstance = isInstance && "listedCount" in value;
    isInstance = isInstance && "location" in value;
    isInstance = isInstance && "mediaCount" in value;
    isInstance = isInstance && "muting" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "normalFollowersCount" in value;
    isInstance = isInstance && "pinnedTweetIdsStr" in value;
    isInstance = isInstance && "possiblySensitive" in value;
    isInstance = isInstance && "profileImageUrlHttps" in value;
    isInstance = isInstance && "profileInterstitialType" in value;
    isInstance = isInstance && "screenName" in value;
    isInstance = isInstance && "statusesCount" in value;
    isInstance = isInstance && "translatorType" in value;
    isInstance = isInstance && "verified" in value;
    isInstance = isInstance && "wantRetweets" in value;
    return isInstance;
}
exports.instanceOfUserLegacy = instanceOfUserLegacy;
function UserLegacyFromJSON(json) {
    return UserLegacyFromJSONTyped(json, false);
}
exports.UserLegacyFromJSON = UserLegacyFromJSON;
function UserLegacyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'blockedBy': json['blocked_by'],
        'blocking': json['blocking'],
        'canDm': json['can_dm'],
        'canMediaTag': json['can_media_tag'],
        'createdAt': json['created_at'],
        'defaultProfile': json['default_profile'],
        'defaultProfileImage': json['default_profile_image'],
        'description': json['description'],
        'entities': json['entities'],
        'fastFollowersCount': json['fast_followers_count'],
        'favouritesCount': json['favourites_count'],
        'followRequestSent': !(0, runtime_1.exists)(json, 'follow_request_sent') ? undefined : json['follow_request_sent'],
        'followedBy': !(0, runtime_1.exists)(json, 'followed_by') ? undefined : json['followed_by'],
        'followersCount': json['followers_count'],
        'following': !(0, runtime_1.exists)(json, 'following') ? undefined : json['following'],
        'friendsCount': json['friends_count'],
        'hasCustomTimelines': json['has_custom_timelines'],
        'isTranslator': json['is_translator'],
        'listedCount': json['listed_count'],
        'location': json['location'],
        'mediaCount': json['media_count'],
        'muting': json['muting'],
        'name': json['name'],
        'normalFollowersCount': json['normal_followers_count'],
        'notifications': !(0, runtime_1.exists)(json, 'notifications') ? undefined : json['notifications'],
        'pinnedTweetIdsStr': json['pinned_tweet_ids_str'],
        'possiblySensitive': json['possibly_sensitive'],
        'profileBannerExtensions': !(0, runtime_1.exists)(json, 'profile_banner_extensions') ? undefined : json['profile_banner_extensions'],
        'profileBannerUrl': !(0, runtime_1.exists)(json, 'profile_banner_url') ? undefined : json['profile_banner_url'],
        'profileImageExtensions': !(0, runtime_1.exists)(json, 'profile_image_extensions') ? undefined : json['profile_image_extensions'],
        'profileImageUrlHttps': json['profile_image_url_https'],
        'profileInterstitialType': json['profile_interstitial_type'],
        '_protected': !(0, runtime_1.exists)(json, 'protected') ? undefined : json['protected'],
        'screenName': json['screen_name'],
        'statusesCount': json['statuses_count'],
        'translatorType': json['translator_type'],
        'url': !(0, runtime_1.exists)(json, 'url') ? undefined : json['url'],
        'verified': json['verified'],
        'wantRetweets': json['want_retweets'],
    };
}
exports.UserLegacyFromJSONTyped = UserLegacyFromJSONTyped;
function UserLegacyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'blocked_by': value.blockedBy,
        'blocking': value.blocking,
        'can_dm': value.canDm,
        'can_media_tag': value.canMediaTag,
        'created_at': value.createdAt,
        'default_profile': value.defaultProfile,
        'default_profile_image': value.defaultProfileImage,
        'description': value.description,
        'entities': value.entities,
        'fast_followers_count': value.fastFollowersCount,
        'favourites_count': value.favouritesCount,
        'follow_request_sent': value.followRequestSent,
        'followed_by': value.followedBy,
        'followers_count': value.followersCount,
        'following': value.following,
        'friends_count': value.friendsCount,
        'has_custom_timelines': value.hasCustomTimelines,
        'is_translator': value.isTranslator,
        'listed_count': value.listedCount,
        'location': value.location,
        'media_count': value.mediaCount,
        'muting': value.muting,
        'name': value.name,
        'normal_followers_count': value.normalFollowersCount,
        'notifications': value.notifications,
        'pinned_tweet_ids_str': value.pinnedTweetIdsStr,
        'possibly_sensitive': value.possiblySensitive,
        'profile_banner_extensions': value.profileBannerExtensions,
        'profile_banner_url': value.profileBannerUrl,
        'profile_image_extensions': value.profileImageExtensions,
        'profile_image_url_https': value.profileImageUrlHttps,
        'profile_interstitial_type': value.profileInterstitialType,
        'protected': value._protected,
        'screen_name': value.screenName,
        'statuses_count': value.statusesCount,
        'translator_type': value.translatorType,
        'url': value.url,
        'verified': value.verified,
        'want_retweets': value.wantRetweets,
    };
}
exports.UserLegacyToJSON = UserLegacyToJSON;
