// twitter-openapi-typescript-generated/dist/models/UserHighlightsTweetsUser.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHighlightsTweetsUserToJSON = exports.UserHighlightsTweetsUserFromJSONTyped = exports.UserHighlightsTweetsUserFromJSON = exports.instanceOfUserHighlightsTweetsUser = void 0;
var UserHighlightsTweetsResult_1 = require("./UserHighlightsTweetsResult");
/**
 * Check if a given object implements the UserHighlightsTweetsUser interface.
 */
function instanceOfUserHighlightsTweetsUser(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfUserHighlightsTweetsUser = instanceOfUserHighlightsTweetsUser;
function UserHighlightsTweetsUserFromJSON(json) {
    return UserHighlightsTweetsUserFromJSONTyped(json, false);
}
exports.UserHighlightsTweetsUserFromJSON = UserHighlightsTweetsUserFromJSON;
function UserHighlightsTweetsUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, UserHighlightsTweetsResult_1.UserHighlightsTweetsResultFromJSON)(json['result']),
    };
}
exports.UserHighlightsTweetsUserFromJSONTyped = UserHighlightsTweetsUserFromJSONTyped;
function UserHighlightsTweetsUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, UserHighlightsTweetsResult_1.UserHighlightsTweetsResultToJSON)(value.result),
    };
}
exports.UserHighlightsTweetsUserToJSON = UserHighlightsTweetsUserToJSON;
