// twitter-openapi-typescript-generated/dist/models/UserResultByScreenNameLegacy.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResultByScreenNameLegacyToJSON = exports.UserResultByScreenNameLegacyFromJSONTyped = exports.UserResultByScreenNameLegacyFromJSON = exports.instanceOfUserResultByScreenNameLegacy = void 0;
var runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UserResultByScreenNameLegacy interface.
 */
function instanceOfUserResultByScreenNameLegacy(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserResultByScreenNameLegacy = instanceOfUserResultByScreenNameLegacy;
function UserResultByScreenNameLegacyFromJSON(json) {
    return UserResultByScreenNameLegacyFromJSONTyped(json, false);
}
exports.UserResultByScreenNameLegacyFromJSON = UserResultByScreenNameLegacyFromJSON;
function UserResultByScreenNameLegacyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'blockedBy': !(0, runtime_1.exists)(json, 'blocked_by') ? undefined : json['blocked_by'],
        'blocking': !(0, runtime_1.exists)(json, 'blocking') ? undefined : json['blocking'],
        'followedBy': !(0, runtime_1.exists)(json, 'followed_by') ? undefined : json['followed_by'],
        'following': !(0, runtime_1.exists)(json, 'following') ? undefined : json['following'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        '_protected': !(0, runtime_1.exists)(json, 'protected') ? undefined : json['protected'],
        'screenName': !(0, runtime_1.exists)(json, 'screen_name') ? undefined : json['screen_name'],
    };
}
exports.UserResultByScreenNameLegacyFromJSONTyped = UserResultByScreenNameLegacyFromJSONTyped;
function UserResultByScreenNameLegacyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'blocked_by': value.blockedBy,
        'blocking': value.blocking,
        'followed_by': value.followedBy,
        'following': value.following,
        'name': value.name,
        'protected': value._protected,
        'screen_name': value.screenName,
    };
}
exports.UserResultByScreenNameLegacyToJSON = UserResultByScreenNameLegacyToJSON;
