// twitter-openapi-typescript-generated/dist/models/UserFeatures.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserFeaturesToJSON = exports.UserFeaturesFromJSONTyped = exports.UserFeaturesFromJSON = exports.instanceOfUserFeatures = void 0;
/**
 * Check if a given object implements the UserFeatures interface.
 */
function instanceOfUserFeatures(value) {
    var isInstance = true;
    isInstance = isInstance && "mediatoolStudioLibrary" in value;
    return isInstance;
}
exports.instanceOfUserFeatures = instanceOfUserFeatures;
function UserFeaturesFromJSON(json) {
    return UserFeaturesFromJSONTyped(json, false);
}
exports.UserFeaturesFromJSON = UserFeaturesFromJSON;
function UserFeaturesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'mediatoolStudioLibrary': json['mediatool_studio_library'],
    };
}
exports.UserFeaturesFromJSONTyped = UserFeaturesFromJSONTyped;
function UserFeaturesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'mediatool_studio_library': value.mediatoolStudioLibrary,
    };
}
exports.UserFeaturesToJSON = UserFeaturesToJSON;
