// twitter-openapi-typescript-generated/dist/models/UserTweetsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTweetsResponseToJSON = exports.UserTweetsResponseFromJSONTyped = exports.UserTweetsResponseFromJSON = exports.instanceOfUserTweetsResponse = void 0;
var UserTweetsData_1 = require("./UserTweetsData");
/**
 * Check if a given object implements the UserTweetsResponse interface.
 */
function instanceOfUserTweetsResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfUserTweetsResponse = instanceOfUserTweetsResponse;
function UserTweetsResponseFromJSON(json) {
    return UserTweetsResponseFromJSONTyped(json, false);
}
exports.UserTweetsResponseFromJSON = UserTweetsResponseFromJSON;
function UserTweetsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, UserTweetsData_1.UserTweetsDataFromJSON)(json['data']),
    };
}
exports.UserTweetsResponseFromJSONTyped = UserTweetsResponseFromJSONTyped;
function UserTweetsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, UserTweetsData_1.UserTweetsDataToJSON)(value.data),
    };
}
exports.UserTweetsResponseToJSON = UserTweetsResponseToJSON;
