// twitter-openapi-typescript-generated/dist/models/UserHighlightsTweetsData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHighlightsTweetsDataToJSON = exports.UserHighlightsTweetsDataFromJSONTyped = exports.UserHighlightsTweetsDataFromJSON = exports.instanceOfUserHighlightsTweetsData = void 0;
var UserHighlightsTweetsUser_1 = require("./UserHighlightsTweetsUser");
/**
 * Check if a given object implements the UserHighlightsTweetsData interface.
 */
function instanceOfUserHighlightsTweetsData(value) {
    var isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfUserHighlightsTweetsData = instanceOfUserHighlightsTweetsData;
function UserHighlightsTweetsDataFromJSON(json) {
    return UserHighlightsTweetsDataFromJSONTyped(json, false);
}
exports.UserHighlightsTweetsDataFromJSON = UserHighlightsTweetsDataFromJSON;
function UserHighlightsTweetsDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, UserHighlightsTweetsUser_1.UserHighlightsTweetsUserFromJSON)(json['user']),
    };
}
exports.UserHighlightsTweetsDataFromJSONTyped = UserHighlightsTweetsDataFromJSONTyped;
function UserHighlightsTweetsDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, UserHighlightsTweetsUser_1.UserHighlightsTweetsUserToJSON)(value.user),
    };
}
exports.UserHighlightsTweetsDataToJSON = UserHighlightsTweetsDataToJSON;
