// twitter-openapi-typescript-generated/dist/models/User.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserToJSON = exports.UserFromJSONTyped = exports.UserFromJSON = exports.instanceOfUser = void 0;
var runtime_1 = require("../runtime");
var TypeName_1 = require("./TypeName");
var UserLegacy_1 = require("./UserLegacy");
/**
 * Check if a given object implements the User interface.
 */
function instanceOfUser(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "affiliatesHighlightedLabel" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "isBlueVerified" in value;
    isInstance = isInstance && "legacy" in value;
    isInstance = isInstance && "restId" in value;
    isInstance = isInstance && "superFollowEligible" in value;
    isInstance = isInstance && "superFollowedBy" in value;
    isInstance = isInstance && "superFollowing" in value;
    return isInstance;
}
exports.instanceOfUser = instanceOfUser;
function UserFromJSON(json) {
    return UserFromJSONTyped(json, false);
}
exports.UserFromJSON = UserFromJSON;
function UserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'affiliatesHighlightedLabel': json['affiliates_highlighted_label'],
        'businessAccount': !(0, runtime_1.exists)(json, 'business_account') ? undefined : json['business_account'],
        'hasGraduatedAccess': !(0, runtime_1.exists)(json, 'has_graduated_access') ? undefined : json['has_graduated_access'],
        'hasNftAvatar': !(0, runtime_1.exists)(json, 'has_nft_avatar') ? undefined : json['has_nft_avatar'],
        'id': json['id'],
        'isBlueVerified': json['is_blue_verified'],
        'legacy': (0, UserLegacy_1.UserLegacyFromJSON)(json['legacy']),
        'restId': json['rest_id'],
        'superFollowEligible': json['super_follow_eligible'],
        'superFollowedBy': json['super_followed_by'],
        'superFollowing': json['super_following'],
    };
}
exports.UserFromJSONTyped = UserFromJSONTyped;
function UserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'affiliates_highlighted_label': value.affiliatesHighlightedLabel,
        'business_account': value.businessAccount,
        'has_graduated_access': value.hasGraduatedAccess,
        'has_nft_avatar': value.hasNftAvatar,
        'id': value.id,
        'is_blue_verified': value.isBlueVerified,
        'legacy': (0, UserLegacy_1.UserLegacyToJSON)(value.legacy),
        'rest_id': value.restId,
        'super_follow_eligible': value.superFollowEligible,
        'super_followed_by': value.superFollowedBy,
        'super_following': value.superFollowing,
    };
}
exports.UserToJSON = UserToJSON;
