// twitter-openapi-typescript-generated/dist/models/UsersResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersResponseDataToJSON = exports.UsersResponseDataFromJSONTyped = exports.UsersResponseDataFromJSON = exports.instanceOfUsersResponseData = void 0;
var UserResults_1 = require("./UserResults");
/**
 * Check if a given object implements the UsersResponseData interface.
 */
function instanceOfUsersResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "users" in value;
    return isInstance;
}
exports.instanceOfUsersResponseData = instanceOfUsersResponseData;
function UsersResponseDataFromJSON(json) {
    return UsersResponseDataFromJSONTyped(json, false);
}
exports.UsersResponseDataFromJSON = UsersResponseDataFromJSON;
function UsersResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'users': (json['users'].map(UserResults_1.UserResultsFromJSON)),
    };
}
exports.UsersResponseDataFromJSONTyped = UsersResponseDataFromJSONTyped;
function UsersResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'users': (value.users.map(UserResults_1.UserResultsToJSON)),
    };
}
exports.UsersResponseDataToJSON = UsersResponseDataToJSON;
