// twitter-openapi-typescript-generated/dist/models/UserHighlightsTweetsResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHighlightsTweetsResultToJSON = exports.UserHighlightsTweetsResultFromJSONTyped = exports.UserHighlightsTweetsResultFromJSON = exports.instanceOfUserHighlightsTweetsResult = void 0;
var TypeName_1 = require("./TypeName");
var UserHighlightsTweetsTimeline_1 = require("./UserHighlightsTweetsTimeline");
/**
 * Check if a given object implements the UserHighlightsTweetsResult interface.
 */
function instanceOfUserHighlightsTweetsResult(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "timeline" in value;
    return isInstance;
}
exports.instanceOfUserHighlightsTweetsResult = instanceOfUserHighlightsTweetsResult;
function UserHighlightsTweetsResultFromJSON(json) {
    return UserHighlightsTweetsResultFromJSONTyped(json, false);
}
exports.UserHighlightsTweetsResultFromJSON = UserHighlightsTweetsResultFromJSON;
function UserHighlightsTweetsResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'timeline': (0, UserHighlightsTweetsTimeline_1.UserHighlightsTweetsTimelineFromJSON)(json['timeline']),
    };
}
exports.UserHighlightsTweetsResultFromJSONTyped = UserHighlightsTweetsResultFromJSONTyped;
function UserHighlightsTweetsResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'timeline': (0, UserHighlightsTweetsTimeline_1.UserHighlightsTweetsTimelineToJSON)(value.timeline),
    };
}
exports.UserHighlightsTweetsResultToJSON = UserHighlightsTweetsResultToJSON;
