// twitter-openapi-typescript-generated/dist/models/UserResponseData.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResponseDataToJSON = exports.UserResponseDataFromJSONTyped = exports.UserResponseDataFromJSON = exports.instanceOfUserResponseData = void 0;
var UserResults_1 = require("./UserResults");
/**
 * Check if a given object implements the UserResponseData interface.
 */
function instanceOfUserResponseData(value) {
    var isInstance = true;
    isInstance = isInstance && "user" in value;
    return isInstance;
}
exports.instanceOfUserResponseData = instanceOfUserResponseData;
function UserResponseDataFromJSON(json) {
    return UserResponseDataFromJSONTyped(json, false);
}
exports.UserResponseDataFromJSON = UserResponseDataFromJSON;
function UserResponseDataFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'user': (0, UserResults_1.UserResultsFromJSON)(json['user']),
    };
}
exports.UserResponseDataFromJSONTyped = UserResponseDataFromJSONTyped;
function UserResponseDataToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'user': (0, UserResults_1.UserResultsToJSON)(value.user),
    };
}
exports.UserResponseDataToJSON = UserResponseDataToJSON;
