// twitter-openapi-typescript-generated/dist/models/UsersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersResponseToJSON = exports.UsersResponseFromJSONTyped = exports.UsersResponseFromJSON = exports.instanceOfUsersResponse = void 0;
var UsersResponseData_1 = require("./UsersResponseData");
/**
 * Check if a given object implements the UsersResponse interface.
 */
function instanceOfUsersResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfUsersResponse = instanceOfUsersResponse;
function UsersResponseFromJSON(json) {
    return UsersResponseFromJSONTyped(json, false);
}
exports.UsersResponseFromJSON = UsersResponseFromJSON;
function UsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, UsersResponseData_1.UsersResponseDataFromJSON)(json['data']),
    };
}
exports.UsersResponseFromJSONTyped = UsersResponseFromJSONTyped;
function UsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, UsersResponseData_1.UsersResponseDataToJSON)(value.data),
    };
}
exports.UsersResponseToJSON = UsersResponseToJSON;
