// twitter-openapi-typescript-generated/dist/models/UserHighlightsTweetsTimeline.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHighlightsTweetsTimelineToJSON = exports.UserHighlightsTweetsTimelineFromJSONTyped = exports.UserHighlightsTweetsTimelineFromJSON = exports.instanceOfUserHighlightsTweetsTimeline = void 0;
var Timeline_1 = require("./Timeline");
/**
 * Check if a given object implements the UserHighlightsTweetsTimeline interface.
 */
function instanceOfUserHighlightsTweetsTimeline(value) {
    var isInstance = true;
    isInstance = isInstance && "timeline" in value;
    return isInstance;
}
exports.instanceOfUserHighlightsTweetsTimeline = instanceOfUserHighlightsTweetsTimeline;
function UserHighlightsTweetsTimelineFromJSON(json) {
    return UserHighlightsTweetsTimelineFromJSONTyped(json, false);
}
exports.UserHighlightsTweetsTimelineFromJSON = UserHighlightsTweetsTimelineFromJSON;
function UserHighlightsTweetsTimelineFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'timeline': (0, Timeline_1.TimelineFromJSON)(json['timeline']),
    };
}
exports.UserHighlightsTweetsTimelineFromJSONTyped = UserHighlightsTweetsTimelineFromJSONTyped;
function UserHighlightsTweetsTimelineToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'timeline': (0, Timeline_1.TimelineToJSON)(value.timeline),
    };
}
exports.UserHighlightsTweetsTimelineToJSON = UserHighlightsTweetsTimelineToJSON;
