// twitter-openapi-typescript-generated/dist/models/UserTweetsResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTweetsResultToJSON = exports.UserTweetsResultFromJSONTyped = exports.UserTweetsResultFromJSON = exports.instanceOfUserTweetsResult = void 0;
var TimelineV2_1 = require("./TimelineV2");
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the UserTweetsResult interface.
 */
function instanceOfUserTweetsResult(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "timelineV2" in value;
    return isInstance;
}
exports.instanceOfUserTweetsResult = instanceOfUserTweetsResult;
function UserTweetsResultFromJSON(json) {
    return UserTweetsResultFromJSONTyped(json, false);
}
exports.UserTweetsResultFromJSON = UserTweetsResultFromJSON;
function UserTweetsResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'timelineV2': (0, TimelineV2_1.TimelineV2FromJSON)(json['timeline_v2']),
    };
}
exports.UserTweetsResultFromJSONTyped = UserTweetsResultFromJSONTyped;
function UserTweetsResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'timeline_v2': (0, TimelineV2_1.TimelineV2ToJSON)(value.timelineV2),
    };
}
exports.UserTweetsResultToJSON = UserTweetsResultToJSON;
