// twitter-openapi-typescript-generated/dist/models/UserHighlightsTweetsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHighlightsTweetsResponseToJSON = exports.UserHighlightsTweetsResponseFromJSONTyped = exports.UserHighlightsTweetsResponseFromJSON = exports.instanceOfUserHighlightsTweetsResponse = void 0;
var UserHighlightsTweetsData_1 = require("./UserHighlightsTweetsData");
/**
 * Check if a given object implements the UserHighlightsTweetsResponse interface.
 */
function instanceOfUserHighlightsTweetsResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfUserHighlightsTweetsResponse = instanceOfUserHighlightsTweetsResponse;
function UserHighlightsTweetsResponseFromJSON(json) {
    return UserHighlightsTweetsResponseFromJSONTyped(json, false);
}
exports.UserHighlightsTweetsResponseFromJSON = UserHighlightsTweetsResponseFromJSON;
function UserHighlightsTweetsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, UserHighlightsTweetsData_1.UserHighlightsTweetsDataFromJSON)(json['data']),
    };
}
exports.UserHighlightsTweetsResponseFromJSONTyped = UserHighlightsTweetsResponseFromJSONTyped;
function UserHighlightsTweetsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, UserHighlightsTweetsData_1.UserHighlightsTweetsDataToJSON)(value.data),
    };
}
exports.UserHighlightsTweetsResponseToJSON = UserHighlightsTweetsResponseToJSON;
