// twitter-openapi-typescript-generated/dist/models/UserResults.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResultsToJSON = exports.UserResultsFromJSONTyped = exports.UserResultsFromJSON = exports.instanceOfUserResults = void 0;
var runtime_1 = require("../runtime");
var UserUnion_1 = require("./UserUnion");
/**
 * Check if a given object implements the UserResults interface.
 */
function instanceOfUserResults(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserResults = instanceOfUserResults;
function UserResultsFromJSON(json) {
    return UserResultsFromJSONTyped(json, false);
}
exports.UserResultsFromJSON = UserResultsFromJSON;
function UserResultsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': !(0, runtime_1.exists)(json, 'result') ? undefined : (0, UserUnion_1.UserUnionFromJSON)(json['result']),
    };
}
exports.UserResultsFromJSONTyped = UserResultsFromJSONTyped;
function UserResultsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, UserUnion_1.UserUnionToJSON)(value.result),
    };
}
exports.UserResultsToJSON = UserResultsToJSON;
