// twitter-openapi-typescript-generated/dist/models/UserResultByScreenNameResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResultByScreenNameResultToJSON = exports.UserResultByScreenNameResultFromJSONTyped = exports.UserResultByScreenNameResultFromJSON = exports.instanceOfUserResultByScreenNameResult = void 0;
var TypeName_1 = require("./TypeName");
var UserResultByScreenNameLegacy_1 = require("./UserResultByScreenNameLegacy");
/**
 * Check if a given object implements the UserResultByScreenNameResult interface.
 */
function instanceOfUserResultByScreenNameResult(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "legacy" in value;
    isInstance = isInstance && "profilemodules" in value;
    isInstance = isInstance && "restId" in value;
    return isInstance;
}
exports.instanceOfUserResultByScreenNameResult = instanceOfUserResultByScreenNameResult;
function UserResultByScreenNameResultFromJSON(json) {
    return UserResultByScreenNameResultFromJSONTyped(json, false);
}
exports.UserResultByScreenNameResultFromJSON = UserResultByScreenNameResultFromJSON;
function UserResultByScreenNameResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'id': json['id'],
        'legacy': (0, UserResultByScreenNameLegacy_1.UserResultByScreenNameLegacyFromJSON)(json['legacy']),
        'profilemodules': json['profilemodules'],
        'restId': json['rest_id'],
    };
}
exports.UserResultByScreenNameResultFromJSONTyped = UserResultByScreenNameResultFromJSONTyped;
function UserResultByScreenNameResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'id': value.id,
        'legacy': (0, UserResultByScreenNameLegacy_1.UserResultByScreenNameLegacyToJSON)(value.legacy),
        'profilemodules': value.profilemodules,
        'rest_id': value.restId,
    };
}
exports.UserResultByScreenNameResultToJSON = UserResultByScreenNameResultToJSON;
