// twitter-openapi-typescript-generated/dist/models/UserUnavailable.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserUnavailableToJSON = exports.UserUnavailableFromJSONTyped = exports.UserUnavailableFromJSON = exports.instanceOfUserUnavailable = void 0;
var TypeName_1 = require("./TypeName");
/**
 * Check if a given object implements the UserUnavailable interface.
 */
function instanceOfUserUnavailable(value) {
    var isInstance = true;
    isInstance = isInstance && "typename" in value;
    isInstance = isInstance && "reason" in value;
    return isInstance;
}
exports.instanceOfUserUnavailable = instanceOfUserUnavailable;
function UserUnavailableFromJSON(json) {
    return UserUnavailableFromJSONTyped(json, false);
}
exports.UserUnavailableFromJSON = UserUnavailableFromJSON;
function UserUnavailableFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'typename': (0, TypeName_1.TypeNameFromJSON)(json['__typename']),
        'reason': json['reason'],
    };
}
exports.UserUnavailableFromJSONTyped = UserUnavailableFromJSONTyped;
function UserUnavailableToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '__typename': (0, TypeName_1.TypeNameToJSON)(value.typename),
        'reason': value.reason,
    };
}
exports.UserUnavailableToJSON = UserUnavailableToJSON;
