// twitter-openapi-typescript-generated/dist/models/UserResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResponseToJSON = exports.UserResponseFromJSONTyped = exports.UserResponseFromJSON = exports.instanceOfUserResponse = void 0;
var UserResponseData_1 = require("./UserResponseData");
/**
 * Check if a given object implements the UserResponse interface.
 */
function instanceOfUserResponse(value) {
    var isInstance = true;
    isInstance = isInstance && "data" in value;
    return isInstance;
}
exports.instanceOfUserResponse = instanceOfUserResponse;
function UserResponseFromJSON(json) {
    return UserResponseFromJSONTyped(json, false);
}
exports.UserResponseFromJSON = UserResponseFromJSON;
function UserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'data': (0, UserResponseData_1.UserResponseDataFromJSON)(json['data']),
    };
}
exports.UserResponseFromJSONTyped = UserResponseFromJSONTyped;
function UserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'data': (0, UserResponseData_1.UserResponseDataToJSON)(value.data),
    };
}
exports.UserResponseToJSON = UserResponseToJSON;
