// twitter-openapi-typescript-generated/dist/models/UserTweetsUser.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTweetsUserToJSON = exports.UserTweetsUserFromJSONTyped = exports.UserTweetsUserFromJSON = exports.instanceOfUserTweetsUser = void 0;
var UserTweetsResult_1 = require("./UserTweetsResult");
/**
 * Check if a given object implements the UserTweetsUser interface.
 */
function instanceOfUserTweetsUser(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfUserTweetsUser = instanceOfUserTweetsUser;
function UserTweetsUserFromJSON(json) {
    return UserTweetsUserFromJSONTyped(json, false);
}
exports.UserTweetsUserFromJSON = UserTweetsUserFromJSON;
function UserTweetsUserFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, UserTweetsResult_1.UserTweetsResultFromJSON)(json['result']),
    };
}
exports.UserTweetsUserFromJSONTyped = UserTweetsUserFromJSONTyped;
function UserTweetsUserToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, UserTweetsResult_1.UserTweetsResultToJSON)(value.result),
    };
}
exports.UserTweetsUserToJSON = UserTweetsUserToJSON;
