// @book000/twitterts/dist/scraper.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const scraper_1 = require("./scraper");
jest.setTimeout(60000);
describe('Scraper', () => {
    let scraper;
    test('login', async () => {
        const username = process.env.TWITTER_USERNAME;
        const password = process.env.TWITTER_PASSWORD;
        const otpSecret = process.env.TWITTER_OTP_SECRET;
        if (!username || !password || !otpSecret) {
            throw new Error('Failed to get environment variables.');
        }
        scraper = new scraper_1.TwitterScraper({
            username,
            password,
            otpSecret,
            puppeteerOptions: {
                userDataDirectory: './data/userdata',
            },
            debugOptions: {
                outputResponse: {
                    enable: true,
                    outputDirectory: './data/responses',
                },
            },
        });
        await expect(scraper.login()).resolves.not.toThrowError();
    });
    afterAll(async () => {
        await scraper.close();
    });
});
