// @book000/twitterts/dist/models/exceptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TwitterRateLimitError = exports.TwitterTimeoutError = exports.TwitterOperationError = exports.ResponseParseError = exports.UserNotFoundError = exports.NotBlockedError = exports.AlreadyBlockedError = exports.NotLikedError = exports.AlreadyLikedError = exports.IllegalArgumentError = exports.TwitterTsError = void 0;
/**
 * twitterts ライブラリのエラー
 */
class TwitterTsError extends Error {
}
exports.TwitterTsError = TwitterTsError;
/**
 * メソッドに無効または不適切な引数が渡された場合
 */
class IllegalArgumentError extends TwitterTsError {
}
exports.IllegalArgumentError = IllegalArgumentError;
/**
 * ユーザーがすでに「いいね！」をしているツイートに「いいね！」をしようとした場合
 */
class AlreadyLikedError extends TwitterTsError {
    constructor() {
        super('Already liked');
    }
}
exports.AlreadyLikedError = AlreadyLikedError;
/**
 * ユーザーがすでに「いいね！」をしていないツイートに「いいね！」を解除しようとした場合
 */
class NotLikedError extends TwitterTsError {
    constructor() {
        super('Not liked');
    }
}
exports.NotLikedError = NotLikedError;
/**
 * ユーザーがすでにブロックしているユーザーをブロックしようとした場合
 */
class AlreadyBlockedError extends TwitterTsError {
    constructor() {
        super('Already blocked');
    }
}
exports.AlreadyBlockedError = AlreadyBlockedError;
/**
 * ユーザーがブロックしていないユーザーをブロック解除しようとした場合
 */
class NotBlockedError extends TwitterTsError {
    constructor() {
        super('Not blocked');
    }
}
exports.NotBlockedError = NotBlockedError;
/**
 * ユーザーが存在しない場合
 */
class UserNotFoundError extends TwitterTsError {
    constructor() {
        super('User not found');
    }
}
exports.UserNotFoundError = UserNotFoundError;
/**
 * レスポンスのパースに失敗した場合
 */
class ResponseParseError extends TwitterTsError {
}
exports.ResponseParseError = ResponseParseError;
/**
 * Twitterでの操作に失敗した場合
 */
class TwitterOperationError extends TwitterTsError {
}
exports.TwitterOperationError = TwitterOperationError;
/**
 * Twitterでの操作でタイムアウトした場合
 */
class TwitterTimeoutError extends TwitterOperationError {
}
exports.TwitterTimeoutError = TwitterTimeoutError;
/**
 * Twitterのレートリミット制限に引っかかった場合
 */
class TwitterRateLimitError extends TwitterOperationError {
}
exports.TwitterRateLimitError = TwitterRateLimitError;
