// twitter-openapi-typescript-generated/dist/apis/UserListApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Twitter OpenAPI
 * Twitter OpenAPI(Swagger) specification
 *
 * The version of the OpenAPI document: 0.0.1
 * Contact: yuki@yuki0311.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListApi = void 0;
var runtime = require("../runtime");
var index_1 = require("../models/index");
/**
 *
 */
var UserListApi = /** @class */ (function (_super) {
    __extends(UserListApi, _super);
    function UserListApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * get tweet favoriters
     */
    UserListApi.prototype.getFavoritersRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.pathQueryId === null || requestParameters.pathQueryId === undefined) {
                            throw new runtime.RequiredError('pathQueryId', 'Required parameter requestParameters.pathQueryId was null or undefined when calling getFavoriters.');
                        }
                        if (requestParameters.variables === null || requestParameters.variables === undefined) {
                            throw new runtime.RequiredError('variables', 'Required parameter requestParameters.variables was null or undefined when calling getFavoriters.');
                        }
                        if (requestParameters.features === null || requestParameters.features === undefined) {
                            throw new runtime.RequiredError('features', 'Required parameter requestParameters.features was null or undefined when calling getFavoriters.');
                        }
                        queryParameters = {};
                        if (requestParameters.variables !== undefined) {
                            queryParameters['variables'] = requestParameters.variables;
                        }
                        if (requestParameters.features !== undefined) {
                            queryParameters['features'] = requestParameters.features;
                        }
                        headerParameters = {};
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-client-language"] = this.configuration.apiKey("x-twitter-client-language"); // ClientLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept"] = this.configuration.apiKey("Accept"); // Accept authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Dest"] = this.configuration.apiKey("Sec-Fetch-Dest"); // SecFetchDest authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Pragma"] = this.configuration.apiKey("Pragma"); // Pragma authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Platform"] = this.configuration.apiKey("Sec-Ch-Ua-Platform"); // SecChUaPlatform authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Mode"] = this.configuration.apiKey("Sec-Fetch-Mode"); // SecFetchMode authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-csrf-token"] = this.configuration.apiKey("x-csrf-token"); // CsrfToken authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-guest-token"] = this.configuration.apiKey("x-guest-token"); // GuestToken authentication
                        }
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("BearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2:
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua"] = this.configuration.apiKey("Sec-Ch-Ua"); // SecChUa authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-active-user"] = this.configuration.apiKey("x-twitter-active-user"); // ActiveUser authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["user-agent"] = this.configuration.apiKey("user-agent"); // UserAgent authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Language"] = this.configuration.apiKey("Accept-Language"); // AcceptLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Site"] = this.configuration.apiKey("Sec-Fetch-Site"); // SecFetchSite authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-auth-type"] = this.configuration.apiKey("x-twitter-auth-type"); // AuthType authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Cache-Control"] = this.configuration.apiKey("Cache-Control"); // CacheControl authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Mobile"] = this.configuration.apiKey("Sec-Ch-Ua-Mobile"); // SecChUaMobile authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Encoding"] = this.configuration.apiKey("Accept-Encoding"); // AcceptEncoding authentication
                        }
                        return [4 /*yield*/, this.request({
                                path: "/graphql/{pathQueryId}/Favoriters".replace("{".concat("pathQueryId", "}"), encodeURIComponent(String(requestParameters.pathQueryId))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetFavoriters200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * get tweet favoriters
     */
    UserListApi.prototype.getFavoriters = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getFavoritersRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * get user list of followers
     */
    UserListApi.prototype.getFollowersRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.pathQueryId === null || requestParameters.pathQueryId === undefined) {
                            throw new runtime.RequiredError('pathQueryId', 'Required parameter requestParameters.pathQueryId was null or undefined when calling getFollowers.');
                        }
                        if (requestParameters.variables === null || requestParameters.variables === undefined) {
                            throw new runtime.RequiredError('variables', 'Required parameter requestParameters.variables was null or undefined when calling getFollowers.');
                        }
                        if (requestParameters.features === null || requestParameters.features === undefined) {
                            throw new runtime.RequiredError('features', 'Required parameter requestParameters.features was null or undefined when calling getFollowers.');
                        }
                        queryParameters = {};
                        if (requestParameters.variables !== undefined) {
                            queryParameters['variables'] = requestParameters.variables;
                        }
                        if (requestParameters.features !== undefined) {
                            queryParameters['features'] = requestParameters.features;
                        }
                        headerParameters = {};
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-client-language"] = this.configuration.apiKey("x-twitter-client-language"); // ClientLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept"] = this.configuration.apiKey("Accept"); // Accept authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Dest"] = this.configuration.apiKey("Sec-Fetch-Dest"); // SecFetchDest authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Pragma"] = this.configuration.apiKey("Pragma"); // Pragma authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Platform"] = this.configuration.apiKey("Sec-Ch-Ua-Platform"); // SecChUaPlatform authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Mode"] = this.configuration.apiKey("Sec-Fetch-Mode"); // SecFetchMode authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-csrf-token"] = this.configuration.apiKey("x-csrf-token"); // CsrfToken authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-guest-token"] = this.configuration.apiKey("x-guest-token"); // GuestToken authentication
                        }
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("BearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2:
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua"] = this.configuration.apiKey("Sec-Ch-Ua"); // SecChUa authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-active-user"] = this.configuration.apiKey("x-twitter-active-user"); // ActiveUser authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["user-agent"] = this.configuration.apiKey("user-agent"); // UserAgent authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Language"] = this.configuration.apiKey("Accept-Language"); // AcceptLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Site"] = this.configuration.apiKey("Sec-Fetch-Site"); // SecFetchSite authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-auth-type"] = this.configuration.apiKey("x-twitter-auth-type"); // AuthType authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Cache-Control"] = this.configuration.apiKey("Cache-Control"); // CacheControl authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Mobile"] = this.configuration.apiKey("Sec-Ch-Ua-Mobile"); // SecChUaMobile authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Encoding"] = this.configuration.apiKey("Accept-Encoding"); // AcceptEncoding authentication
                        }
                        return [4 /*yield*/, this.request({
                                path: "/graphql/{pathQueryId}/Followers".replace("{".concat("pathQueryId", "}"), encodeURIComponent(String(requestParameters.pathQueryId))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetFollowers200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * get user list of followers
     */
    UserListApi.prototype.getFollowers = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getFollowersRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * get followers you know
     */
    UserListApi.prototype.getFollowersYouKnowRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.pathQueryId === null || requestParameters.pathQueryId === undefined) {
                            throw new runtime.RequiredError('pathQueryId', 'Required parameter requestParameters.pathQueryId was null or undefined when calling getFollowersYouKnow.');
                        }
                        if (requestParameters.variables === null || requestParameters.variables === undefined) {
                            throw new runtime.RequiredError('variables', 'Required parameter requestParameters.variables was null or undefined when calling getFollowersYouKnow.');
                        }
                        if (requestParameters.features === null || requestParameters.features === undefined) {
                            throw new runtime.RequiredError('features', 'Required parameter requestParameters.features was null or undefined when calling getFollowersYouKnow.');
                        }
                        queryParameters = {};
                        if (requestParameters.variables !== undefined) {
                            queryParameters['variables'] = requestParameters.variables;
                        }
                        if (requestParameters.features !== undefined) {
                            queryParameters['features'] = requestParameters.features;
                        }
                        headerParameters = {};
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-client-language"] = this.configuration.apiKey("x-twitter-client-language"); // ClientLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept"] = this.configuration.apiKey("Accept"); // Accept authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Dest"] = this.configuration.apiKey("Sec-Fetch-Dest"); // SecFetchDest authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Pragma"] = this.configuration.apiKey("Pragma"); // Pragma authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Platform"] = this.configuration.apiKey("Sec-Ch-Ua-Platform"); // SecChUaPlatform authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Mode"] = this.configuration.apiKey("Sec-Fetch-Mode"); // SecFetchMode authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-csrf-token"] = this.configuration.apiKey("x-csrf-token"); // CsrfToken authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-guest-token"] = this.configuration.apiKey("x-guest-token"); // GuestToken authentication
                        }
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("BearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2:
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua"] = this.configuration.apiKey("Sec-Ch-Ua"); // SecChUa authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-active-user"] = this.configuration.apiKey("x-twitter-active-user"); // ActiveUser authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["user-agent"] = this.configuration.apiKey("user-agent"); // UserAgent authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Language"] = this.configuration.apiKey("Accept-Language"); // AcceptLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Site"] = this.configuration.apiKey("Sec-Fetch-Site"); // SecFetchSite authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-auth-type"] = this.configuration.apiKey("x-twitter-auth-type"); // AuthType authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Cache-Control"] = this.configuration.apiKey("Cache-Control"); // CacheControl authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Mobile"] = this.configuration.apiKey("Sec-Ch-Ua-Mobile"); // SecChUaMobile authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Encoding"] = this.configuration.apiKey("Accept-Encoding"); // AcceptEncoding authentication
                        }
                        return [4 /*yield*/, this.request({
                                path: "/graphql/{pathQueryId}/FollowersYouKnow".replace("{".concat("pathQueryId", "}"), encodeURIComponent(String(requestParameters.pathQueryId))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetFollowers200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * get followers you know
     */
    UserListApi.prototype.getFollowersYouKnow = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getFollowersYouKnowRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * get user list of following
     */
    UserListApi.prototype.getFollowingRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.pathQueryId === null || requestParameters.pathQueryId === undefined) {
                            throw new runtime.RequiredError('pathQueryId', 'Required parameter requestParameters.pathQueryId was null or undefined when calling getFollowing.');
                        }
                        if (requestParameters.variables === null || requestParameters.variables === undefined) {
                            throw new runtime.RequiredError('variables', 'Required parameter requestParameters.variables was null or undefined when calling getFollowing.');
                        }
                        if (requestParameters.features === null || requestParameters.features === undefined) {
                            throw new runtime.RequiredError('features', 'Required parameter requestParameters.features was null or undefined when calling getFollowing.');
                        }
                        queryParameters = {};
                        if (requestParameters.variables !== undefined) {
                            queryParameters['variables'] = requestParameters.variables;
                        }
                        if (requestParameters.features !== undefined) {
                            queryParameters['features'] = requestParameters.features;
                        }
                        headerParameters = {};
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-client-language"] = this.configuration.apiKey("x-twitter-client-language"); // ClientLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept"] = this.configuration.apiKey("Accept"); // Accept authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Dest"] = this.configuration.apiKey("Sec-Fetch-Dest"); // SecFetchDest authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Pragma"] = this.configuration.apiKey("Pragma"); // Pragma authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Platform"] = this.configuration.apiKey("Sec-Ch-Ua-Platform"); // SecChUaPlatform authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Mode"] = this.configuration.apiKey("Sec-Fetch-Mode"); // SecFetchMode authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-csrf-token"] = this.configuration.apiKey("x-csrf-token"); // CsrfToken authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-guest-token"] = this.configuration.apiKey("x-guest-token"); // GuestToken authentication
                        }
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("BearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2:
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua"] = this.configuration.apiKey("Sec-Ch-Ua"); // SecChUa authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-active-user"] = this.configuration.apiKey("x-twitter-active-user"); // ActiveUser authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["user-agent"] = this.configuration.apiKey("user-agent"); // UserAgent authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Language"] = this.configuration.apiKey("Accept-Language"); // AcceptLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Site"] = this.configuration.apiKey("Sec-Fetch-Site"); // SecFetchSite authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-auth-type"] = this.configuration.apiKey("x-twitter-auth-type"); // AuthType authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Cache-Control"] = this.configuration.apiKey("Cache-Control"); // CacheControl authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Mobile"] = this.configuration.apiKey("Sec-Ch-Ua-Mobile"); // SecChUaMobile authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Encoding"] = this.configuration.apiKey("Accept-Encoding"); // AcceptEncoding authentication
                        }
                        return [4 /*yield*/, this.request({
                                path: "/graphql/{pathQueryId}/Following".replace("{".concat("pathQueryId", "}"), encodeURIComponent(String(requestParameters.pathQueryId))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetFollowers200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * get user list of following
     */
    UserListApi.prototype.getFollowing = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getFollowingRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * get tweet retweeters
     */
    UserListApi.prototype.getRetweetersRaw = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, token, tokenString, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.pathQueryId === null || requestParameters.pathQueryId === undefined) {
                            throw new runtime.RequiredError('pathQueryId', 'Required parameter requestParameters.pathQueryId was null or undefined when calling getRetweeters.');
                        }
                        if (requestParameters.variables === null || requestParameters.variables === undefined) {
                            throw new runtime.RequiredError('variables', 'Required parameter requestParameters.variables was null or undefined when calling getRetweeters.');
                        }
                        if (requestParameters.features === null || requestParameters.features === undefined) {
                            throw new runtime.RequiredError('features', 'Required parameter requestParameters.features was null or undefined when calling getRetweeters.');
                        }
                        queryParameters = {};
                        if (requestParameters.variables !== undefined) {
                            queryParameters['variables'] = requestParameters.variables;
                        }
                        if (requestParameters.features !== undefined) {
                            queryParameters['features'] = requestParameters.features;
                        }
                        headerParameters = {};
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-client-language"] = this.configuration.apiKey("x-twitter-client-language"); // ClientLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept"] = this.configuration.apiKey("Accept"); // Accept authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Dest"] = this.configuration.apiKey("Sec-Fetch-Dest"); // SecFetchDest authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Pragma"] = this.configuration.apiKey("Pragma"); // Pragma authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Platform"] = this.configuration.apiKey("Sec-Ch-Ua-Platform"); // SecChUaPlatform authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Mode"] = this.configuration.apiKey("Sec-Fetch-Mode"); // SecFetchMode authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-csrf-token"] = this.configuration.apiKey("x-csrf-token"); // CsrfToken authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-guest-token"] = this.configuration.apiKey("x-guest-token"); // GuestToken authentication
                        }
                        if (!(this.configuration && this.configuration.accessToken)) return [3 /*break*/, 2];
                        token = this.configuration.accessToken;
                        return [4 /*yield*/, token("BearerAuth", [])];
                    case 1:
                        tokenString = _a.sent();
                        if (tokenString) {
                            headerParameters["Authorization"] = "Bearer ".concat(tokenString);
                        }
                        _a.label = 2;
                    case 2:
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua"] = this.configuration.apiKey("Sec-Ch-Ua"); // SecChUa authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-active-user"] = this.configuration.apiKey("x-twitter-active-user"); // ActiveUser authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["user-agent"] = this.configuration.apiKey("user-agent"); // UserAgent authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Language"] = this.configuration.apiKey("Accept-Language"); // AcceptLanguage authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Fetch-Site"] = this.configuration.apiKey("Sec-Fetch-Site"); // SecFetchSite authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["x-twitter-auth-type"] = this.configuration.apiKey("x-twitter-auth-type"); // AuthType authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Cache-Control"] = this.configuration.apiKey("Cache-Control"); // CacheControl authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Sec-Ch-Ua-Mobile"] = this.configuration.apiKey("Sec-Ch-Ua-Mobile"); // SecChUaMobile authentication
                        }
                        if (this.configuration && this.configuration.apiKey) {
                            headerParameters["Accept-Encoding"] = this.configuration.apiKey("Accept-Encoding"); // AcceptEncoding authentication
                        }
                        return [4 /*yield*/, this.request({
                                path: "/graphql/{pathQueryId}/Retweeters".replace("{".concat("pathQueryId", "}"), encodeURIComponent(String(requestParameters.pathQueryId))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            }, initOverrides)];
                    case 3:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, index_1.GetRetweeters200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * get tweet retweeters
     */
    UserListApi.prototype.getRetweeters = function (requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getRetweetersRaw(requestParameters, initOverrides)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return UserListApi;
}(runtime.BaseAPI));
exports.UserListApi = UserListApi;
